{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1764fcdb-f0ca-46b3-907a-dfad53504b14",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<frozen importlib._bootstrap>:219: RuntimeWarning: scipy._lib.messagestream.MessageStream size changed, may indicate binary incompatibility. Expected 56 from C header, got 64 from PyObject\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from matplotlib import pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad3c14a8-3088-4f74-90d0-f7d64acdf00f",
   "metadata": {},
   "source": [
    "# General init (to delete what's not needed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fcde73cf-4ee9-4c58-91d2-8987f7cb14d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Names and colors\n",
    "\n",
    "colors_dict = {\"Oncodrive3D\" : \"#ffad48\", \n",
    "              \"HotMAPS\" : \"#024b79\", \n",
    "              \"OncodriveCLUSTL\" : \"#43b7c2\", \n",
    "              \"smRegions\" : \"lightgray\", \n",
    "              \"CBaSE\" : \"lightgray\", \n",
    "              \"dNdScv\" : \"lightgray\", \n",
    "              \"MutPanning\" : \"lightgray\", \n",
    "              \"OncodriveFML\" : \"lightgray\",\n",
    "              \"Random\" : \"pink\"}\n",
    "\n",
    "names_dict = {\"o3d\" : \"Oncodrive3D\", \n",
    "             \"hotmaps\" : \"HotMAPS\", \n",
    "             \"clustl\" : \"OncodriveCLUSTL\", \n",
    "             \"smreg\" : \"smRegions\", \n",
    "             \"cbase\" : \"CBaSE\", \n",
    "             \"dndscv\" : \"dNdScv\", \n",
    "             \"mutpan\" : \"MutPanning\", \n",
    "             \"fml\" : \"OncodriveFML\",\n",
    "             \"random\" : \"Random\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "69372ebb-2588-40ff-aadc-42b47a575eea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Name</th>\n",
       "      <th>Entrez GeneId</th>\n",
       "      <th>Genome Location</th>\n",
       "      <th>Tier</th>\n",
       "      <th>Hallmark</th>\n",
       "      <th>Chr Band</th>\n",
       "      <th>Somatic</th>\n",
       "      <th>Germline</th>\n",
       "      <th>Tumour Types(Somatic)</th>\n",
       "      <th>Tumour Types(Germline)</th>\n",
       "      <th>Cancer Syndrome</th>\n",
       "      <th>Tissue Type</th>\n",
       "      <th>Molecular Genetics</th>\n",
       "      <th>Role in Cancer</th>\n",
       "      <th>Mutation Types</th>\n",
       "      <th>Translocation Partner</th>\n",
       "      <th>Other Germline Mut</th>\n",
       "      <th>Other Syndrome</th>\n",
       "      <th>Synonyms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>10:50799421-50885675</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10q11.23</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oncogene</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACF,ACF64,ACF65,APOBEC1CF,ASP,CCDS73133.1,ENSG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABI1</td>\n",
       "      <td>abl-interactor 1</td>\n",
       "      <td>10006.0</td>\n",
       "      <td>10:26746593-26860935</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>10p12.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>TSG, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>KMT2A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABI-1,CCDS7150.1,E3B1,ENSG00000136754.17,NM_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>v-abl Abelson murine leukemia viral oncogene h...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>9:130713946-130885683</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>9q34.12</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CML, ALL, T-ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T, Mis</td>\n",
       "      <td>BCR, ETV6, NUP214</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABL,CCDS35165.1,ENSG00000097007.17,JTK7,NM_007...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>c-abl oncogene 2, non-receptor tyrosine kinase</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1:179099327-179229601</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1q25.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>ETV6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABLL,ARG,CCDS30947.1,ENSG00000143322.19,NM_007...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACKR3</td>\n",
       "      <td>atypical chemokine receptor 3</td>\n",
       "      <td>57007.0</td>\n",
       "      <td>2:236569641-236582358</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>2q37.3</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lipoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>HMGA2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS2516.1,CMKOR1,CXCR7,ENSG00000144476.5,GPR1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>ZNF429</td>\n",
       "      <td>zinc finger protein 429</td>\n",
       "      <td>353088.0</td>\n",
       "      <td>19:21505564-21538078</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19p12</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS42537.1,ENSG00000197013.9,NM_001001415.2,N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>739</th>\n",
       "      <td>ZNF479</td>\n",
       "      <td>zinc finger protein 479</td>\n",
       "      <td>90827.0</td>\n",
       "      <td>7:57119614-57139864</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7p11.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lung cancer, bladder carcinoma, prostate carci...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS43590.1,ENSG00000185177.12,KR19,NM_033273....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>740</th>\n",
       "      <td>ZNF521</td>\n",
       "      <td>zinc finger protein 521</td>\n",
       "      <td>25925.0</td>\n",
       "      <td>18:25061926-25352152</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18q11.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS32806.1,EHZF,ENSG00000198795.10,Evi3,NM_01...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>741</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>zinc and ring finger 3</td>\n",
       "      <td>84133.0</td>\n",
       "      <td>22:28883592-29057487</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22q12.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colorectal cancer, adrenocortical carcinoma, g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSG</td>\n",
       "      <td>N, F, Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK747E2.3,CCDS56225.1,ENSG00000183579.15,FLJ22...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>742</th>\n",
       "      <td>ZRSR2</td>\n",
       "      <td>zinc finger (CCCH type), RNA-binding motif and...</td>\n",
       "      <td>8233.0</td>\n",
       "      <td>X:15790472-15823260</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Xp22.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MDS, CLL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Rec</td>\n",
       "      <td>TSG</td>\n",
       "      <td>F, S, Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS14172.1,ENSG00000169249.12,NM_005089.3,NP_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>743 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Symbol                                               Name  \\\n",
       "0          A1CF                     APOBEC1 complementation factor   \n",
       "1          ABI1                                   abl-interactor 1   \n",
       "2          ABL1  v-abl Abelson murine leukemia viral oncogene h...   \n",
       "3          ABL2     c-abl oncogene 2, non-receptor tyrosine kinase   \n",
       "4         ACKR3                      atypical chemokine receptor 3   \n",
       "..          ...                                                ...   \n",
       "738      ZNF429                            zinc finger protein 429   \n",
       "739      ZNF479                            zinc finger protein 479   \n",
       "740      ZNF521                            zinc finger protein 521   \n",
       "741       ZNRF3                             zinc and ring finger 3   \n",
       "742       ZRSR2  zinc finger (CCCH type), RNA-binding motif and...   \n",
       "\n",
       "     Entrez GeneId        Genome Location  Tier Hallmark  Chr Band Somatic  \\\n",
       "0          29974.0   10:50799421-50885675     2      NaN  10q11.23     yes   \n",
       "1          10006.0   10:26746593-26860935     1      Yes   10p12.1     yes   \n",
       "2             25.0  9:130713946-130885683     1      Yes   9q34.12     yes   \n",
       "3             27.0  1:179099327-179229601     1      NaN    1q25.2     yes   \n",
       "4          57007.0  2:236569641-236582358     1      Yes    2q37.3     yes   \n",
       "..             ...                    ...   ...      ...       ...     ...   \n",
       "738       353088.0   19:21505564-21538078     2      NaN     19p12     yes   \n",
       "739        90827.0    7:57119614-57139864     2      NaN    7p11.2     yes   \n",
       "740        25925.0   18:25061926-25352152     1      NaN   18q11.2     yes   \n",
       "741        84133.0   22:28883592-29057487     2      NaN   22q12.1     yes   \n",
       "742         8233.0    X:15790472-15823260     1      Yes    Xp22.2     yes   \n",
       "\n",
       "    Germline                              Tumour Types(Somatic)  \\\n",
       "0        NaN                                           melanoma   \n",
       "1        NaN                                                AML   \n",
       "2        NaN                                    CML, ALL, T-ALL   \n",
       "3        NaN                                                AML   \n",
       "4        NaN                                             lipoma   \n",
       "..       ...                                                ...   \n",
       "738      NaN                                                GBM   \n",
       "739      NaN  lung cancer, bladder carcinoma, prostate carci...   \n",
       "740      NaN                                                ALL   \n",
       "741      NaN  colorectal cancer, adrenocortical carcinoma, g...   \n",
       "742      NaN                                           MDS, CLL   \n",
       "\n",
       "    Tumour Types(Germline) Cancer Syndrome Tissue Type Molecular Genetics  \\\n",
       "0                      NaN             NaN           E                NaN   \n",
       "1                      NaN             NaN           L                Dom   \n",
       "2                      NaN             NaN           L                Dom   \n",
       "3                      NaN             NaN           L                Dom   \n",
       "4                      NaN             NaN           M                Dom   \n",
       "..                     ...             ...         ...                ...   \n",
       "738                    NaN             NaN           O                NaN   \n",
       "739                    NaN             NaN           E                NaN   \n",
       "740                    NaN             NaN           L                Dom   \n",
       "741                    NaN             NaN           E                NaN   \n",
       "742                    NaN             NaN           L                Rec   \n",
       "\n",
       "       Role in Cancer Mutation Types Translocation Partner Other Germline Mut  \\\n",
       "0            oncogene            Mis                   NaN                NaN   \n",
       "1         TSG, fusion              T                 KMT2A                NaN   \n",
       "2    oncogene, fusion         T, Mis     BCR, ETV6, NUP214                NaN   \n",
       "3    oncogene, fusion              T                  ETV6                NaN   \n",
       "4    oncogene, fusion              T                 HMGA2                NaN   \n",
       "..                ...            ...                   ...                ...   \n",
       "738               NaN            Mis                   NaN                NaN   \n",
       "739               NaN            Mis                   NaN                NaN   \n",
       "740  oncogene, fusion              T                  PAX5                NaN   \n",
       "741               TSG      N, F, Mis                   NaN                NaN   \n",
       "742               TSG      F, S, Mis                   NaN                NaN   \n",
       "\n",
       "    Other Syndrome                                           Synonyms  \n",
       "0              NaN  ACF,ACF64,ACF65,APOBEC1CF,ASP,CCDS73133.1,ENSG...  \n",
       "1              NaN  ABI-1,CCDS7150.1,E3B1,ENSG00000136754.17,NM_00...  \n",
       "2              NaN  ABL,CCDS35165.1,ENSG00000097007.17,JTK7,NM_007...  \n",
       "3              NaN  ABLL,ARG,CCDS30947.1,ENSG00000143322.19,NM_007...  \n",
       "4              NaN  CCDS2516.1,CMKOR1,CXCR7,ENSG00000144476.5,GPR1...  \n",
       "..             ...                                                ...  \n",
       "738            NaN  CCDS42537.1,ENSG00000197013.9,NM_001001415.2,N...  \n",
       "739            NaN  CCDS43590.1,ENSG00000185177.12,KR19,NM_033273....  \n",
       "740            NaN  CCDS32806.1,EHZF,ENSG00000198795.10,Evi3,NM_01...  \n",
       "741            NaN  BK747E2.3,CCDS56225.1,ENSG00000183579.15,FLJ22...  \n",
       "742            NaN  CCDS14172.1,ENSG00000169249.12,NM_005089.3,NP_...  \n",
       "\n",
       "[743 rows x 20 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>PARENT</th>\n",
       "      <th>NAMES</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA</td>\n",
       "      <td>SOFT_TISSUE</td>\n",
       "      <td>Aggressive Angiomyxoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AASTR</td>\n",
       "      <td>DIFG</td>\n",
       "      <td>Anaplastic Astrocytoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABC</td>\n",
       "      <td>DLBCLNOS</td>\n",
       "      <td>Activated B-cell Type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL</td>\n",
       "      <td>AMLNOS</td>\n",
       "      <td>Acute Basophilic Leukemia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACA</td>\n",
       "      <td>ADRENAL_GLAND</td>\n",
       "      <td>Adrenocortical Adenoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>WDLS</td>\n",
       "      <td>LIPO</td>\n",
       "      <td>Well-Differentiated Liposarcoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>WDTC</td>\n",
       "      <td>THYROID</td>\n",
       "      <td>Well-Differentiated Thyroid Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>885</th>\n",
       "      <td>WM</td>\n",
       "      <td>LPL</td>\n",
       "      <td>Waldenstrom Macroglobulinemia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>WPSCC</td>\n",
       "      <td>PSCC</td>\n",
       "      <td>Warty Penile Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>WT</td>\n",
       "      <td>KIDNEY</td>\n",
       "      <td>Wilms' Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>888 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID         PARENT                                 NAMES\n",
       "0       AA    SOFT_TISSUE                Aggressive Angiomyxoma\n",
       "1    AASTR           DIFG                Anaplastic Astrocytoma\n",
       "2      ABC       DLBCLNOS                 Activated B-cell Type\n",
       "3      ABL         AMLNOS             Acute Basophilic Leukemia\n",
       "4      ACA  ADRENAL_GLAND                Adrenocortical Adenoma\n",
       "..     ...            ...                                   ...\n",
       "883   WDLS           LIPO       Well-Differentiated Liposarcoma\n",
       "884   WDTC        THYROID    Well-Differentiated Thyroid Cancer\n",
       "885     WM            LPL         Waldenstrom Macroglobulinemia\n",
       "886  WPSCC           PSCC  Warty Penile Squamous Cell Carcinoma\n",
       "887     WT         KIDNEY                          Wilms' Tumor\n",
       "\n",
       "[888 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo Symbol</th>\n",
       "      <th>OncoKB Annotated</th>\n",
       "      <th>MSK-IMPACT</th>\n",
       "      <th>MSK-HEME</th>\n",
       "      <th>FOUNDATION ONE</th>\n",
       "      <th>FOUNDATION ONE HEME</th>\n",
       "      <th>Vogelstein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALK</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AMER1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>APC</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149</th>\n",
       "      <td>ZFP36L1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150</th>\n",
       "      <td>ZFP36L2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1151</th>\n",
       "      <td>ZNF24</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1152</th>\n",
       "      <td>ZNF331</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1153</th>\n",
       "      <td>ZNF750</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1154 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo Symbol OncoKB Annotated MSK-IMPACT MSK-HEME FOUNDATION ONE  \\\n",
       "0           ABL1              Yes        Yes      Yes            Yes   \n",
       "1           AKT1              Yes        Yes      Yes            Yes   \n",
       "2            ALK              Yes        Yes      Yes            Yes   \n",
       "3          AMER1              Yes        Yes      Yes            Yes   \n",
       "4            APC              Yes        Yes      Yes            Yes   \n",
       "...          ...              ...        ...      ...            ...   \n",
       "1149     ZFP36L1              Yes         No       No             No   \n",
       "1150     ZFP36L2              Yes         No       No             No   \n",
       "1151       ZNF24               No         No       No             No   \n",
       "1152      ZNF331               No         No       No             No   \n",
       "1153      ZNF750              Yes         No       No             No   \n",
       "\n",
       "     FOUNDATION ONE HEME Vogelstein  \n",
       "0                    Yes        Yes  \n",
       "1                    Yes        Yes  \n",
       "2                    Yes        Yes  \n",
       "3                    Yes        Yes  \n",
       "4                    Yes        Yes  \n",
       "...                  ...        ...  \n",
       "1149                  No         No  \n",
       "1150                  No         No  \n",
       "1151                 Yes         No  \n",
       "1152                  No         No  \n",
       "1153                  No         No  \n",
       "\n",
       "[1154 rows x 7 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SUBDIR = \"human_mane_raw\"\n",
    "RUN = \"run_2024-07-01_16-04-14\"\n",
    "# lst_methods = [\"o3d\", \"hotmaps\", \"clustl\", \"smreg\", \"cbase\", \"seismic\", \"dndscv\", \"mutpan\", \"fml\", \"random\"]\n",
    "# lst_methods = [\"o3d\", \"hotmaps\", \"clustl\", \"random\"]\n",
    "\n",
    "datasets = f\"/workspace/projects/clustering_3d/o3d_analysys/datasets/\"\n",
    "input_path = f\"{datasets}/input/cancer_202404\"\n",
    "cohort_df = pd.read_csv(f\"{input_path}/cohorts.tsv\", sep=\"\\t\")\n",
    "cohort_df = cohort_df[~cohort_df.COHORT.str.startswith(\"ICGC\")].reset_index(drop=True)\n",
    "o3d_output_dir = f\"{datasets}/output/cancer_202404/o3d_output/{SUBDIR}/{RUN}\"\n",
    "intogen_output = f\"{datasets}/output/cancer_202404/intogen_output\"\n",
    "genetable_path = f\"/workspace/projects/clustering_3d/o3d_analysys/o3d_paper/data/results/{SUBDIR}/{RUN}.genetable.tsv\"\n",
    "drivers_df = pd.read_table(f\"{input_path}/drivers.tsv\")\n",
    "uniq_drivers_df = pd.read_table(f\"{input_path}/unique_drivers.tsv\")\n",
    "\n",
    "cgc_df = pd.read_csv(f\"{input_path}/CGC_list_v99.tsv\", sep = \"\\t\")\n",
    "display(cgc_df)\n",
    "\n",
    "oncotree_df = pd.read_csv(f\"{datasets}/input/cancer_202404/oncotree.tsv\", sep=\"\\t\")\n",
    "display(oncotree_df)\n",
    "\n",
    "oncokb_df = pd.read_csv(f\"{input_path}/oncokb_list_20240712.tsv\", sep = \"\\t\")\n",
    "oncokb_df = oncokb_df[[\"Hugo Symbol\", \"OncoKB Annotated\", \"MSK-IMPACT\", \"MSK-HEME\", \"FOUNDATION ONE\", \"FOUNDATION ONE HEME\", \"Vogelstein\"]]\n",
    "oncokb_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e6325b9f-9ae8-47fd-a2cf-a88ce1febae9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>o3d_status</th>\n",
       "      <th>CGC</th>\n",
       "      <th>Fish</th>\n",
       "      <th>o3d_rank</th>\n",
       "      <th>o3d_pval</th>\n",
       "      <th>o3d_qval</th>\n",
       "      <th>cbase_rank</th>\n",
       "      <th>cbase_pval</th>\n",
       "      <th>...</th>\n",
       "      <th>clustl_pval</th>\n",
       "      <th>clustl_qval</th>\n",
       "      <th>fml_rank</th>\n",
       "      <th>fml_pval</th>\n",
       "      <th>fml_qval</th>\n",
       "      <th>smreg_rank</th>\n",
       "      <th>smreg_pval</th>\n",
       "      <th>smreg_qval</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Cohort</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3062778</th>\n",
       "      <td>NFE2L2</td>\n",
       "      <td>Q16236</td>\n",
       "      <td>Processed</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002892</td>\n",
       "      <td>0.380525</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.001441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>TCGA_WXS_LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3062779</th>\n",
       "      <td>TP53</td>\n",
       "      <td>K7PPA8</td>\n",
       "      <td>Processed</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>0.073971</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.001441</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.534979e-56</td>\n",
       "      <td>1.202361e-51</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>TCGA_WXS_LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3062780</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>P42336</td>\n",
       "      <td>Processed</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.007465</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.000310</td>\n",
       "      <td>0.148955</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.013395e-12</td>\n",
       "      <td>5.052445e-08</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>TCGA_WXS_LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3062781</th>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>K7PML8</td>\n",
       "      <td>Processed</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000399</td>\n",
       "      <td>0.266155</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.001441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>TCGA_WXS_LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3062782</th>\n",
       "      <td>HRAS</td>\n",
       "      <td>P01112</td>\n",
       "      <td>Processed</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>990.0</td>\n",
       "      <td>0.033670</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003380</td>\n",
       "      <td>0.380525</td>\n",
       "      <td>2313.0</td>\n",
       "      <td>0.129510</td>\n",
       "      <td>0.927878</td>\n",
       "      <td>2046.0</td>\n",
       "      <td>1.413890e-01</td>\n",
       "      <td>7.547557e-01</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>TCGA_WXS_LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4582141</th>\n",
       "      <td>XKRY2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGNG</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4582142</th>\n",
       "      <td>BCORL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGNG</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4582143</th>\n",
       "      <td>CYorf15A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGNG</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4582144</th>\n",
       "      <td>CYorf15B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGNG</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4582145</th>\n",
       "      <td>SELENOF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGNG</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>686729 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene Uniprot_ID o3d_status  CGC  Fish  o3d_rank  o3d_pval  \\\n",
       "3062778    NFE2L2     Q16236  Processed    1     0       0.0       0.0   \n",
       "3062779      TP53     K7PPA8  Processed    1     0       1.0       0.0   \n",
       "3062780    PIK3CA     P42336  Processed    1     0       2.0       0.0   \n",
       "3062781    CDKN2A     K7PML8  Processed    1     0       3.0       0.0   \n",
       "3062782      HRAS     P01112  Processed    1     0       4.0       0.0   \n",
       "...           ...        ...        ...  ...   ...       ...       ...   \n",
       "4582141     XKRY2        NaN        NaN    0     1       NaN       NaN   \n",
       "4582142    BCORL2        NaN        NaN    0     0       NaN       NaN   \n",
       "4582143  CYorf15A        NaN        NaN    0     0       NaN       NaN   \n",
       "4582144  CYorf15B        NaN        NaN    0     0       NaN       NaN   \n",
       "4582145   SELENOF        NaN        NaN    0     0       NaN       NaN   \n",
       "\n",
       "         o3d_qval  cbase_rank  cbase_pval  ...  clustl_pval  clustl_qval  \\\n",
       "3062778       0.0         6.0    0.000000  ...     0.002892     0.380525   \n",
       "3062779       0.0        11.0    0.000000  ...     0.000055     0.073971   \n",
       "3062780       0.0        53.0    0.000388  ...     0.000004     0.007465   \n",
       "3062781       0.0         5.0    0.000000  ...     0.000399     0.266155   \n",
       "3062782       0.0       990.0    0.033670  ...     0.003380     0.380525   \n",
       "...           ...         ...         ...  ...          ...          ...   \n",
       "4582141       NaN         NaN         NaN  ...          NaN          NaN   \n",
       "4582142       NaN         NaN         NaN  ...          NaN          NaN   \n",
       "4582143       NaN         NaN         NaN  ...          NaN          NaN   \n",
       "4582144       NaN         NaN         NaN  ...          NaN          NaN   \n",
       "4582145       NaN         NaN         NaN  ...          NaN          NaN   \n",
       "\n",
       "         fml_rank  fml_pval  fml_qval  smreg_rank    smreg_pval    smreg_qval  \\\n",
       "3062778       3.0  0.000001  0.001441         NaN           NaN           NaN   \n",
       "3062779       1.0  0.000001  0.001441         0.0  9.534979e-56  1.202361e-51   \n",
       "3062780      29.0  0.000310  0.148955         1.0  8.013395e-12  5.052445e-08   \n",
       "3062781       5.0  0.000001  0.001441         NaN           NaN           NaN   \n",
       "3062782    2313.0  0.129510  0.927878      2046.0  1.413890e-01  7.547557e-01   \n",
       "...           ...       ...       ...         ...           ...           ...   \n",
       "4582141       NaN       NaN       NaN         NaN           NaN           NaN   \n",
       "4582142       NaN       NaN       NaN         NaN           NaN           NaN   \n",
       "4582143       NaN       NaN       NaN         NaN           NaN           NaN   \n",
       "4582144       NaN       NaN       NaN         NaN           NaN           NaN   \n",
       "4582145       NaN       NaN       NaN         NaN           NaN           NaN   \n",
       "\n",
       "         Cancer         Cohort  \n",
       "3062778    LUSC  TCGA_WXS_LUSC  \n",
       "3062779    LUSC  TCGA_WXS_LUSC  \n",
       "3062780    LUSC  TCGA_WXS_LUSC  \n",
       "3062781    LUSC  TCGA_WXS_LUSC  \n",
       "3062782    LUSC  TCGA_WXS_LUSC  \n",
       "...         ...            ...  \n",
       "4582141    PGNG  TCGA_WXS_PGNG  \n",
       "4582142    PGNG  TCGA_WXS_PGNG  \n",
       "4582143    PGNG  TCGA_WXS_PGNG  \n",
       "4582144    PGNG  TCGA_WXS_PGNG  \n",
       "4582145    PGNG  TCGA_WXS_PGNG  \n",
       "\n",
       "[686729 rows x 31 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genetable = pd.read_csv(genetable_path, sep=\"\\t\", dtype={1 : str, 2 : str, 3 : int})\n",
    "genetable = genetable[[not cohort.startswith(\"ICGC\") for cohort in genetable[\"Cohort\"]]].reset_index(drop=True)\n",
    "genetable_subset = genetable[genetable[\"Cohort\"].str.startswith(\"TCGA\")]\n",
    "genetable_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "55df6679-9d67-4335-84cc-0ce3e4cc98e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['BLCA', 'EGC', 'LMS', 'STAD', 'MEL', 'AML', 'HCC', 'UCEC', 'EPM',\n",
       "       'COADREAD', 'GB', 'ANSC', 'RCC', 'OS', 'BCC', 'RMS', 'PRAD', 'ACC',\n",
       "       'WDTC', 'ES', 'UTUC', 'GIST', 'LIPO', 'MBL', 'ANGS', 'GBC', 'ALL',\n",
       "       'LUNG', 'BRCA', 'RBL', 'CLLSLL', 'BLADDER', 'SCLC', 'NSCLC',\n",
       "       'STOMACH', 'SKCM', 'NETNOS', 'SOFT_TISSUE', 'SARCNOS', 'COAD',\n",
       "       'HNSC', 'SIC', 'CHOL', 'ESCC', 'DLBCLNOS', 'GBM', 'HGGNOS', 'NBL',\n",
       "       'ESCA', 'CSCC', 'SKIN', 'LUAD', 'CEAD', 'PLMESO', 'LGGNOS', 'PAAD',\n",
       "       'OVT', 'LUSC', 'SACA', 'CESC', 'CCRCC', 'PANCREAS', 'ACYC', 'MT',\n",
       "       'PROSTATE', 'READ', 'PCM', 'BL', 'VULVA', 'PAST', 'ATRT', 'PANET',\n",
       "       'NHL', 'CHRCC', 'MDS', 'NPC', 'LIHB', 'WT', 'MGCT', 'UCS', 'UM',\n",
       "       'THYM', 'PRCC', 'PGNG'], dtype=object)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cohort_df.CANCER_TYPE.unique())\n",
    "cohort_df.CANCER_TYPE.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cccc8d6-1456-41f9-9596-5876e6c84bc2",
   "metadata": {},
   "source": [
    "# Get data\n",
    "__I need:__\n",
    "* Clustering scores of a given genes across all cohorts\n",
    "* Cluster of a given gene across all cohorts\n",
    "* Missense mutations probability of a given genes across all cohorts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "51f18fee-3af9-4705-8807-c54b7cbe94c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/workspace/projects/clustering_3d/o3d_analysys/datasets//output/cancer_202404/o3d_output/human_mane_raw/run_2024-07-01_16-04-14'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o3d_output_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "508c72a7-0f8a-4d1b-92f8-b27003a3c7b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_55141/1608040957.py:19: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df_cohort = pd.read_csv(path_csv)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path doesn't exist: /workspace/projects/clustering_3d/o3d_analysys/datasets//output/cancer_202404/o3d_output/human_mane_raw/run_2024-07-01_16-04-14/PEDCBIOP_WXS_TALL_REL/PEDCBIOP_WXS_TALL_REL.3d_clustering_pos.csv\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>F</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Mut_in_gene</th>\n",
       "      <th>Score_obs_sim</th>\n",
       "      <th>C</th>\n",
       "      <th>C_ext</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Cohort</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1047</th>\n",
       "      <td>FBXW7</td>\n",
       "      <td>Q969H0</td>\n",
       "      <td>1</td>\n",
       "      <td>505</td>\n",
       "      <td>8</td>\n",
       "      <td>3.204794</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HARTWIG_WGS_BLCA_2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1048</th>\n",
       "      <td>FBXW7</td>\n",
       "      <td>Q969H0</td>\n",
       "      <td>1</td>\n",
       "      <td>465</td>\n",
       "      <td>8</td>\n",
       "      <td>3.409612</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HARTWIG_WGS_BLCA_2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1049</th>\n",
       "      <td>FBXW7</td>\n",
       "      <td>Q969H0</td>\n",
       "      <td>1</td>\n",
       "      <td>176</td>\n",
       "      <td>8</td>\n",
       "      <td>0.410806</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HARTWIG_WGS_BLCA_2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1050</th>\n",
       "      <td>FBXW7</td>\n",
       "      <td>Q969H0</td>\n",
       "      <td>1</td>\n",
       "      <td>280</td>\n",
       "      <td>8</td>\n",
       "      <td>0.390281</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HARTWIG_WGS_BLCA_2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>P22607</td>\n",
       "      <td>1</td>\n",
       "      <td>248</td>\n",
       "      <td>17</td>\n",
       "      <td>4.734346</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HARTWIG_WGS_BLCA_2023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EPAS1</td>\n",
       "      <td>Q99814</td>\n",
       "      <td>1</td>\n",
       "      <td>531</td>\n",
       "      <td>8</td>\n",
       "      <td>5.687067</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EPAS1</td>\n",
       "      <td>Q99814</td>\n",
       "      <td>1</td>\n",
       "      <td>530</td>\n",
       "      <td>8</td>\n",
       "      <td>6.057751</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EPAS1</td>\n",
       "      <td>Q99814</td>\n",
       "      <td>1</td>\n",
       "      <td>532</td>\n",
       "      <td>8</td>\n",
       "      <td>6.386988</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HRAS</td>\n",
       "      <td>P01112</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>18</td>\n",
       "      <td>7.463746</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HRAS</td>\n",
       "      <td>P01112</td>\n",
       "      <td>1</td>\n",
       "      <td>61</td>\n",
       "      <td>18</td>\n",
       "      <td>6.826611</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA_WXS_PGNG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7252 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Uniprot_ID  F  Pos  Mut_in_gene  Score_obs_sim  C  C_ext  Cancer  \\\n",
       "1047  FBXW7     Q969H0  1  505            8       3.204794  1    0.0     NaN   \n",
       "1048  FBXW7     Q969H0  1  465            8       3.409612  1    0.0     NaN   \n",
       "1049  FBXW7     Q969H0  1  176            8       0.410806  0    NaN     NaN   \n",
       "1050  FBXW7     Q969H0  1  280            8       0.390281  0    NaN     NaN   \n",
       "1057  FGFR3     P22607  1  248           17       4.734346  1    0.0     NaN   \n",
       "...     ...        ... ..  ...          ...            ... ..    ...     ...   \n",
       "1     EPAS1     Q99814  1  531            8       5.687067  1    0.0     NaN   \n",
       "2     EPAS1     Q99814  1  530            8       6.057751  1    0.0     NaN   \n",
       "3     EPAS1     Q99814  1  532            8       6.386988  1    0.0     NaN   \n",
       "5      HRAS     P01112  1   13           18       7.463746  1    0.0     NaN   \n",
       "6      HRAS     P01112  1   61           18       6.826611  1    0.0     NaN   \n",
       "\n",
       "                     Cohort  \n",
       "1047  HARTWIG_WGS_BLCA_2023  \n",
       "1048  HARTWIG_WGS_BLCA_2023  \n",
       "1049  HARTWIG_WGS_BLCA_2023  \n",
       "1050  HARTWIG_WGS_BLCA_2023  \n",
       "1057  HARTWIG_WGS_BLCA_2023  \n",
       "...                     ...  \n",
       "1             TCGA_WXS_PGNG  \n",
       "2             TCGA_WXS_PGNG  \n",
       "3             TCGA_WXS_PGNG  \n",
       "5             TCGA_WXS_PGNG  \n",
       "6             TCGA_WXS_PGNG  \n",
       "\n",
       "[7252 rows x 10 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_o3d_result(cohorts_df, o3d_output_dir):\n",
    "\n",
    "    lst_dfs = []\n",
    "\n",
    "    for cohort in cohorts_df[\"COHORT\"].values:\n",
    "        cohort = \"TCGA_WXS_SOFT_TISSUE\" if cohort == \"TCGA_WXS_SARC\" else cohort\n",
    "        \n",
    "        path_csv = f\"{o3d_output_dir}/{cohort}/{cohort}.3d_clustering_pos.csv\"\n",
    "        path_genes_csv = f\"{o3d_output_dir}/{cohort}/{cohort}.3d_clustering_genes.csv\"\n",
    "        if os.path.exists(path_csv) and os.path.exists(path_genes_csv):\n",
    "\n",
    "            # Get detected genes\n",
    "            df_genes_cohort = pd.read_csv(path_genes_csv)\n",
    "            df_genes_cohort = df_genes_cohort[df_genes_cohort[\"C_gene\"] == 1] \n",
    "            genes_detected = df_genes_cohort.apply(lambda x: x.Gene, axis=1).values\n",
    "\n",
    "            if len(genes_detected) > 0:\n",
    "                # Get per-residue result\n",
    "                df_cohort = pd.read_csv(path_csv)\n",
    "                cols = [\"Gene\", \"Uniprot_ID\", \"F\", \"Pos\", \"Mut_in_gene\", \"Score_obs_sim\", \"C\", \"C_ext\", 'Cancer', 'Cohort'] #\"Mut_in_res\", \"Mut_in_vol\", \"pval\"\n",
    "                df_cohort = df_cohort.loc[df_cohort.Gene.isin(genes_detected), cols]\n",
    "                lst_dfs.append(df_cohort)\n",
    "                \n",
    "        else:\n",
    "            print(f\"Path doesn't exist: {path_csv}\")\n",
    "\n",
    "    return pd.concat(lst_dfs)\n",
    "\n",
    "o3d_pos_result = get_o3d_result(cohort_df, o3d_output_dir)\n",
    "o3d_pos_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "a5ef9a2b-b41a-4b11-8160-b43d54eb35bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "273\n",
      "135\n"
     ]
    }
   ],
   "source": [
    "print(len(o3d_pos_result.Gene.unique()))\n",
    "print(len(o3d_pos_result.Cohort.unique()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ce8dcb4-4eaf-4dd2-9fb8-1944e25b7e4a",
   "metadata": {},
   "source": [
    "## To do:\n",
    "* I got a panda df including the per-residue result of each gene-cohort detected as significant\n",
    "* Now I should get something like a dictionary gene_dict[gene] : gene[cohort] : gene_cohort[score, miss_prob, cluster, mut_in_gene]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aff9e6f1-7ddc-4658-8dd9-8263940cf116",
   "metadata": {},
   "outputs": [],
   "source": [
    "# From BLCA_NORMAL (to del)\n",
    "\n",
    "o3d_seq_df = pd.read_table(f\"{o3d_datasets}/seq_for_mut_prob.tsv\")\n",
    "o3d_annot_df = pd.read_csv(f\"{o3d_annotations}/uniprot_feat.tsv\", sep=\"\\t\")\n",
    "\n",
    "o3d_prob = f\"{deepcsa_run_dir}/oncodrive3d/run/{run_name}/{run_name}.miss_prob.processed.json\"\n",
    "o3d_prob = json.load(open(o3d_prob, encoding=\"utf-8\"))\n",
    "\n",
    "o3d_score = f\"{deepcsa_run_dir}/oncodrive3d/run/{run_name}/{run_name}.3d_clustering_pos.csv\"\n",
    "o3d_score = pd.read_csv(o3d_score)[[\"Gene\", \"Pos\", \"Score\", \"Score_obs_sim\", \"pval\", \"C\", \"C_ext\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8201d67-90f0-4049-8e6a-b1f4e6c4869b",
   "metadata": {},
   "source": [
    "# Plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bf5099b-938d-4fae-b554-2d7a5d659fc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Plot\n",
    "# ====\n",
    "\n",
    "def init_plot_pars():\n",
    "    \n",
    "    plot_pars = {\"fsize\"                : (20,10),\n",
    "                 \"hspace\"               : 0.1,                    # General space between all tracks\n",
    "                 \"ofml_cbar_coord\"      : (0.01, 0.35, 0.9, 1),   # Box to anchor coordinate for OncodriveFML color bar\n",
    "                 \"track_title_x_coord\"  : 0.83,                   # x-coordinate (respect to protein len) for track txt title\n",
    "                 \"score_txt_x_coord\"    : 1.13,                   # as track title but for track score txt\n",
    "                 \"track_title_fontsize\" : 14,\n",
    "                 \"ylabel_fontsize\"      : 13.5,\n",
    "                 \"xlabel_fontsize\"      : 13.5,\n",
    "                 \"ylabel_pad\"           : 38,\n",
    "                 \"ticksize\"             : 10.5,\n",
    "                 \"legend_fontsize\"      : 12,\n",
    "                 \"legend_frameon\"       : True,\n",
    "                 \"dpi\"                  : 300\n",
    "                }\n",
    "    plot_pars[\"colors\"] = {\"ofml\"        : \"viridis_r\",\n",
    "                           \"omega_trunc\" : \"#FA5E32\",\n",
    "                           \"omega_synon\" : \"#89E4A2\",\n",
    "                           \"omega_miss\"  : \"#FABE4A\",\n",
    "                           \"o3d_score\"   : \"#6DBDCC\",\n",
    "                           \"o3d_cluster\" : \"skyblue\",\n",
    "                           \"o3d_prob\"    : \"darkgray\",\n",
    "                           \"frameshift\"  : \"#E4ACF4\",\n",
    "                           \"inframe\"     : \"C5\",\n",
    "                           \"hv_lines\"    : \"lightgray\"      # General horizontal and vertical lines (e.g., needle plot vline)\n",
    "                           }\n",
    "    plot_pars[\"h_ratios\"] = {\"omega_trunc\" : 0.3,\n",
    "                             \"omega_synon\" : 0.3,\n",
    "                             \"omega_miss\"  : 0.3,\n",
    "                             \"space1\"      : 0.015,\n",
    "                             \"o3d\"         : 0.5,\n",
    "                             \"space2\"      : 0.015,\n",
    "                             \"ofml\"        : 0.5,\n",
    "                             \"space3\"      : 0.015,\n",
    "                             \"indels\"      : 0.5,\n",
    "                             \"space4\"      : 0.015,\n",
    "                             \"domain\"      : 0.07\n",
    "                            }\n",
    "    \n",
    "    return plot_pars\n",
    "\n",
    "\n",
    "def plot_count_track(pos_track, \n",
    "                     axes, \n",
    "                     ax=0, \n",
    "                     neg_track=None, \n",
    "                     gene_len=None,\n",
    "                     col_pos_track=\"gray\",\n",
    "                     col_neg_track=\"gray\",\n",
    "                     label_pos_track=None,\n",
    "                     label_neg_track=None,\n",
    "                     ymargin=None,\n",
    "                     hv_lines=\"lightgray\"):\n",
    "    \n",
    "    axes[ax].vlines(pos_track[\"Pos\"], ymin=0, ymax=pos_track[\"Count\"], color=hv_lines, lw=1, zorder=1, alpha=0.5)\n",
    "    axes[ax].scatter(pos_track[\"Pos\"], pos_track[\"Count\"], color='white', zorder=3, lw=1, ec=\"white\") # To cover the overlapping needle top part\n",
    "    axes[ax].scatter(pos_track[\"Pos\"].values, pos_track[\"Count\"].values, color=col_pos_track, zorder=4, \n",
    "                     alpha=0.7, lw=0.1, ec=\"black\", s=60, label=label_pos_track)\n",
    "    \n",
    "    if isinstance(neg_track, pd.DataFrame):\n",
    "        if isinstance(gene_len, int):\n",
    "            axes[ax].hlines(0, xmin=0, xmax=gene_len, color=\"gray\", lw=0.6, zorder=1)\n",
    "        axes[ax].vlines(neg_track[\"Pos\"], ymin=-neg_track[\"Count\"], ymax=0, color=hv_lines, lw=1, zorder=1, alpha=0.5)\n",
    "        axes[ax].scatter(neg_track[\"Pos\"], -neg_track[\"Count\"], color='white', zorder=3, lw=1, ec=\"white\") # To cover the overlapping needle top part\n",
    "        axes[ax].scatter(pos_track[\"Pos\"].values, -neg_track[\"Count\"].values, color=col_neg_track, zorder=4, \n",
    "                         alpha=0.7, lw=0.1, ec=\"black\", s=60, label=label_neg_track)\n",
    "        axes[ax].set_yticklabels(abs(axes[ax].get_yticks()))\n",
    "        \n",
    "    if ymargin is not None:\n",
    "        axes[ax].set_ylim(-np.max(neg_track[\"Count\"])-ymargin, np.max(pos_track[\"Count\"])+ymargin)\n",
    "        \n",
    "        \n",
    "def add_ax_text(score, pvalue, y_text, x_text, ax, y_shift=0.2, y_adjust=0, equal_less=True):\n",
    "\n",
    "    ax.text(x_text, y_text+(y_text*y_shift)+y_adjust, \n",
    "            fr'$\\mathit{{Score}}$ = {np.round(score, 2)}', ha='center', va='center', fontsize=13.5, color=\"black\")\n",
    "    \n",
    "    equal = \"≤\" if equal_less else \"=\"\n",
    "    ax.text(x_text, y_text-(y_text*y_shift)+y_adjust, \n",
    "            fr'$\\mathit{{p}}$-value {equal} {pvalue}', ha='center', va='center', fontsize=13.5, color=\"black\")\n",
    "\n",
    "    \n",
    "def get_transcript_ids(gene, maf_df_f, o3d_seq_df):\n",
    "    \n",
    "    canonical_tr = maf_df_f[maf_df_f[\"SYMBOL\"] == gene].canonical_Feature.unique()[0]\n",
    "    o3d_tr = o3d_seq_df[o3d_seq_df[\"Gene\"] == gene].Ens_Transcr_ID.values[0]\n",
    "    \n",
    "    return canonical_tr, o3d_tr\n",
    "\n",
    "\n",
    "def plot_gene_selection_signals(maf_trunc_count_gene,\n",
    "                                maf_miss_count_gene,\n",
    "                                maf_synon_count_gene,\n",
    "                                frameshift_indels_count_gene,\n",
    "                                inframe_indels_count_gene,\n",
    "                                ofml_muts_score_gene,\n",
    "                                o3d_score_gene,\n",
    "                                o3d_prob_gene,\n",
    "                                plot_pars,\n",
    "                                domain_df=None,\n",
    "                                add_track_title_text=True,\n",
    "                                add_score_text=False,\n",
    "                                rm_spines=False,\n",
    "                                light_spines=False,\n",
    "                                output_path=None):\n",
    "    \n",
    "    \n",
    "    gene_len = len(o3d_prob_gene)\n",
    "    fig, axes = plt.subplots(len(plot_pars[\"h_ratios\"]), 1, \n",
    "                             figsize=plot_pars[\"fsize\"], \n",
    "                             sharex=True, \n",
    "                             gridspec_kw={'hspace': plot_pars[\"hspace\"], \n",
    "                                          'height_ratios': plot_pars[\"h_ratios\"].values()})\n",
    "    colors = plot_pars[\"colors\"]\n",
    "\n",
    "\n",
    "        \n",
    "    # Oncodrive3D\n",
    "    # ===========\n",
    "\n",
    "    ax=4\n",
    "    axes[ax].plot(range(1, gene_len+1), o3d_score_gene[\"O3D_score_norm\"], zorder=2, color=colors[\"o3d_score\"], lw=1, label=\"Clustering score\")                       \n",
    "    axes[ax].plot(range(1, gene_len+1), -o3d_prob_gene, zorder=3, color=colors[\"o3d_prob\"], lw=1, label=\"Missense mut probability\")     \n",
    "    axes[ax].fill_between(o3d_score_gene['Pos'], 0, o3d_score_gene[\"O3D_score_norm\"], where=(o3d_score_gene['Cluster'] == 1), \n",
    "                          color=colors[\"o3d_cluster\"], alpha=0.3, label='Cluster', zorder=0, lw=2)\n",
    "    if add_score_text:\n",
    "        add_ax_text(score=np.round(o3d_score, 2), \n",
    "                    pvalue=o3d_pvalue,\n",
    "                    y_text=(np.max(o3d_score_gene[\"O3D_score_norm\"]) + np.max(o3d_prob_gene))/2, \n",
    "                    x_text=gene_len*plot_pars[\"score_txt_x_coord\"], \n",
    "                    ax=axes[ax], \n",
    "                    y_shift=0.15,\n",
    "                    y_adjust=-0.0075,\n",
    "                    equal_less=True)\n",
    "    if add_track_title_text:\n",
    "        axes[ax].text(gene_len*plot_pars[\"track_title_x_coord\"], 0.015, \n",
    "                      fr'$\\mathbf{{3D}}$-$\\mathbf{{Clustering}}$', \n",
    "                      ha='center', va='center', fontsize=plot_pars[\"track_title_fontsize\"], color=\"black\")\n",
    "        \n",
    "    axes[ax].set_ylabel('Score &\\nprobability', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=45, labelpad=plot_pars[\"ylabel_pad\"], va='center')\n",
    "    axes[ax].yaxis.set_major_locator(MaxNLocator(integer=False, nbins=4))\n",
    "    tick_labels = [np.round(abs(tick), 3) for tick in axes[ax].get_yticks()]\n",
    "    axes[ax].set_yticklabels(tick_labels)\n",
    "    axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "    \n",
    "    axes[ax].legend(loc=\"upper left\", fontsize=plot_pars[\"legend_fontsize\"], frameon=plot_pars[\"legend_frameon\"])\n",
    "    #axes[ax].legend(bbox_to_anchor=[0, 0.35, 1, 0])\n",
    "    if rm_spines:\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "    elif light_spines:\n",
    "        axes[ax].spines['top'].set_color(colors[\"hv_lines\"])\n",
    "        axes[ax].spines['right'].set_color(colors[\"hv_lines\"])\n",
    "\n",
    "        \n",
    "    # OncodriveFML\n",
    "    # ============\n",
    "\n",
    "    ax=6\n",
    "    axes[ax].vlines(ofml_muts_score_gene[\"Pos\"].values, ymin=0, ymax=ofml_muts_score_gene[\"Count\"], color=colors[\"hv_lines\"], lw=1, zorder=1, alpha=0.5)\n",
    "    fml_scatter = axes[ax].scatter(ofml_muts_score_gene[\"Pos\"].values, ofml_muts_score_gene[\"Count\"].values, \n",
    "                                   c=ofml_muts_score_gene[\"CADD_score\"].values, cmap=colors[\"ofml\"], zorder=4, \n",
    "                                   alpha=0.7, lw=0.1, ec=\"black\", s=60, label=\"SNV\")\n",
    "    \n",
    "    y_text=np.round(np.max(ofml_muts_score_gene[\"Count\"])/2)\n",
    "    if add_score_text:\n",
    "        add_ax_text(score=np.round(ofml_score, 2), \n",
    "                    pvalue=ofml_pvalue,\n",
    "                    y_text=y_text, \n",
    "                    x_text=gene_len*plot_pars[\"score_txt_x_coord\"], \n",
    "                    ax=axes[ax], \n",
    "                    y_adjust=3,\n",
    "                    equal_less=True)\n",
    "\n",
    "    if add_track_title_text:\n",
    "        axes[ax].text(gene_len*plot_pars[\"track_title_x_coord\"], y_text+y_text*0.8, \n",
    "                      fr'$\\mathbf{{Functional}}$ $\\mathbf{{Impact}}$', \n",
    "                      ha='center', va='center', fontsize=plot_pars[\"track_title_fontsize\"], color=\"black\")\n",
    "\n",
    "    inset_ax = inset_axes(axes[ax], width=\"10%\", height=\"10%\", loc='center left', \n",
    "                          bbox_to_anchor=plot_pars[\"ofml_cbar_coord\"], bbox_transform=axes[ax].transAxes, borderpad=0)\n",
    "    cbar = plt.colorbar(fml_scatter, cax=inset_ax, orientation='horizontal')\n",
    "    cbar.set_label('Impact score', fontsize=plot_pars[\"legend_fontsize\"])\n",
    "    cbar.ax.tick_params(labelsize=plot_pars[\"ticksize\"])\n",
    "\n",
    "    n_max = np.max(ofml_muts_score_gene[\"Count\"])\n",
    "    j = 14.5\n",
    "    i = 19.33\n",
    "    axes[ax].set_ylim(-n_max/i, n_max + n_max/j)    \n",
    "  \n",
    "    axes[ax].set_ylabel('SNV count', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=45, labelpad=plot_pars[\"ylabel_pad\"], va='center')\n",
    "    axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "    \n",
    "    if rm_spines:\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "    elif light_spines:\n",
    "        axes[ax].spines['top'].set_color(colors[\"hv_lines\"])\n",
    "        axes[ax].spines['right'].set_color(colors[\"hv_lines\"])        \n",
    "        \n",
    "\n",
    "    # Frameshift enrichment\n",
    "    # =====================\n",
    "\n",
    "    ax=8\n",
    "    plot_count_track(frameshift_indels_count_gene, axes, ax=ax, neg_track=inframe_indels_count_gene, gene_len=gene_len,\n",
    "                     col_pos_track=colors[\"frameshift\"], col_neg_track=colors[\"inframe\"], label_pos_track=\"Frameshift\", label_neg_track=\"Inframe\",\n",
    "                     ymargin=1, hv_lines=colors[\"hv_lines\"])\n",
    "    if add_score_text:\n",
    "        add_ax_text(score=np.round(indels_score, 2), \n",
    "                    pvalue=np.round(indels_pvalue, 2),\n",
    "                    y_text=(np.max(frameshift_indels_count_gene[\"Count\"]) + np.max(inframe_indels_count_gene[\"Count\"]))/2, \n",
    "                    x_text=gene_len*plot_pars[\"score_txt_x_coord\"], \n",
    "                    ax=axes[ax], \n",
    "                    y_shift=0.24,\n",
    "                    y_adjust=0.4,\n",
    "                    equal_less=False)\n",
    "    if add_track_title_text:\n",
    "        axes[ax].text(gene_len*plot_pars[\"track_title_x_coord\"], 3.9, \n",
    "                      fr'$\\mathbf{{Frameshift}}$ $\\mathbf{{enrichment}}$', \n",
    "                      ha='center', va='center', fontsize=plot_pars[\"track_title_fontsize\"], color=\"black\")\n",
    "    axes[ax].legend(loc=\"upper left\", fontsize=plot_pars[\"legend_fontsize\"], frameon=plot_pars[\"legend_frameon\"])\n",
    "    axes[ax].set_ylabel('Indels count', fontsize=plot_pars[\"ylabel_fontsize\"], rotation=45, labelpad=plot_pars[\"ylabel_pad\"], va='center')\n",
    "    \n",
    "    axes[ax].yaxis.set_major_locator(MaxNLocator(integer=True, nbins=5))\n",
    "    tick_labels = [abs(int(tick)) for tick in axes[ax].get_yticks()]\n",
    "    axes[ax].set_yticklabels(tick_labels)\n",
    "    \n",
    "    if rm_spines:\n",
    "        axes[ax].spines['top'].set_visible(False)\n",
    "        axes[ax].spines['right'].set_visible(False)\n",
    "    elif light_spines:\n",
    "        axes[ax].spines['top'].set_color(colors[\"hv_lines\"])\n",
    "        axes[ax].spines['right'].set_color(colors[\"hv_lines\"])\n",
    "\n",
    "\n",
    "    # Domain\n",
    "    # ======\n",
    "    \n",
    "    if isinstance(domain_df, pd.DataFrame):\n",
    "        ax=10\n",
    "        domain_color_dict = {}\n",
    "\n",
    "        for n, name in enumerate(domain_df[\"Description\"].unique()):\n",
    "            domain_color_dict[name] = f\"C{n}\"\n",
    "\n",
    "        n = 0\n",
    "        added_domain = []\n",
    "        for i, row in domain_df.iterrows():\n",
    "            if pd.Series([row[\"Description\"], row[\"Begin\"], row[\"End\"]]).isnull().any():\n",
    "                continue\n",
    "\n",
    "            name = row[\"Description\"]\n",
    "            start = int(row[\"Begin\"])\n",
    "            end = int(row[\"End\"])\n",
    "            axes[ax].fill_between(range(start, end+1), -0.45, 0.45,  alpha=0.5, color=domain_color_dict[name])\n",
    "            if name not in added_domain:\n",
    "                y = -0.04\n",
    "                axes[ax].text(((start + end) / 2)+0.5, y, name, ha='center', va='center', fontsize=10, color=\"black\")\n",
    "                added_domain.append(name)\n",
    "        axes[ax].set_yticks([])  \n",
    "    else:\n",
    "        axes[10].remove()\n",
    "\n",
    "        \n",
    "    # Spaces\n",
    "    # =====\n",
    "    axes[3].remove()\n",
    "    axes[5].remove()\n",
    "    axes[7].remove()\n",
    "    axes[9].remove()\n",
    "\n",
    "    \n",
    "    # X axes\n",
    "    # ======\n",
    "    axes[ax].tick_params(axis='y', labelsize=plot_pars[\"ticksize\"])\n",
    "    axes[ax].tick_params(axis='x', labelsize=plot_pars[\"ticksize\"])\n",
    "    axes[ax].set_xlabel('Protein position', fontsize=plot_pars[\"xlabel_fontsize\"]) \n",
    "\n",
    "    if output_path is not None:\n",
    "        print(f\"Saving {output_path}\")\n",
    "        plt.savefig(output_path, dpi=plot_pars[\"dpi\"], bbox_inches='tight')\n",
    "    plt.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "o3d_nb",
   "language": "python",
   "name": "o3d_nb"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
