{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe8a86d1-e152-40f3-9e19-0af72c121bfb",
   "metadata": {},
   "source": [
    "# Trying intersection of callers (FAILED)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9faaf593-39f8-43a2-b380-b8abd89f6436",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/spellegrini/miniconda3/envs/clustering_3d/lib/python3.10/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.23.5\n",
      "  warnings.warn(f\"A NumPy version >={np_minversion} and <{np_maxversion}\"\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import os.path\n",
    "import seaborn as sns\n",
    "import math\n",
    "import re\n",
    "import gzip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fc0ab2ca-b4bd-4ec8-a4f2-39f000059dc3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/jobs/spellegrini/8892130/ipykernel_1283/1484548907.py:3: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  pd.read_csv(f\"{input_maf}/OTHER_WXS_TCGA_FULL.in.maf\", sep=\"\\t\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1</td>\n",
       "      <td>861377</td>\n",
       "      <td>861377</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-X2-A95T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>p.R19Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1</td>\n",
       "      <td>865589</td>\n",
       "      <td>865589</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-EW-A1PH</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>p.L43M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1</td>\n",
       "      <td>866453</td>\n",
       "      <td>866453</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-EW-A6SD</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1</td>\n",
       "      <td>871146</td>\n",
       "      <td>871146</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-FD-A6TD</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1</td>\n",
       "      <td>874447</td>\n",
       "      <td>874447</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-BF-A1Q0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>p.R153H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401917</th>\n",
       "      <td>TMLHE</td>\n",
       "      <td>X</td>\n",
       "      <td>154754142</td>\n",
       "      <td>154754142</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-QN-A5NN</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401918</th>\n",
       "      <td>TMLHE</td>\n",
       "      <td>X</td>\n",
       "      <td>154754220</td>\n",
       "      <td>154754220</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-SX-A71V</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401919</th>\n",
       "      <td>TMLHE</td>\n",
       "      <td>X</td>\n",
       "      <td>154774805</td>\n",
       "      <td>154774805</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-E7-A4XJ</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>p.A45S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401920</th>\n",
       "      <td>TMLHE</td>\n",
       "      <td>X</td>\n",
       "      <td>154774839</td>\n",
       "      <td>154774839</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-CV-7414</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>p.S33R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401921</th>\n",
       "      <td>TMLHE</td>\n",
       "      <td>X</td>\n",
       "      <td>154774895</td>\n",
       "      <td>154774895</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-CH-5748</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>p.D15Y</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>401922 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol Chromosome  Start_Position  End_Position Reference_Allele  \\\n",
       "0           SAMD11          1          861377        861377                G   \n",
       "1           SAMD11          1          865589        865589                C   \n",
       "2           SAMD11          1          866453        866453                C   \n",
       "3           SAMD11          1          871146        871146                C   \n",
       "4           SAMD11          1          874447        874447                G   \n",
       "...            ...        ...             ...           ...              ...   \n",
       "401917       TMLHE          X       154754142     154754142                C   \n",
       "401918       TMLHE          X       154754220     154754220                G   \n",
       "401919       TMLHE          X       154774805     154774805                C   \n",
       "401920       TMLHE          X       154774839     154774839                G   \n",
       "401921       TMLHE          X       154774895     154774895                C   \n",
       "\n",
       "       Tumor_Seq_Allele2 Tumor_Sample_Barcode Variant_Classification  \\\n",
       "0                      A         TCGA-X2-A95T      Missense_Mutation   \n",
       "1                      A         TCGA-EW-A1PH      Missense_Mutation   \n",
       "2                      A         TCGA-EW-A6SD     synonymous_variant   \n",
       "3                      A         TCGA-FD-A6TD  splice_region_variant   \n",
       "4                      A         TCGA-BF-A1Q0      Missense_Mutation   \n",
       "...                  ...                  ...                    ...   \n",
       "401917                 T         TCGA-QN-A5NN     synonymous_variant   \n",
       "401918                 T         TCGA-SX-A71V            stop_gained   \n",
       "401919                 A         TCGA-E7-A4XJ      Missense_Mutation   \n",
       "401920                 T         TCGA-CV-7414      Missense_Mutation   \n",
       "401921                 A         TCGA-CH-5748      Missense_Mutation   \n",
       "\n",
       "          Transcript_ID HGVSp_Short  \n",
       "0       ENST00000342066      p.R19Q  \n",
       "1       ENST00000342066      p.L43M  \n",
       "2       ENST00000342066           .  \n",
       "3       ENST00000342066           .  \n",
       "4       ENST00000342066     p.R153H  \n",
       "...                 ...         ...  \n",
       "401917  ENST00000334398           .  \n",
       "401918  ENST00000334398           .  \n",
       "401919  ENST00000334398      p.A45S  \n",
       "401920  ENST00000334398      p.S33R  \n",
       "401921  ENST00000334398      p.D15Y  \n",
       "\n",
       "[401922 rows x 10 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "input_mut_profile = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_ch/input/mut_profile\"\n",
    "input_maf = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_ch/input/maf\"\n",
    "pd.read_csv(f\"{input_maf}/OTHER_WXS_TCGA_FULL.in.maf\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a2bc1dc2-54ac-40c0-b52c-54220e8b1073",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols_maf = [\"Hugo_Symbol\", \"Chromosome\", \"Reference_Allele\", \n",
    "            \"Tumor_Seq_Allele2\", \"Tumor_Sample_Barcode\",\t\"Variant_Classification\", \"Transcript_ID\", \"HGVSp_Short\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c1414dab-5bbc-4d41-a9b1-59db8f74b442",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>COHORT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COL16A1</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000362776</td>\n",
       "      <td>p.P761P</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NRP1</td>\n",
       "      <td>10</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000265371</td>\n",
       "      <td>p.E784K</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NCR3LG1</td>\n",
       "      <td>11</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000341637</td>\n",
       "      <td>p.H85N</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAV3</td>\n",
       "      <td>12</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Essential_Splice</td>\n",
       "      <td>ENSP00000381007</td>\n",
       "      <td>p..</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FAM83G</td>\n",
       "      <td>17</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000373647</td>\n",
       "      <td>p.S116S</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667415</th>\n",
       "      <td>CDCA2</td>\n",
       "      <td>8</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000328228</td>\n",
       "      <td>p.K627N</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667416</th>\n",
       "      <td>MROH1</td>\n",
       "      <td>8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000435565</td>\n",
       "      <td>p.A64V</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667417</th>\n",
       "      <td>C9orf78</td>\n",
       "      <td>9</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000361524</td>\n",
       "      <td>p.I180I</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667418</th>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>9</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000277541</td>\n",
       "      <td>p.A1944T</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667419</th>\n",
       "      <td>DDX53</td>\n",
       "      <td>X</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000368667</td>\n",
       "      <td>p.R624C</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>667420 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol Chromosome Reference_Allele Tumor_Seq_Allele2  \\\n",
       "0          COL16A1          1                C                 A   \n",
       "1             NRP1         10                G                 A   \n",
       "2          NCR3LG1         11                C                 A   \n",
       "3             NAV3         12                G                 A   \n",
       "4           FAM83G         17                C                 A   \n",
       "...            ...        ...              ...               ...   \n",
       "667415       CDCA2          8                G                 C   \n",
       "667416       MROH1          8                C                 T   \n",
       "667417     C9orf78          9                C                 T   \n",
       "667418      NOTCH1          9                G                 A   \n",
       "667419       DDX53          X                C                 T   \n",
       "\n",
       "               Tumor_Sample_Barcode Variant_Classification    Transcript_ID  \\\n",
       "0       CPCT02010349T_CPCT02010349R             Synonymous  ENSP00000362776   \n",
       "1       CPCT02010349T_CPCT02010349R               Missense  ENSP00000265371   \n",
       "2       CPCT02010349T_CPCT02010349R               Missense  ENSP00000341637   \n",
       "3       CPCT02010349T_CPCT02010349R       Essential_Splice  ENSP00000381007   \n",
       "4       CPCT02010349T_CPCT02010349R             Synonymous  ENSP00000373647   \n",
       "...                             ...                    ...              ...   \n",
       "667415                 TCGA-ZX-AA5X               Missense  ENSP00000328228   \n",
       "667416                 TCGA-ZX-AA5X               Missense  ENSP00000435565   \n",
       "667417                 TCGA-ZX-AA5X             Synonymous  ENSP00000361524   \n",
       "667418                 TCGA-ZX-AA5X               Missense  ENSP00000277541   \n",
       "667419                 TCGA-ZX-AA5X               Missense  ENSP00000368667   \n",
       "\n",
       "       HGVSp_Short                COHORT  \n",
       "0          p.P761P  OTHER_WGS_HMF_MUTECT  \n",
       "1          p.E784K  OTHER_WGS_HMF_MUTECT  \n",
       "2           p.H85N  OTHER_WGS_HMF_MUTECT  \n",
       "3              p..  OTHER_WGS_HMF_MUTECT  \n",
       "4          p.S116S  OTHER_WGS_HMF_MUTECT  \n",
       "...            ...                   ...  \n",
       "667415     p.K627N   OTHER_WXS_TCGA_FULL  \n",
       "667416      p.A64V   OTHER_WXS_TCGA_FULL  \n",
       "667417     p.I180I   OTHER_WXS_TCGA_FULL  \n",
       "667418    p.A1944T   OTHER_WXS_TCGA_FULL  \n",
       "667419     p.R624C   OTHER_WXS_TCGA_FULL  \n",
       "\n",
       "[667420 rows x 9 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_ch_mut = pd.read_csv(\"/workspace/projects/boostdm_ch/output/run_20220127/discovery/mutations.tsv\", sep=\"\\t\")\n",
    "all_ch_mut[\"aachange\"] = \"p.\" + all_ch_mut[\"aachange\"]\n",
    "all_ch_mut = all_ch_mut.rename(columns={\"gene\" : \"Hugo_Symbol\",\n",
    "                                       \"chr\" : \"Chromosome\", \n",
    "                                       \"ref_cod\" : \"Reference_Allele\", \n",
    "                                       \"mut_cod\" : \"Tumor_Seq_Allele2\", \n",
    "                                       \"impact\" : \"Variant_Classification\",\n",
    "                                       \"sampleID\" : \"Tumor_Sample_Barcode\",\n",
    "                                       \"pid\" : \"Transcript_ID\",\n",
    "                                        \"aachange\" : \"HGVSp_Short\"})\n",
    "all_ch_mut = all_ch_mut[cols_maf + [\"COHORT\"]]\n",
    "all_ch_mut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b02bdadc-32e1-44e5-bd40-a033df2941dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>Prot_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COL16A1</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000362776</td>\n",
       "      <td>p.P761P</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NRP1</td>\n",
       "      <td>10</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000265371</td>\n",
       "      <td>p.E784K</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NCR3LG1</td>\n",
       "      <td>11</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000341637</td>\n",
       "      <td>p.H85N</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAV3</td>\n",
       "      <td>12</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Essential_Splice</td>\n",
       "      <td>ENSP00000381007</td>\n",
       "      <td>p..</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FAM83G</td>\n",
       "      <td>17</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000373647</td>\n",
       "      <td>p.S116S</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667415</th>\n",
       "      <td>CDCA2</td>\n",
       "      <td>8</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000328228</td>\n",
       "      <td>p.K627N</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667416</th>\n",
       "      <td>MROH1</td>\n",
       "      <td>8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000435565</td>\n",
       "      <td>p.A64V</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667417</th>\n",
       "      <td>C9orf78</td>\n",
       "      <td>9</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000361524</td>\n",
       "      <td>p.I180I</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667418</th>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>9</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000277541</td>\n",
       "      <td>p.A1944T</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>1944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667419</th>\n",
       "      <td>DDX53</td>\n",
       "      <td>X</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000368667</td>\n",
       "      <td>p.R624C</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>624</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>667420 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol Chromosome Reference_Allele Tumor_Seq_Allele2  \\\n",
       "0          COL16A1          1                C                 A   \n",
       "1             NRP1         10                G                 A   \n",
       "2          NCR3LG1         11                C                 A   \n",
       "3             NAV3         12                G                 A   \n",
       "4           FAM83G         17                C                 A   \n",
       "...            ...        ...              ...               ...   \n",
       "667415       CDCA2          8                G                 C   \n",
       "667416       MROH1          8                C                 T   \n",
       "667417     C9orf78          9                C                 T   \n",
       "667418      NOTCH1          9                G                 A   \n",
       "667419       DDX53          X                C                 T   \n",
       "\n",
       "               Tumor_Sample_Barcode Variant_Classification    Transcript_ID  \\\n",
       "0       CPCT02010349T_CPCT02010349R             Synonymous  ENSP00000362776   \n",
       "1       CPCT02010349T_CPCT02010349R               Missense  ENSP00000265371   \n",
       "2       CPCT02010349T_CPCT02010349R               Missense  ENSP00000341637   \n",
       "3       CPCT02010349T_CPCT02010349R       Essential_Splice  ENSP00000381007   \n",
       "4       CPCT02010349T_CPCT02010349R             Synonymous  ENSP00000373647   \n",
       "...                             ...                    ...              ...   \n",
       "667415                 TCGA-ZX-AA5X               Missense  ENSP00000328228   \n",
       "667416                 TCGA-ZX-AA5X               Missense  ENSP00000435565   \n",
       "667417                 TCGA-ZX-AA5X             Synonymous  ENSP00000361524   \n",
       "667418                 TCGA-ZX-AA5X               Missense  ENSP00000277541   \n",
       "667419                 TCGA-ZX-AA5X               Missense  ENSP00000368667   \n",
       "\n",
       "       HGVSp_Short                COHORT Prot_pos  \n",
       "0          p.P761P  OTHER_WGS_HMF_MUTECT      761  \n",
       "1          p.E784K  OTHER_WGS_HMF_MUTECT      784  \n",
       "2           p.H85N  OTHER_WGS_HMF_MUTECT       85  \n",
       "3              p..  OTHER_WGS_HMF_MUTECT           \n",
       "4          p.S116S  OTHER_WGS_HMF_MUTECT      116  \n",
       "...            ...                   ...      ...  \n",
       "667415     p.K627N   OTHER_WXS_TCGA_FULL      627  \n",
       "667416      p.A64V   OTHER_WXS_TCGA_FULL       64  \n",
       "667417     p.I180I   OTHER_WXS_TCGA_FULL      180  \n",
       "667418    p.A1944T   OTHER_WXS_TCGA_FULL     1944  \n",
       "667419     p.R624C   OTHER_WXS_TCGA_FULL      624  \n",
       "\n",
       "[667420 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_ch_mut[\"Prot_pos\"] = all_ch_mut.apply(lambda x: re.sub(r'\\D+', \"\", x[\"HGVSp_Short\"]), axis=1)\n",
    "all_ch_mut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0d66f39c-6891-4266-87f6-503b30cc7638",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>Prot_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COL16A1</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000362776</td>\n",
       "      <td>p.P761P</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NRP1</td>\n",
       "      <td>10</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000265371</td>\n",
       "      <td>p.E784K</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NCR3LG1</td>\n",
       "      <td>11</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000341637</td>\n",
       "      <td>p.H85N</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAV3</td>\n",
       "      <td>12</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Essential_Splice</td>\n",
       "      <td>ENSP00000381007</td>\n",
       "      <td>p..</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FAM83G</td>\n",
       "      <td>17</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000373647</td>\n",
       "      <td>p.S116S</td>\n",
       "      <td>OTHER_WGS_HMF_MUTECT</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51626</th>\n",
       "      <td>RAB44</td>\n",
       "      <td>6</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>WIDE01010464T_WIDE01010464R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000481054</td>\n",
       "      <td>p.T553T</td>\n",
       "      <td>OTHER_WGS_HMF_MOSAIC</td>\n",
       "      <td>553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51627</th>\n",
       "      <td>POR</td>\n",
       "      <td>7</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>WIDE01010464T_WIDE01010464R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000419970</td>\n",
       "      <td>p.L405V</td>\n",
       "      <td>OTHER_WGS_HMF_MOSAIC</td>\n",
       "      <td>405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51628</th>\n",
       "      <td>SLC16A2</td>\n",
       "      <td>X</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>WIDE01010464T_WIDE01010464R</td>\n",
       "      <td>Nonsense</td>\n",
       "      <td>ENSP00000465734</td>\n",
       "      <td>p.R314*</td>\n",
       "      <td>OTHER_WGS_HMF_MOSAIC</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51629</th>\n",
       "      <td>UBE4B</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>WIDE01010481T_WIDE01010481R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000343001</td>\n",
       "      <td>p.R717Q</td>\n",
       "      <td>OTHER_WGS_HMF_MOSAIC</td>\n",
       "      <td>717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51630</th>\n",
       "      <td>PPM1D</td>\n",
       "      <td>17</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>WIDE01010481T_WIDE01010481R</td>\n",
       "      <td>Nonsense</td>\n",
       "      <td>ENSP00000306682</td>\n",
       "      <td>p.R572*</td>\n",
       "      <td>OTHER_WGS_HMF_MOSAIC</td>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51631 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol Chromosome Reference_Allele Tumor_Seq_Allele2  \\\n",
       "0         COL16A1          1                C                 A   \n",
       "1            NRP1         10                G                 A   \n",
       "2         NCR3LG1         11                C                 A   \n",
       "3            NAV3         12                G                 A   \n",
       "4          FAM83G         17                C                 A   \n",
       "...           ...        ...              ...               ...   \n",
       "51626       RAB44          6                A                 G   \n",
       "51627         POR          7                C                 G   \n",
       "51628     SLC16A2          X                C                 T   \n",
       "51629       UBE4B          1                G                 A   \n",
       "51630       PPM1D         17                C                 T   \n",
       "\n",
       "              Tumor_Sample_Barcode Variant_Classification    Transcript_ID  \\\n",
       "0      CPCT02010349T_CPCT02010349R             Synonymous  ENSP00000362776   \n",
       "1      CPCT02010349T_CPCT02010349R               Missense  ENSP00000265371   \n",
       "2      CPCT02010349T_CPCT02010349R               Missense  ENSP00000341637   \n",
       "3      CPCT02010349T_CPCT02010349R       Essential_Splice  ENSP00000381007   \n",
       "4      CPCT02010349T_CPCT02010349R             Synonymous  ENSP00000373647   \n",
       "...                            ...                    ...              ...   \n",
       "51626  WIDE01010464T_WIDE01010464R             Synonymous  ENSP00000481054   \n",
       "51627  WIDE01010464T_WIDE01010464R               Missense  ENSP00000419970   \n",
       "51628  WIDE01010464T_WIDE01010464R               Nonsense  ENSP00000465734   \n",
       "51629  WIDE01010481T_WIDE01010481R               Missense  ENSP00000343001   \n",
       "51630  WIDE01010481T_WIDE01010481R               Nonsense  ENSP00000306682   \n",
       "\n",
       "      HGVSp_Short                COHORT Prot_pos  \n",
       "0         p.P761P  OTHER_WGS_HMF_MUTECT      761  \n",
       "1         p.E784K  OTHER_WGS_HMF_MUTECT      784  \n",
       "2          p.H85N  OTHER_WGS_HMF_MUTECT       85  \n",
       "3             p..  OTHER_WGS_HMF_MUTECT           \n",
       "4         p.S116S  OTHER_WGS_HMF_MUTECT      116  \n",
       "...           ...                   ...      ...  \n",
       "51626     p.T553T  OTHER_WGS_HMF_MOSAIC      553  \n",
       "51627     p.L405V  OTHER_WGS_HMF_MOSAIC      405  \n",
       "51628     p.R314*  OTHER_WGS_HMF_MOSAIC      314  \n",
       "51629     p.R717Q  OTHER_WGS_HMF_MOSAIC      717  \n",
       "51630     p.R572*  OTHER_WGS_HMF_MOSAIC      572  \n",
       "\n",
       "[51631 rows x 10 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgs_df = all_ch_mut[all_ch_mut[\"COHORT\"].str.startswith(\"OTHER_WGS_HMF\")]\n",
    "wgs_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "48153456-27f3-4719-9478-0c486e74299f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/jobs/spellegrini/8892130/ipykernel_1283/1515262322.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  wgs_df[\"Caller\"] = wgs_df.COHORT.apply(lambda x: x.split(\"OTHER_WGS_HMF_\")[-1])\n",
      "/tmp/jobs/spellegrini/8892130/ipykernel_1283/1515262322.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  wgs_df[\"Caller\"] = wgs_df.COHORT.apply(lambda x: x.split(\"OTHER_WGS_HMF_\")[-1])\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Prot_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>COL16A1</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.P761P</td>\n",
       "      <td>761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NRP1</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.E784K</td>\n",
       "      <td>784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NCR3LG1</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.H85N</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>FAM83G</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.S116S</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SLCO2A1</td>\n",
       "      <td>CPCT02010349T_CPCT02010349R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.R314R</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11416</th>\n",
       "      <td>ANKRD60</td>\n",
       "      <td>WIDE01010464T_WIDE01010464R</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.R310R</td>\n",
       "      <td>310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11417</th>\n",
       "      <td>CXCL14</td>\n",
       "      <td>WIDE01010464T_WIDE01010464R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.C37R</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11421</th>\n",
       "      <td>SLC16A2</td>\n",
       "      <td>WIDE01010464T_WIDE01010464R</td>\n",
       "      <td>Nonsense</td>\n",
       "      <td>p.R314*</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11422</th>\n",
       "      <td>UBE4B</td>\n",
       "      <td>WIDE01010481T_WIDE01010481R</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.R717Q</td>\n",
       "      <td>717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11423</th>\n",
       "      <td>PPM1D</td>\n",
       "      <td>WIDE01010481T_WIDE01010481R</td>\n",
       "      <td>Nonsense</td>\n",
       "      <td>p.R572*</td>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3262 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol         Tumor_Sample_Barcode Variant_Classification  \\\n",
       "1         COL16A1  CPCT02010349T_CPCT02010349R             Synonymous   \n",
       "2            NRP1  CPCT02010349T_CPCT02010349R               Missense   \n",
       "3         NCR3LG1  CPCT02010349T_CPCT02010349R               Missense   \n",
       "6          FAM83G  CPCT02010349T_CPCT02010349R             Synonymous   \n",
       "15        SLCO2A1  CPCT02010349T_CPCT02010349R             Synonymous   \n",
       "...           ...                          ...                    ...   \n",
       "11416     ANKRD60  WIDE01010464T_WIDE01010464R             Synonymous   \n",
       "11417      CXCL14  WIDE01010464T_WIDE01010464R               Missense   \n",
       "11421     SLC16A2  WIDE01010464T_WIDE01010464R               Nonsense   \n",
       "11422       UBE4B  WIDE01010481T_WIDE01010481R               Missense   \n",
       "11423       PPM1D  WIDE01010481T_WIDE01010481R               Nonsense   \n",
       "\n",
       "      HGVSp_Short Prot_pos  \n",
       "1         p.P761P      761  \n",
       "2         p.E784K      784  \n",
       "3          p.H85N       85  \n",
       "6         p.S116S      116  \n",
       "15        p.R314R      314  \n",
       "...           ...      ...  \n",
       "11416     p.R310R      310  \n",
       "11417      p.C37R       37  \n",
       "11421     p.R314*      314  \n",
       "11422     p.R717Q      717  \n",
       "11423     p.R572*      572  \n",
       "\n",
       "[3262 rows x 5 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgs_df[\"Caller\"] = wgs_df.COHORT.apply(lambda x: x.split(\"OTHER_WGS_HMF_\")[-1])\n",
    "wgs_df_mutect = wgs_df[wgs_df[\"Caller\"] == \"MUTECT\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "wgs_df_mosaic = wgs_df[wgs_df[\"Caller\"] == \"MOSAIC\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "wgs_df_full = wgs_df[wgs_df[\"Caller\"] == \"FULL\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "\n",
    "wgs_df[\"Caller\"] = wgs_df.COHORT.apply(lambda x: x.split(\"OTHER_WGS_HMF_\")[-1])\n",
    "wgs_df_mutect = wgs_df[wgs_df[\"Caller\"] == \"MUTECT\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "wgs_df_mosaic = wgs_df[wgs_df[\"Caller\"] == \"MOSAIC\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "wgs_df_full = wgs_df[wgs_df[\"Caller\"] == \"FULL\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "\n",
    "wgs_pos_mosaic = wgs_df_mosaic.apply(lambda x: f'{x[\"Hugo_Symbol\"]}-{x[\"Prot_pos\"]}', axis=1).reset_index(drop=True)\n",
    "wgs_pos_mutect = wgs_df_mutect.apply(lambda x: f'{x[\"Hugo_Symbol\"]}-{x[\"Prot_pos\"]}', axis=1).reset_index(drop=True)\n",
    "wgs_pos_full = wgs_df_full.apply(lambda x: f'{x[\"Hugo_Symbol\"]}-{x[\"Prot_pos\"]}', axis=1).reset_index(drop=True)\n",
    "wgs_intersect_df = wgs_df_mosaic[wgs_pos_mosaic.isin(wgs_pos_mutect) & wgs_pos_mosaic.isin(wgs_pos_full)]\n",
    "wgs_intersect_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "8eea058a-141d-43b6-8699-e748615d5da1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Hugo_Symbol\n",
       "DNMT3A      46\n",
       "TTN         15\n",
       "ATM          6\n",
       "COL6A3       5\n",
       "CSMD1        5\n",
       "MYH8         4\n",
       "EYS          4\n",
       "SLC3A1       4\n",
       "SF3B1        4\n",
       "FAT1         4\n",
       "MTA2         4\n",
       "FBN2         4\n",
       "DNAH9        4\n",
       "NOS1         4\n",
       "ADGRV1       4\n",
       "PDZRN3       3\n",
       "APOB         3\n",
       "SETD1B       3\n",
       "DNAH6        3\n",
       "TET2         3\n",
       "DNAH5        3\n",
       "NEB          3\n",
       "ADAMTSL1     3\n",
       "APC          3\n",
       "RP1          3\n",
       "NFASC        3\n",
       "DOCK8        3\n",
       "CXorf67      3\n",
       "MUC16        3\n",
       "CCDC168      3\n",
       "USH2A        3\n",
       "MUC5B        3\n",
       "PLXNA4       3\n",
       "TP53         3\n",
       "PTPRK        3\n",
       "TP63         3\n",
       "ABCC11       3\n",
       "KLHDC7A      3\n",
       "RYR3         3\n",
       "DSCAML1      3\n",
       "SVEP1        3\n",
       "PABPC5       2\n",
       "TRIO         2\n",
       "LAMA1        2\n",
       "DNAH10       2\n",
       "PTPRT        2\n",
       "TPH1         2\n",
       "DNAH7        2\n",
       "KRT85        2\n",
       "TPP2         2\n",
       "Name: Tumor_Sample_Barcode, dtype: int64"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgs_intersect_df[wgs_intersect_df[\"Variant_Classification\"] == \"Missense\"].groupby(\"Hugo_Symbol\").count().iloc[:, 0].sort_values(ascending=False)[:50]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "9dffa32a-2d96-4aab-bd81-068dfb944c0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>Prot_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>51631</th>\n",
       "      <td>PADI6</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000483125</td>\n",
       "      <td>p.K498N</td>\n",
       "      <td>OTHER_WXS_TCGA_MOSAIC</td>\n",
       "      <td>498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51632</th>\n",
       "      <td>C4BPB</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000243611</td>\n",
       "      <td>p.P142T</td>\n",
       "      <td>OTHER_WXS_TCGA_MOSAIC</td>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51633</th>\n",
       "      <td>NLRC5</td>\n",
       "      <td>16</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000262510</td>\n",
       "      <td>p.C1403C</td>\n",
       "      <td>OTHER_WXS_TCGA_MOSAIC</td>\n",
       "      <td>1403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51634</th>\n",
       "      <td>COL3A1</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000304408</td>\n",
       "      <td>p.C1268C</td>\n",
       "      <td>OTHER_WXS_TCGA_MOSAIC</td>\n",
       "      <td>1268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51635</th>\n",
       "      <td>TNIK</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000399511</td>\n",
       "      <td>p.T1351T</td>\n",
       "      <td>OTHER_WXS_TCGA_MOSAIC</td>\n",
       "      <td>1351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667415</th>\n",
       "      <td>CDCA2</td>\n",
       "      <td>8</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000328228</td>\n",
       "      <td>p.K627N</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667416</th>\n",
       "      <td>MROH1</td>\n",
       "      <td>8</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000435565</td>\n",
       "      <td>p.A64V</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667417</th>\n",
       "      <td>C9orf78</td>\n",
       "      <td>9</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>ENSP00000361524</td>\n",
       "      <td>p.I180I</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667418</th>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>9</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000277541</td>\n",
       "      <td>p.A1944T</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>1944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667419</th>\n",
       "      <td>DDX53</td>\n",
       "      <td>X</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>ENSP00000368667</td>\n",
       "      <td>p.R624C</td>\n",
       "      <td>OTHER_WXS_TCGA_FULL</td>\n",
       "      <td>624</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>615789 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol Chromosome Reference_Allele Tumor_Seq_Allele2  \\\n",
       "51631        PADI6          1                G                 T   \n",
       "51632        C4BPB          1                C                 A   \n",
       "51633        NLRC5         16                C                 T   \n",
       "51634       COL3A1          2                C                 T   \n",
       "51635         TNIK          3                C                 A   \n",
       "...            ...        ...              ...               ...   \n",
       "667415       CDCA2          8                G                 C   \n",
       "667416       MROH1          8                C                 T   \n",
       "667417     C9orf78          9                C                 T   \n",
       "667418      NOTCH1          9                G                 A   \n",
       "667419       DDX53          X                C                 T   \n",
       "\n",
       "       Tumor_Sample_Barcode Variant_Classification    Transcript_ID  \\\n",
       "51631          TCGA-02-0003               Missense  ENSP00000483125   \n",
       "51632          TCGA-02-0003               Missense  ENSP00000243611   \n",
       "51633          TCGA-02-0003             Synonymous  ENSP00000262510   \n",
       "51634          TCGA-02-0003             Synonymous  ENSP00000304408   \n",
       "51635          TCGA-02-0003             Synonymous  ENSP00000399511   \n",
       "...                     ...                    ...              ...   \n",
       "667415         TCGA-ZX-AA5X               Missense  ENSP00000328228   \n",
       "667416         TCGA-ZX-AA5X               Missense  ENSP00000435565   \n",
       "667417         TCGA-ZX-AA5X             Synonymous  ENSP00000361524   \n",
       "667418         TCGA-ZX-AA5X               Missense  ENSP00000277541   \n",
       "667419         TCGA-ZX-AA5X               Missense  ENSP00000368667   \n",
       "\n",
       "       HGVSp_Short                 COHORT Prot_pos  \n",
       "51631      p.K498N  OTHER_WXS_TCGA_MOSAIC      498  \n",
       "51632      p.P142T  OTHER_WXS_TCGA_MOSAIC      142  \n",
       "51633     p.C1403C  OTHER_WXS_TCGA_MOSAIC     1403  \n",
       "51634     p.C1268C  OTHER_WXS_TCGA_MOSAIC     1268  \n",
       "51635     p.T1351T  OTHER_WXS_TCGA_MOSAIC     1351  \n",
       "...            ...                    ...      ...  \n",
       "667415     p.K627N    OTHER_WXS_TCGA_FULL      627  \n",
       "667416      p.A64V    OTHER_WXS_TCGA_FULL       64  \n",
       "667417     p.I180I    OTHER_WXS_TCGA_FULL      180  \n",
       "667418    p.A1944T    OTHER_WXS_TCGA_FULL     1944  \n",
       "667419     p.R624C    OTHER_WXS_TCGA_FULL      624  \n",
       "\n",
       "[615789 rows x 10 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wxs_df = all_ch_mut[all_ch_mut[\"COHORT\"].str.startswith(\"OTHER_WXS_TCGA\")]\n",
    "wxs_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "2ba884da-ae61-4e22-bc60-2310e94967e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/jobs/spellegrini/8869446/ipykernel_51029/432867690.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  wxs_df[\"Caller\"] = wxs_df.COHORT.apply(lambda x: x.split(\"OTHER_WXS_TCGA_\")[-1])\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Prot_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PADI6</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.K498N</td>\n",
       "      <td>498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C4BPB</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.P142T</td>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NLRC5</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.C1403C</td>\n",
       "      <td>1403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COL3A1</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.C1268C</td>\n",
       "      <td>1268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TNIK</td>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.T1351T</td>\n",
       "      <td>1351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223603</th>\n",
       "      <td>FLNC</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.G1343G</td>\n",
       "      <td>1343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223604</th>\n",
       "      <td>FASTK</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.P298S</td>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223605</th>\n",
       "      <td>MROH1</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.A64V</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223606</th>\n",
       "      <td>C9orf78</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>p.I180I</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223607</th>\n",
       "      <td>DDX53</td>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>Missense</td>\n",
       "      <td>p.R624C</td>\n",
       "      <td>624</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223596 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol Tumor_Sample_Barcode Variant_Classification HGVSp_Short  \\\n",
       "0            PADI6         TCGA-02-0003               Missense     p.K498N   \n",
       "1            C4BPB         TCGA-02-0003               Missense     p.P142T   \n",
       "2            NLRC5         TCGA-02-0003             Synonymous    p.C1403C   \n",
       "3           COL3A1         TCGA-02-0003             Synonymous    p.C1268C   \n",
       "4             TNIK         TCGA-02-0003             Synonymous    p.T1351T   \n",
       "...            ...                  ...                    ...         ...   \n",
       "223603        FLNC         TCGA-ZX-AA5X             Synonymous    p.G1343G   \n",
       "223604       FASTK         TCGA-ZX-AA5X               Missense     p.P298S   \n",
       "223605       MROH1         TCGA-ZX-AA5X               Missense      p.A64V   \n",
       "223606     C9orf78         TCGA-ZX-AA5X             Synonymous     p.I180I   \n",
       "223607       DDX53         TCGA-ZX-AA5X               Missense     p.R624C   \n",
       "\n",
       "       Prot_pos  \n",
       "0           498  \n",
       "1           142  \n",
       "2          1403  \n",
       "3          1268  \n",
       "4          1351  \n",
       "...         ...  \n",
       "223603     1343  \n",
       "223604      298  \n",
       "223605       64  \n",
       "223606      180  \n",
       "223607      624  \n",
       "\n",
       "[223596 rows x 5 columns]"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wxs_df[\"Caller\"] = wxs_df.COHORT.apply(lambda x: x.split(\"OTHER_WXS_TCGA_\")[-1])\n",
    "wxs_df_mosaic = wxs_df[wxs_df[\"Caller\"] == \"MOSAIC\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "wxs_df_full = wxs_df[wxs_df[\"Caller\"] == \"FULL\"].drop(\n",
    "    columns=[\"Caller\", \"COHORT\", \"Transcript_ID\", \"Chromosome\", \"Reference_Allele\", \"Tumor_Seq_Allele2\"]).reset_index(drop=True)\n",
    "\n",
    "wxs_pos_mosaic = wxs_df_mosaic.apply(lambda x: f'{x[\"Hugo_Symbol\"]}-{x[\"Prot_pos\"]}', axis=1).reset_index(drop=True)\n",
    "wxs_pos_full = wxs_df_full.apply(lambda x: f'{x[\"Hugo_Symbol\"]}-{x[\"Prot_pos\"]}', axis=1).reset_index(drop=True)\n",
    "wxs_ix_intersect = wxs_pos_mosaic.isin(wxs_pos_full)\n",
    "wxs_df_intersect = wxs_df_mosaic[wxs_ix_intersect]\n",
    "wxs_df_intersect"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "733fa1d2-a0a1-4016-b136-b3614fb82f38",
   "metadata": {},
   "source": [
    "# Mouse"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e3d07ef-a0b5-42df-9ac6-15c7997598a5",
   "metadata": {},
   "source": [
    "### Convert .txt simple format into .vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a7bdbc0e-6641-42d4-a17f-cf19baae5f8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sort_vcf(vcf_file):\n",
    "    \"\"\"\n",
    "    Sort the vcf file from by chromosomes and position.\n",
    "    \"\"\"\n",
    "    \n",
    "    df = pd.DataFrame(vcf_file)\n",
    "\n",
    "    # Define the custom sort order for Chromosome\n",
    "    chromosome_order = {'1': 1, '2': 2, '3': 3, '4': 4, '5': 5, '6': 6, '7': 7, '8': 8, '9': 9, '10': 10,\n",
    "                        '11': 11, '12': 12, '13': 13, '14': 14, '15': 15, '16': 16, '17': 17, '18': 18, '19': 19,\n",
    "                        '20': 20, '21': 21, 'X': 22, 'Y': 23, 'M': 24}\n",
    "\n",
    "    # Convert 'Chromosome' column to a temporary categorical column for sorting\n",
    "    df['#CHROM'] = df['#CHROM'].astype(str)\n",
    "    df['Chromosome_sort'] = df['#CHROM'].map(chromosome_order)\n",
    "    # Sort and drop temporary col\n",
    "    sorted_df = df.sort_values(by=['Chromosome_sort', 'POS'], ignore_index=True)\n",
    "    sorted_df.drop('Chromosome_sort', axis=1, inplace=True)\n",
    "\n",
    "    return sorted_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c7b57a23-ad59-417a-9f3e-70ed10b5a46e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/jobs/spellegrini/8869446/ipykernel_15677/696072087.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  mouse_df = pd.read_csv(path, sep=\"\\t\")\n"
     ]
    }
   ],
   "source": [
    "mouse_path = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data\" \n",
    "path = f\"{mouse_path}/Liver_DEN_strelka_July2017_mm10.txt\"\n",
    "mouse_df = pd.read_csv(path, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "efb157dc-d065-45d7-8beb-e4cd7c230e6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "mouse_df = mouse_df[[\"CHROMOSOME\", \"POSITION\", \"SAMPLE\", \"REF\", \"ALT\"]]\n",
    "mouse_df = mouse_df.rename(columns={\"CHROMOSOME\" : \"#CHROM\",\n",
    "                                    \"POSITION\" : \"POS\",\n",
    "                                    \"SAMPLE\" : \"ID\"})\n",
    "mouse_df[\"QUAL\"] = \".\"\n",
    "mouse_df[\"FILTER\"] = \"PASS\"\n",
    "mouse_df[\"INFO\"] = \".\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "51351893-7c03-45cd-809f-1caa6a47bb40",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 39.7 s, sys: 11.1 s, total: 50.8 s\n",
      "Wall time: 50.8 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>3000671</td>\n",
       "      <td>do9806_93131_N1</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>3000832</td>\n",
       "      <td>do9414_91603_N1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>3000943</td>\n",
       "      <td>do9786_91932_N7</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>3000976</td>\n",
       "      <td>do9219_90886_N4</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>3001034</td>\n",
       "      <td>do9429_91618_N2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921896</th>\n",
       "      <td>M</td>\n",
       "      <td>3682</td>\n",
       "      <td>do9171_90022_N2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921897</th>\n",
       "      <td>M</td>\n",
       "      <td>8218</td>\n",
       "      <td>do9744_92188_N6</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921898</th>\n",
       "      <td>M</td>\n",
       "      <td>13756</td>\n",
       "      <td>do9212_90810_N5</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921899</th>\n",
       "      <td>M</td>\n",
       "      <td>14328</td>\n",
       "      <td>do9822_93244_N3</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921900</th>\n",
       "      <td>M</td>\n",
       "      <td>14372</td>\n",
       "      <td>do9809_93131_N5</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22921901 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #CHROM      POS               ID REF ALT QUAL FILTER INFO\n",
       "0             1  3000671  do9806_93131_N1   A   G    .   PASS    .\n",
       "1             1  3000832  do9414_91603_N1   T   C    .   PASS    .\n",
       "2             1  3000943  do9786_91932_N7   T   C    .   PASS    .\n",
       "3             1  3000976  do9219_90886_N4   A   T    .   PASS    .\n",
       "4             1  3001034  do9429_91618_N2   G   A    .   PASS    .\n",
       "...         ...      ...              ...  ..  ..  ...    ...  ...\n",
       "22921896      M     3682  do9171_90022_N2   C   T    .   PASS    .\n",
       "22921897      M     8218  do9744_92188_N6   C   T    .   PASS    .\n",
       "22921898      M    13756  do9212_90810_N5   C   T    .   PASS    .\n",
       "22921899      M    14328  do9822_93244_N3   G   A    .   PASS    .\n",
       "22921900      M    14372  do9809_93131_N5   G   A    .   PASS    .\n",
       "\n",
       "[22921901 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "mouse_df = sort_vcf(mouse_df)\n",
    "mouse_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "2a4b2066-facb-4d3b-bfb6-1067593d94ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',\n",
       "       '13', '14', '15', '16', '17', '18', '19', 'X', 'Y', 'M'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mouse_df[\"#CHROM\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "eb60ec11-fced-49d8-abf1-c7fcac75e543",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 6s, sys: 1.71 s, total: 1min 8s\n",
      "Wall time: 1min 8s\n"
     ]
    }
   ],
   "source": [
    "# %%time\n",
    "# mouse_df.to_csv(f\"{mouse_path}/liver_strelka.vcf\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25184506-be6b-4d45-a456-35ffcf99a8f7",
   "metadata": {},
   "source": [
    "### Load VEP output\n",
    "* Run VEP using vep_and_parse_mouse.sh in build/preprocessing/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a3ecda0-8f66-4ec9-9d02-196f41a9898c",
   "metadata": {},
   "source": [
    "#### Reload it and check if I can run it with vep_input for O3D (all transcripts should be there)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0cfccb13-be4c-47e1-beb2-0e3f1fda6604",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/jobs/spellegrini/10373218/ipykernel_682040/3485451430.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  liver_df = pd.read_table(path)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>CANCER_TYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>do9714_91620_N3</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>do9714_91620_N3</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>do9786_91932_N7</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>do9786_91932_N7</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>3278912</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>do9386_90793_N2</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921896</th>\n",
       "      <td>3</td>\n",
       "      <td>4767854</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>do9442_91837_N5</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921897</th>\n",
       "      <td>Y</td>\n",
       "      <td>90793815</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>do12459_91948_N4</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921898</th>\n",
       "      <td>Y</td>\n",
       "      <td>90793868</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>do9172_90170_N2</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921899</th>\n",
       "      <td>Y</td>\n",
       "      <td>90794273</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>do9201_90802_N1</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921900</th>\n",
       "      <td>Y</td>\n",
       "      <td>90805575</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>do9197_90797_N2</td>\n",
       "      <td>subs</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22921901 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROMOSOME  POSITION REF ALT            SAMPLE  TYPE CANCER_TYPE\n",
       "0                10   3145729   A   G   do9714_91620_N3  subs       Liver\n",
       "1                10   3145729   A   T   do9714_91620_N3  subs       Liver\n",
       "2                10   3145729   A   G   do9786_91932_N7  subs       Liver\n",
       "3                10   3145729   A   T   do9786_91932_N7  subs       Liver\n",
       "4                10   3278912   T   C   do9386_90793_N2  subs       Liver\n",
       "...             ...       ...  ..  ..               ...   ...         ...\n",
       "22921896          3   4767854   A   T   do9442_91837_N5  subs       Liver\n",
       "22921897          Y  90793815   T   A  do12459_91948_N4  subs       Liver\n",
       "22921898          Y  90793868   C   T   do9172_90170_N2  subs       Liver\n",
       "22921899          Y  90794273   C   G   do9201_90802_N1  subs       Liver\n",
       "22921900          Y  90805575   T   A   do9197_90797_N2  subs       Liver\n",
       "\n",
       "[22921901 rows x 7 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "path = \"/workspace/projects/clustering_3d/o3d_analysys/datasets/input/cancer_mouse/raw_data/liver_strelka_mm39/Liver_DEN_strelka_July2017_mm10.txt\"\n",
    "liver_df = pd.read_table(path)\n",
    "liver_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "6425758d-23b7-487e-a21c-0eac1bbb8ffd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>do9710_91945_N1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000193812</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>do9848_93260_N7</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000193812</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>do9811_93131_N8</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000193812</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>do9181_90447_N3</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000193812</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>do9698_90447_N3</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000193812</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11858509</th>\n",
       "      <td>Mid1-ps1</td>\n",
       "      <td>do9449_91839_N1</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>ENSMUST00000167967</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11858510</th>\n",
       "      <td>Gm47283</td>\n",
       "      <td>do12459_91948_N4</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>ENSMUST00000179483</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11858511</th>\n",
       "      <td>Gm47283</td>\n",
       "      <td>do9172_90170_N2</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>ENSMUST00000179483</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11858512</th>\n",
       "      <td>Gm47283</td>\n",
       "      <td>do9201_90802_N1</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>ENSMUST00000179483</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11858513</th>\n",
       "      <td>Gm47283</td>\n",
       "      <td>do9197_90797_N2</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>ENSMUST00000179483</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11858514 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Hugo_Symbol Tumor_Sample_Barcode  \\\n",
       "0         4933401J01Rik      do9710_91945_N1   \n",
       "1         4933401J01Rik      do9848_93260_N7   \n",
       "2         4933401J01Rik      do9811_93131_N8   \n",
       "3         4933401J01Rik      do9181_90447_N3   \n",
       "4         4933401J01Rik      do9698_90447_N3   \n",
       "...                 ...                  ...   \n",
       "11858509       Mid1-ps1      do9449_91839_N1   \n",
       "11858510        Gm47283     do12459_91948_N4   \n",
       "11858511        Gm47283      do9172_90170_N2   \n",
       "11858512        Gm47283      do9201_90802_N1   \n",
       "11858513        Gm47283      do9197_90797_N2   \n",
       "\n",
       "                                Variant_Classification       Transcript_ID  \\\n",
       "0                                upstream_gene_variant  ENSMUST00000193812   \n",
       "1                                upstream_gene_variant  ENSMUST00000193812   \n",
       "2                                upstream_gene_variant  ENSMUST00000193812   \n",
       "3                                upstream_gene_variant  ENSMUST00000193812   \n",
       "4                                upstream_gene_variant  ENSMUST00000193812   \n",
       "...                                                ...                 ...   \n",
       "11858509  intron_variant,non_coding_transcript_variant  ENSMUST00000167967   \n",
       "11858510  intron_variant,non_coding_transcript_variant  ENSMUST00000179483   \n",
       "11858511  intron_variant,non_coding_transcript_variant  ENSMUST00000179483   \n",
       "11858512  intron_variant,non_coding_transcript_variant  ENSMUST00000179483   \n",
       "11858513  intron_variant,non_coding_transcript_variant  ENSMUST00000179483   \n",
       "\n",
       "         HGVSp_Short  \n",
       "0                NaN  \n",
       "1                NaN  \n",
       "2                NaN  \n",
       "3                NaN  \n",
       "4                NaN  \n",
       "...              ...  \n",
       "11858509         NaN  \n",
       "11858510         NaN  \n",
       "11858511         NaN  \n",
       "11858512         NaN  \n",
       "11858513         NaN  \n",
       "\n",
       "[11858514 rows x 5 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out_vep = pd.read_csv(f\"{mouse_path}/backup/liver_strelka.maf\", sep='\\t')\n",
    "out_vep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "81f66802-6c86-4d16-97d2-8add733bd6b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ENSMUST00000002487'], dtype=object)"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out_vep[out_vep[\"Hugo_Symbol\"] == \"Braf\"].Transcript_ID.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "963dce96-2aea-440e-bc9b-f60feeaf18d2",
   "metadata": {},
   "source": [
    "#### End"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "16fb0ead-507a-4848-a135-340d3f441a08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Xkr4</td>\n",
       "      <td>do9866_94316_N3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000070533</td>\n",
       "      <td>p.E639V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Xkr4</td>\n",
       "      <td>do9819_93243_N2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000070533</td>\n",
       "      <td>p.I587V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Xkr4</td>\n",
       "      <td>do9213_90880_N1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000070533</td>\n",
       "      <td>p.T575A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Xkr4</td>\n",
       "      <td>do9442_91837_N5</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000070533</td>\n",
       "      <td>p.E562G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Xkr4</td>\n",
       "      <td>do9766_93092_N7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000070533</td>\n",
       "      <td>p.D526Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175108</th>\n",
       "      <td>4933400A11Rik</td>\n",
       "      <td>do9714_91620_N3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000068874</td>\n",
       "      <td>p.H34L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175109</th>\n",
       "      <td>Mid1</td>\n",
       "      <td>do9236_91097_N1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000112105</td>\n",
       "      <td>p.T149I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175110</th>\n",
       "      <td>Mid1</td>\n",
       "      <td>do9464_91932_N1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000112105</td>\n",
       "      <td>p.D324G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175111</th>\n",
       "      <td>Mid1</td>\n",
       "      <td>do9877_91942_N3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000112105</td>\n",
       "      <td>p.L349P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175112</th>\n",
       "      <td>Mid1</td>\n",
       "      <td>do9836_93258_N10</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSMUST00000112105</td>\n",
       "      <td>p.Y555C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>175113 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Hugo_Symbol Tumor_Sample_Barcode Variant_Classification  \\\n",
       "0                Xkr4      do9866_94316_N3       missense_variant   \n",
       "1                Xkr4      do9819_93243_N2       missense_variant   \n",
       "2                Xkr4      do9213_90880_N1       missense_variant   \n",
       "3                Xkr4      do9442_91837_N5       missense_variant   \n",
       "4                Xkr4      do9766_93092_N7       missense_variant   \n",
       "...               ...                  ...                    ...   \n",
       "175108  4933400A11Rik      do9714_91620_N3       missense_variant   \n",
       "175109           Mid1      do9236_91097_N1       missense_variant   \n",
       "175110           Mid1      do9464_91932_N1       missense_variant   \n",
       "175111           Mid1      do9877_91942_N3       missense_variant   \n",
       "175112           Mid1     do9836_93258_N10       missense_variant   \n",
       "\n",
       "             Transcript_ID HGVSp_Short  \n",
       "0       ENSMUST00000070533     p.E639V  \n",
       "1       ENSMUST00000070533     p.I587V  \n",
       "2       ENSMUST00000070533     p.T575A  \n",
       "3       ENSMUST00000070533     p.E562G  \n",
       "4       ENSMUST00000070533     p.D526Y  \n",
       "...                    ...         ...  \n",
       "175108  ENSMUST00000068874      p.H34L  \n",
       "175109  ENSMUST00000112105     p.T149I  \n",
       "175110  ENSMUST00000112105     p.D324G  \n",
       "175111  ENSMUST00000112105     p.L349P  \n",
       "175112  ENSMUST00000112105     p.Y555C  \n",
       "\n",
       "[175113 rows x 5 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out_vep = pd.read_csv(f\"{mouse_path}/backup/liver_strelka.maf\", sep='\\t')\n",
    "out_vep = out_vep[out_vep[\"Variant_Classification\"] == \"missense_variant\"].reset_index(drop=True)\n",
    "out_vep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e16b0513-ef16-4ea4-8a5c-d85ae7498c34",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21615"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "uniprot_ids = os.listdir(\"/workspace/projects/clustering_3d/clustering_3d/datasets_mouse/pdb_structures\")\n",
    "uniprot_ids = [uni_id.split(\"-\")[1] for uni_id in list(set(uniprot_ids)) if uni_id.endswith(\".pdb\")]\n",
    "len(uniprot_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5c560463-6d88-4e9d-87cf-35a28baa3e67",
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "import requests\n",
    "import gzip\n",
    "import io\n",
    "\n",
    "def get_response_jobid(response):\n",
    "    \"\"\"\n",
    "    Get jobId after submitting ID Mapping job to UniprotKB. \n",
    "    \"\"\"\n",
    "    \n",
    "    try:\n",
    "        data = response.json()\n",
    "        job_id = data.get(\"jobId\")\n",
    "    except:\n",
    "        job_id = None\n",
    "        \n",
    "    return job_id\n",
    "\n",
    "\n",
    "def get_mapping_jobid(uniprot_ids):\n",
    "    \"\"\"\n",
    "    Submit an ID Mapping job to UniprotKB.\n",
    "    \"\"\"\n",
    "    \n",
    "    command = f\"https://rest.uniprot.org/idmapping/run?from=UniProtKB_AC-ID&to=UniProtKB&ids={','.join(uniprot_ids)}\"\n",
    "    response = requests.post(command)\n",
    "    job_id = get_response_jobid(response)\n",
    "    \n",
    "    i = 60\n",
    "    while job_id is None:\n",
    "        time.sleep(1) \n",
    "        job_id = get_response_jobid(response)\n",
    "        if i % 60 == 0:\n",
    "            print(f\"Requesting ID mapping job to UniprotKB for IDs.. [waited {i-59}s]\")\n",
    "        i += 1\n",
    "    \n",
    "    return job_id\n",
    "\n",
    "\n",
    "def load_df_from_url(url):\n",
    "    \"\"\"\n",
    "    Load a pandas dataframe from url.\n",
    "    \"\"\"\n",
    "\n",
    "    try:\n",
    "        response = requests.get(url)\n",
    "        decompressed_data = gzip.decompress(response.content)\n",
    "        df = pd.read_csv(io.BytesIO(decompressed_data), sep='\\t')\n",
    "    except:\n",
    "        df = None\n",
    "    \n",
    "    return df\n",
    "\n",
    "\n",
    "def split_lst_into_chunks(lst, chunk_size = 5000):\n",
    "    \"\"\"\n",
    "    Simple split a list into list of list of chunk_size elements.\n",
    "    \"\"\"\n",
    "    \n",
    "    return [lst[i:i+chunk_size] for i in range(0, len(lst), chunk_size)]\n",
    "\n",
    "\n",
    "def uniprot_to_hudo_df(uniprot_ids):\n",
    "    \"\"\"\n",
    "    Given a list of Uniprot IDs (from any species), request an Id \n",
    "    mapping job to UniprotKB to retrieve the corresponding Hugo \n",
    "    symbols and additional protein info. Return a pandas dataframe.\n",
    "    It is recommended to provide batches of IDs up to 5000 elements.\n",
    "    \"\"\"\n",
    "    \n",
    "    job_id = get_mapping_jobid(uniprot_ids)\n",
    "    url = f\"https://rest.uniprot.org/idmapping/uniprotkb/results/stream/{job_id}?compressed=true&fields=accession%2Cid%2Cprotein_name%2Cgene_names%2Corganism_name%2Clength&format=tsv\"\n",
    "    df = load_df_from_url(url)\n",
    "    \n",
    "    i = 60\n",
    "    while df is None:\n",
    "        time.sleep(1)\n",
    "        df = load_df_from_url(url)\n",
    "        if i % 60 == 0:\n",
    "            print(f\"Waiting for UniprotKB mapping job to produce url.. [waited {i-59}s]\")\n",
    "        i =+ 1 \n",
    "        \n",
    "    return df\n",
    "\n",
    "\n",
    "def convert_dict_hugo_to_uniprot(dict_uniprot_hugo):\n",
    "    \"\"\"\n",
    "    Convert a Uniprot IDs to Hugo symbol dictionary to a Hugo symbo to \n",
    "    Uniprot IDs dictionary, if multiple Hugo symbols are mapped to the \n",
    "    same Uniprot ID, add them as multiple keys.\n",
    "    \"\"\"\n",
    "    \n",
    "    dict_hugo_uniprot = {}\n",
    "\n",
    "    for uni_id, gene in dict_uniprot_hugo.items():\n",
    "        if type(gene) == str:\n",
    "            for g in gene.split(\" \"):\n",
    "                dict_hugo_uniprot[g] = uni_id\n",
    "                \n",
    "    return dict_hugo_uniprot\n",
    "\n",
    "\n",
    "def uniprot_to_hugo(uniprot_ids, hugo_as_keys=False):\n",
    "    \"\"\"\n",
    "    Given a list of Uniprot IDs (any species.), request an Id mapping \n",
    "    job to UniprotKB to retrieve the corresponding Hugo symbols. \n",
    "    Return a dictionary of Uniprot IDs to Hugo symbols or vice versa.\n",
    "    \"\"\"\n",
    "    \n",
    "    # Split uniprot IDs into chunks\n",
    "    uniprot_ids_lst = split_lst_into_chunks(uniprot_ids)\n",
    "    \n",
    "    # Get a dataframe including all IDs mapping info\n",
    "    df_lst = []\n",
    "    for ids in uniprot_ids_lst:\n",
    "        df = uniprot_to_hudo_df(ids)\n",
    "        df_lst.append(df)\n",
    "    df = pd.concat(df_lst)\n",
    "\n",
    "    # Get a dictionary for Uniprot ID to Hugo symbols\n",
    "    dictio = {}\n",
    "    for i,r in df[[\"Entry\", \"Gene Names\"]].iterrows():\n",
    "        uni_id, gene = r \n",
    "        dictio[uni_id] = gene\n",
    "    \n",
    "    # Convert to a dictionary of Hugo symbols to Uniprot IDs\n",
    "    if hugo_as_keys:\n",
    "        dictio = convert_dict_hugo_to_uniprot(dictio)\n",
    "            \n",
    "    return dictio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b6b258f7-0966-4722-8a9c-6818aa8ff1d3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Waiting for UniprotKB mapping job to produce url.. [waited 1s]\n",
      "CPU times: user 812 ms, sys: 28.1 ms, total: 840 ms\n",
      "Wall time: 25.2 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "dictio = uniprot_to_hugo(uniprot_ids[:5000], hugo_as_keys=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8a3c8841-afef-4dad-a653-56e727d06273",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Waiting for UniprotKB mapping job to produce url.. [waited 1s]\n",
      "CPU times: user 802 ms, sys: 15.5 ms, total: 818 ms\n",
      "Wall time: 24.5 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "dictio = uniprot_to_hugo(uniprot_ids[5000:10000], hugo_as_keys=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cccae36a-6eab-48ed-836b-7cd5c4f102dc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Waiting for UniprotKB mapping job to produce url.. [waited 1s]\n",
      "CPU times: user 816 ms, sys: 15.7 ms, total: 832 ms\n",
      "Wall time: 25.3 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "dictio = uniprot_to_hugo(uniprot_ids[10000:15000], hugo_as_keys=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "230cfeb2-7859-4fb6-9068-75b89a5836b0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Waiting for UniprotKB mapping job to produce url.. [waited 1s]\n",
      "CPU times: user 829 ms, sys: 19.9 ms, total: 848 ms\n",
      "Wall time: 25.6 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "dictio = uniprot_to_hugo(uniprot_ids[15000:20000], hugo_as_keys=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "620e8d52-95cf-4a95-8236-0a40ef721cb2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Waiting for UniprotKB mapping job to produce url.. [waited 1s]\n",
      "CPU times: user 324 ms, sys: 14.2 ms, total: 338 ms\n",
      "Wall time: 9.33 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "dictio = uniprot_to_hugo(uniprot_ids[20000:], hugo_as_keys=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a7337c14-bdc3-4853-806a-bb85e3d6588b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1.65 s, sys: 18.3 ms, total: 1.66 s\n",
      "Wall time: 18.4 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "dictio = uniprot_to_hugo(uniprot_ids, hugo_as_keys=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c37ebf92-4a2b-49fc-81af-060f7c63010a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save\n",
    "import json\n",
    "output = \"/workspace/projects/clustering_3d/clustering_3d/datasets_mouse/af_uniprot_to_gene_id.json\"\n",
    "with open(output, \"w\") as fp:\n",
    "    json.dump(dictio, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "975670f7-48ad-4394-9b3b-31f5a5ee258f",
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_to_gene_dict = dictio\n",
    "gene_to_uniprot_dict = convert_dict_hugo_to_uniprot(dictio)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1585789-630b-4be3-acd6-aeb241a01a06",
   "metadata": {},
   "source": [
    "### Seq df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "80c6bd8c-afd5-41df-b5c7-38b5ecf84e7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import gzip\n",
    "import io\n",
    "import csv\n",
    "import time\n",
    "import requests\n",
    "from difflib import SequenceMatcher\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "95769e86-7ac1-424e-9244-9cf16e38bdfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "import json\n",
    "import argparse\n",
    "import warnings\n",
    "from progressbar import progressbar\n",
    "import requests\n",
    "import time\n",
    "import re\n",
    "\n",
    "\n",
    "# Import from utils\n",
    "\n",
    "\n",
    "def rounded_up_division(num, den):\n",
    "    \"\"\"\n",
    "    Simply round up the result of the division.\n",
    "    \"\"\"\n",
    "    \n",
    "    return -(-num // den)\n",
    "\n",
    "\n",
    "def get_pos_fragments(mut_gene_df):\n",
    "    \"\"\"\n",
    "    Get the corresponding fragment of each position of the protein.\n",
    "    \"\"\"\n",
    "    \n",
    "    max_f = rounded_up_division(max(mut_gene_df[\"Pos\"]), 1400)\n",
    "    bins = [n * 1400 for n in range(0, max_f+1)]\n",
    "    group_names = list(range(1, max_f+1))\n",
    "    \n",
    "    return pd.cut(mut_gene_df[\"Pos\"], bins, labels = group_names)\n",
    "\n",
    "\n",
    "def get_af_id_from_pdb(path_structure):\n",
    "    \"\"\"\n",
    "    Get AlphaFold 2 identifier (UniprotID_F) from path\n",
    "    \"\"\"\n",
    "\n",
    "    return path_structure.split(\"AF-\")[1].split(\"-model\")[0]   \n",
    "\n",
    "\n",
    "def get_seq_from_pdb(path_structure):\n",
    "    \"\"\"\n",
    "    Get sequense of amino acid residues from PDB structure.\n",
    "    \"\"\"\n",
    "    \n",
    "    return np.array([record.seq for record in SeqIO.parse(path_structure, 'pdb-seqres')][0])\n",
    "\n",
    "\n",
    "def get_pdb_path_list_from_dir(path_dir):\n",
    "    \"\"\"\n",
    "    Takes as input a path of a given directory and it \n",
    "    outputs a list of paths of the contained PDB files.\n",
    "    \"\"\"\n",
    "\n",
    "    pdb_files = os.listdir(path_dir)\n",
    "    pdb_path_list = [f\"{path_dir}{f}\" for f in pdb_files if re.search('.\\.pdb$', f) is not None]\n",
    "    return pdb_path_list\n",
    "\n",
    "\n",
    "# Backtranslation\n",
    "\n",
    "def translate_dna(dna_seq):\n",
    "    \"\"\"\n",
    "    Translate dna sequence to protein.\n",
    "    \"\"\"\n",
    "    \n",
    "    dna_seq = Seq(dna_seq)\n",
    "    return str(dna_seq.translate()) \n",
    "\n",
    "\n",
    "def get_seq_similarity(a, b, decimals=3):\n",
    "    \"\"\"\n",
    "    Compute the similarity ratio between sequences a and b.\n",
    "    \"\"\"\n",
    "    \n",
    "    return round(SequenceMatcher(a=a, b=b).ratio(), decimals)\n",
    "\n",
    "\n",
    "## Other functions\n",
    "\n",
    "\n",
    "def get_seq_df_from_dir(input_path, uniprot_to_gene_dict):\n",
    "    \"\"\"\n",
    "    Parse any PDB structure from a given directory and create \n",
    "    a dataframe including HUGO symbol, Uniprot-ID, AF fragment,\n",
    "    and protein sequence.\n",
    "    \"\"\"\n",
    "    \n",
    "    # Get all PDB path in directory whose IDs can be mapped to HUGO symbols\n",
    "    list_prot_path = get_pdb_path_list_from_dir(input_path)[:1000]            ############################################################ SUBSET\n",
    "    \n",
    "    list_prot_path = [path for path in list_prot_path if get_af_id_from_pdb(path).split(\"-F\")[0] in uniprot_to_gene_dict.keys()]\n",
    "    pdb_not_mapped = set([get_af_id_from_pdb(path).split(\"-F\")[0] for path in list_prot_path if get_af_id_from_pdb(path).split(\"-F\")[0] not in uniprot_to_gene_dict.keys()])\n",
    "    if len(pdb_not_mapped) > 0:                                 \n",
    "        warnings.warn(f\"WARNING........... {len(pdb_not_mapped)} Uniprot-ID not found in the Uniprot-HUGO mapping dictionary\")\n",
    "\n",
    "    # Get Uniprot ID, HUGO, F and protein sequence of any PDB in dir\n",
    "    gene_lst = []\n",
    "    uni_id_lst = []\n",
    "    f_lst = []\n",
    "    seq_lst = []\n",
    "\n",
    "    for path_structure in progressbar(list_prot_path):\n",
    "        uniprot_id, f = get_af_id_from_pdb(path_structure).split(\"-F\")\n",
    "        gene = uniprot_to_gene_dict[uniprot_id]\n",
    "        seq = \"\".join(list(get_seq_from_pdb(path_structure)))\n",
    "        gene_lst.append(gene)\n",
    "        uni_id_lst.append(uniprot_id)\n",
    "        f_lst.append(f)\n",
    "        seq_lst.append(seq)\n",
    "\n",
    "    seq_df = pd.DataFrame({\"Gene\" : gene_lst, \"Uniprot_ID\" : uni_id_lst, \"F\" : f_lst, \"Seq\" : seq_lst}).sort_values([\"Gene\", \"F\"])\n",
    "\n",
    "    return seq_df\n",
    "\n",
    "\n",
    "def backtranseq(protein_seqs, organism = \"Homo sapiens\"):\n",
    "    \"\"\"\n",
    "    Perform backtranslation from proteins to \n",
    "    DNA sequences using EMBOS backtranseq.    \n",
    "    \"\"\"\n",
    "    \n",
    "    # Define the API endpoints\n",
    "    run_url = \"https://www.ebi.ac.uk/Tools/services/rest/emboss_backtranseq/run\"\n",
    "    status_url = \"https://www.ebi.ac.uk/Tools/services/rest/emboss_backtranseq/status/\"\n",
    "    result_url = \"https://www.ebi.ac.uk/Tools/services/rest/emboss_backtranseq/result/\"\n",
    "    \n",
    "    # Define the parameters for the API request (an email address must be included)\n",
    "    params = {\"email\": \"emaple.email@irbbarcelona.org\",\n",
    "              \"sequence\": protein_seqs,\n",
    "              \"outseqformat\": \"plain\",\n",
    "              \"molecule\": \"dna\",\n",
    "              \"organism\": organism}\n",
    "    \n",
    "    # Submit the job request and retrieve the job ID\n",
    "    response = requests.post(run_url, data=params)\n",
    "    job_id = response.text.strip()\n",
    "\n",
    "    # Wait for the job to complete\n",
    "    status = \"\"\n",
    "    while status != \"FINISHED\":\n",
    "        time.sleep(10)\n",
    "        result = requests.get(status_url + job_id)\n",
    "        status = result.text.strip()\n",
    "\n",
    "    # Retrieve the results of the job\n",
    "    result = requests.get(result_url + job_id + \"/out\")\n",
    "    dna_seq = result.text.strip()\n",
    "    \n",
    "    return dna_seq\n",
    "\n",
    "\n",
    "def batch_backtranseq(df, batch_size, organism = \"Homo sapiens\"):\n",
    "    \"\"\"\n",
    "    Given a dataframe including protein sequences, it divides the \n",
    "    sequences into batches of a given size and run EMBOSS backtranseq \n",
    "    (https://www.ebi.ac.uk/Tools/st/emboss_backtranseq/) to translate \n",
    "    them into DNA sequences.\n",
    "    \"\"\"\n",
    "    \n",
    "    batches = df.groupby(df.index // batch_size)\n",
    "    lst_batches = []\n",
    "        \n",
    "    # Iterate over batches\n",
    "    for _, batch in progressbar(batches):\n",
    "\n",
    "        # Get input format for backtranseq \n",
    "        batch_seq = \"\\n\".join(batch.reset_index(drop=True).apply(lambda x: f'>\\n{x[\"Seq\"]}', axis=1).values)\n",
    "\n",
    "        # Run backtranseq\n",
    "        batch_dna = backtranseq(batch_seq, organism=organism)\n",
    "\n",
    "        # Parse output\n",
    "        batch_dna = re.split(\">EMBOSS_\\d+\", batch_dna.replace(\"\\n\", \"\"))[1:]\n",
    "\n",
    "        batch[\"Seq_dna\"] = batch_dna\n",
    "        lst_batches.append(batch)\n",
    "        \n",
    "    return pd.concat(lst_batches)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "75760c59-5f6d-41fd-a92f-fc9aaf6ba606",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100% (991 of 991) |######################| Elapsed Time: 0:00:01 Time:  0:00:01\n"
     ]
    }
   ],
   "source": [
    "input = \"/workspace/projects/clustering_3d/clustering_3d/datasets_mouse/pdb_structures/\"\n",
    "seq_df = get_seq_df_from_dir(input, uniprot_to_gene_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "bea2328e-7081-4330-81e4-c7c444156401",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>F</th>\n",
       "      <th>Seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>1700014B07Rik</td>\n",
       "      <td>A0A5F8MPC3</td>\n",
       "      <td>1</td>\n",
       "      <td>METAPAFPRAALPGPRLRAAGPARLTGRQQSPLRPRPPPSWRGRTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>1700017N19Rik</td>\n",
       "      <td>A0A087WPJ1</td>\n",
       "      <td>1</td>\n",
       "      <td>MAEVGNDCYFFVNSTCIKGSQCRFRHCEEALGSDTVCSLWRERKCL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>1700018G05Rik</td>\n",
       "      <td>A0A1B0GSI2</td>\n",
       "      <td>1</td>\n",
       "      <td>MIVMHLVGHPQLTGTEGVATAHTKENSSRGPRETREPAPCTRSWRS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>1700019A02Rik</td>\n",
       "      <td>A0A087WPV9</td>\n",
       "      <td>1</td>\n",
       "      <td>MVGSSWCLHPHGGLLEIIEMNDKLRNRQEIVNGLNIFLEENEEEAY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>1700025B11Rik</td>\n",
       "      <td>A0A087WNZ6</td>\n",
       "      <td>1</td>\n",
       "      <td>MSTKNEEQNEDQSESVVIPHIQDHHCLAILAFCFFFPLGYLAYRSS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>Zscan4-ps2</td>\n",
       "      <td>A0A1C7CYU6</td>\n",
       "      <td>1</td>\n",
       "      <td>MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>Zscan4-ps3</td>\n",
       "      <td>A0A1B0GRH6</td>\n",
       "      <td>1</td>\n",
       "      <td>MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>Zscan4e</td>\n",
       "      <td>A0A1B0GRE8</td>\n",
       "      <td>1</td>\n",
       "      <td>MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754</th>\n",
       "      <td>NaN</td>\n",
       "      <td>A0A286YDM5</td>\n",
       "      <td>1</td>\n",
       "      <td>MEVRNVTEFVLLGLTSNPRSQVLLFILFLVIYLLTLAGNLLMLLLI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>NaN</td>\n",
       "      <td>A0A5F8MPE6</td>\n",
       "      <td>1</td>\n",
       "      <td>MEKSSNTTGSGGPKSDSQLPEKLRRTFSARAAKIKARKANRETSAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>991 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  Uniprot_ID  F  \\\n",
       "886  1700014B07Rik  A0A5F8MPC3  1   \n",
       "218  1700017N19Rik  A0A087WPJ1  1   \n",
       "602  1700018G05Rik  A0A1B0GSI2  1   \n",
       "223  1700019A02Rik  A0A087WPV9  1   \n",
       "203  1700025B11Rik  A0A087WNZ6  1   \n",
       "..             ...         ... ..   \n",
       "628     Zscan4-ps2  A0A1C7CYU6  1   \n",
       "568     Zscan4-ps3  A0A1B0GRH6  1   \n",
       "565        Zscan4e  A0A1B0GRE8  1   \n",
       "754            NaN  A0A286YDM5  1   \n",
       "888            NaN  A0A5F8MPE6  1   \n",
       "\n",
       "                                                   Seq  \n",
       "886  METAPAFPRAALPGPRLRAAGPARLTGRQQSPLRPRPPPSWRGRTA...  \n",
       "218  MAEVGNDCYFFVNSTCIKGSQCRFRHCEEALGSDTVCSLWRERKCL...  \n",
       "602  MIVMHLVGHPQLTGTEGVATAHTKENSSRGPRETREPAPCTRSWRS...  \n",
       "223  MVGSSWCLHPHGGLLEIIEMNDKLRNRQEIVNGLNIFLEENEEEAY...  \n",
       "203  MSTKNEEQNEDQSESVVIPHIQDHHCLAILAFCFFFPLGYLAYRSS...  \n",
       "..                                                 ...  \n",
       "628  MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...  \n",
       "568  MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...  \n",
       "565  MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...  \n",
       "754  MEVRNVTEFVLLGLTSNPRSQVLLFILFLVIYLLTLAGNLLMLLLI...  \n",
       "888  MEKSSNTTGSGGPKSDSQLPEKLRRTFSARAAKIKARKANRETSAA...  \n",
       "\n",
       "[991 rows x 4 columns]"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "7d632d2e-3497-428a-a044-c4ace66dd443",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0% (0 of 2) |                          | Elapsed Time: 0:00:00 ETA:  --:--:--"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Performing back translation\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100% (2 of 2) |##########################| Elapsed Time: 0:00:50 Time:  0:00:50\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>F</th>\n",
       "      <th>Seq</th>\n",
       "      <th>Seq_dna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>1700017N19Rik</td>\n",
       "      <td>A0A087WPJ1</td>\n",
       "      <td>1</td>\n",
       "      <td>MAEVGNDCYFFVNSTCIKGSQCRFRHCEEALGSDTVCSLWRERKCL...</td>\n",
       "      <td>ATGGCCGAGGTGGGCAACGACTGCTACTTCTTCGTGAACAGCACCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>1700019A02Rik</td>\n",
       "      <td>A0A087WPV9</td>\n",
       "      <td>1</td>\n",
       "      <td>MVGSSWCLHPHGGLLEIIEMNDKLRNRQEIVNGLNIFLEENEEEAY...</td>\n",
       "      <td>ATGGTGGGCAGCAGCTGGTGCCTGCACCCCCACGGCGGCCTGCTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>1700025B11Rik</td>\n",
       "      <td>A0A087WNZ6</td>\n",
       "      <td>1</td>\n",
       "      <td>MSTKNEEQNEDQSESVVIPHIQDHHCLAILAFCFFFPLGYLAYRSS...</td>\n",
       "      <td>ATGAGCACCAAGAACGAGGAGCAGAACGAGGACCAGAGCGAGAGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>2210017I01Rik</td>\n",
       "      <td>A0A0A6YVV1</td>\n",
       "      <td>1</td>\n",
       "      <td>MSSQQHQQKCKLSANCPPKCPQQGPQVPTSCLPLSAPPAPACCVST...</td>\n",
       "      <td>ATGAGCAGCCAGCAGCACCAGCAGAAGTGCAAGCTGAGCGCCAACT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>2310009B15Rik</td>\n",
       "      <td>A0A0A6YWI4</td>\n",
       "      <td>1</td>\n",
       "      <td>MAAQGIWTLARVSLGRPPPRAEGFRQQLNSTLGSTSEGARGGSAAD...</td>\n",
       "      <td>ATGGCCGCCCAGGGCATCTGGACCCTGGCCAGGGTGAGCCTGGGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>Zscan4-ps2</td>\n",
       "      <td>A0A1C7CYU6</td>\n",
       "      <td>1</td>\n",
       "      <td>MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...</td>\n",
       "      <td>ATGGCCAGCCAGCAGGCCCCCGCCAAGGACCTGCAGACCAACAACC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>Zscan4-ps3</td>\n",
       "      <td>A0A1B0GRH6</td>\n",
       "      <td>1</td>\n",
       "      <td>MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...</td>\n",
       "      <td>ATGGCCAGCCAGCAGGCCCCCGCCAAGGACCTGCAGACCAACAACC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>Zscan4e</td>\n",
       "      <td>A0A1B0GRE8</td>\n",
       "      <td>1</td>\n",
       "      <td>MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...</td>\n",
       "      <td>ATGGCCAGCCAGCAGGCCCCCGCCAAGGACCTGCAGACCAACAACC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754</th>\n",
       "      <td>NaN</td>\n",
       "      <td>A0A286YDM5</td>\n",
       "      <td>1</td>\n",
       "      <td>MEVRNVTEFVLLGLTSNPRSQVLLFILFLVIYLLTLAGNLLMLLLI...</td>\n",
       "      <td>ATGGAGGTGAGGAACGTGACCGAGTTCGTGCTGCTGGGCCTGACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>NaN</td>\n",
       "      <td>A0A5F8MPE6</td>\n",
       "      <td>1</td>\n",
       "      <td>MEKSSNTTGSGGPKSDSQLPEKLRRTFSARAAKIKARKANRETSAA...</td>\n",
       "      <td>ATGGAGAAGAGCAGCAACACCACCGGCAGCGGCGGCCCCAAGAGCG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>991 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  Uniprot_ID  F  \\\n",
       "218  1700017N19Rik  A0A087WPJ1  1   \n",
       "223  1700019A02Rik  A0A087WPV9  1   \n",
       "203  1700025B11Rik  A0A087WNZ6  1   \n",
       "288  2210017I01Rik  A0A0A6YVV1  1   \n",
       "296  2310009B15Rik  A0A0A6YWI4  1   \n",
       "..             ...         ... ..   \n",
       "628     Zscan4-ps2  A0A1C7CYU6  1   \n",
       "568     Zscan4-ps3  A0A1B0GRH6  1   \n",
       "565        Zscan4e  A0A1B0GRE8  1   \n",
       "754            NaN  A0A286YDM5  1   \n",
       "888            NaN  A0A5F8MPE6  1   \n",
       "\n",
       "                                                   Seq  \\\n",
       "218  MAEVGNDCYFFVNSTCIKGSQCRFRHCEEALGSDTVCSLWRERKCL...   \n",
       "223  MVGSSWCLHPHGGLLEIIEMNDKLRNRQEIVNGLNIFLEENEEEAY...   \n",
       "203  MSTKNEEQNEDQSESVVIPHIQDHHCLAILAFCFFFPLGYLAYRSS...   \n",
       "288  MSSQQHQQKCKLSANCPPKCPQQGPQVPTSCLPLSAPPAPACCVST...   \n",
       "296  MAAQGIWTLARVSLGRPPPRAEGFRQQLNSTLGSTSEGARGGSAAD...   \n",
       "..                                                 ...   \n",
       "628  MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...   \n",
       "568  MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...   \n",
       "565  MASQQAPAKDLQTNNLEFTPTDSSGVQWAEDISNSPSAQLNFSPSN...   \n",
       "754  MEVRNVTEFVLLGLTSNPRSQVLLFILFLVIYLLTLAGNLLMLLLI...   \n",
       "888  MEKSSNTTGSGGPKSDSQLPEKLRRTFSARAAKIKARKANRETSAA...   \n",
       "\n",
       "                                               Seq_dna  \n",
       "218  ATGGCCGAGGTGGGCAACGACTGCTACTTCTTCGTGAACAGCACCT...  \n",
       "223  ATGGTGGGCAGCAGCTGGTGCCTGCACCCCCACGGCGGCCTGCTGG...  \n",
       "203  ATGAGCACCAAGAACGAGGAGCAGAACGAGGACCAGAGCGAGAGCG...  \n",
       "288  ATGAGCAGCCAGCAGCACCAGCAGAAGTGCAAGCTGAGCGCCAACT...  \n",
       "296  ATGGCCGCCCAGGGCATCTGGACCCTGGCCAGGGTGAGCCTGGGCA...  \n",
       "..                                                 ...  \n",
       "628  ATGGCCAGCCAGCAGGCCCCCGCCAAGGACCTGCAGACCAACAACC...  \n",
       "568  ATGGCCAGCCAGCAGGCCCCCGCCAAGGACCTGCAGACCAACAACC...  \n",
       "565  ATGGCCAGCCAGCAGGCCCCCGCCAAGGACCTGCAGACCAACAACC...  \n",
       "754  ATGGAGGTGAGGAACGTGACCGAGTTCGTGCTGCTGGGCCTGACCA...  \n",
       "888  ATGGAGAAGAGCAGCAACACCACCGGCAGCGGCGGCCCCAAGAGCG...  \n",
       "\n",
       "[991 rows x 5 columns]"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Annotate df with DNA sequences\n",
    "print(\"Performing back translation\")\n",
    "seq_df = batch_backtranseq(seq_df, 500, organism = \"Mus musculus\")\n",
    "seq_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "040507fd-57d0-4f85-af4d-7e4e8042c79c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_extra_genes_to_seq_df(seq_df, uniprot_to_gene_dict):\n",
    "    \"\"\"\n",
    "    If multiple genes are mapping to a given Uniprot_ID, add \n",
    "    each gene name with corresponding sequence info to the seq_df.\n",
    "    \"\"\"\n",
    "    \n",
    "    lst_extra_genes = []\n",
    "\n",
    "    for _, seq_row in seq_df.iterrows():\n",
    "\n",
    "        uni_id = seq_row[\"Uniprot_ID\"]\n",
    "        gene_id = uniprot_to_gene_dict[uni_id]\n",
    "        if pd.isnull(gene_id) == False:\n",
    "\n",
    "            gene_id = gene_id.split(\" \")\n",
    "            if len(gene_id) > 1:\n",
    "\n",
    "                for gene in gene_id:\n",
    "                    if gene != seq_row[\"Gene\"]:\n",
    "                        lst_extra_genes.append((gene, seq_row[\"Uniprot_ID\"], seq_row[\"F\"], seq_row[\"Seq\"], seq_row[\"Seq_dna\"]))\n",
    "\n",
    "    seq_df_extra_genes = pd.DataFrame(lst_extra_genes, columns=[\"Gene\", \"Uniprot_ID\", \"F\", \"Seq\", \"Seq_dna\"])\n",
    "    \n",
    "    return pd.concat((seq_df, seq_df_extra_genes)).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "fe210346-4686-48b1-83d1-9f5967055761",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>F</th>\n",
       "      <th>Seq</th>\n",
       "      <th>Seq_dna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1700017N19Rik</td>\n",
       "      <td>A0A087WPJ1</td>\n",
       "      <td>1</td>\n",
       "      <td>MAEVGNDCYFFVNSTCIKGSQCRFRHCEEALGSDTVCSLWRERKCL...</td>\n",
       "      <td>ATGGCCGAGGTGGGCAACGACTGCTACTTCTTCGTGAACAGCACCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1700019A02Rik</td>\n",
       "      <td>A0A087WPV9</td>\n",
       "      <td>1</td>\n",
       "      <td>MVGSSWCLHPHGGLLEIIEMNDKLRNRQEIVNGLNIFLEENEEEAY...</td>\n",
       "      <td>ATGGTGGGCAGCAGCTGGTGCCTGCACCCCCACGGCGGCCTGCTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1700025B11Rik</td>\n",
       "      <td>A0A087WNZ6</td>\n",
       "      <td>1</td>\n",
       "      <td>MSTKNEEQNEDQSESVVIPHIQDHHCLAILAFCFFFPLGYLAYRSS...</td>\n",
       "      <td>ATGAGCACCAAGAACGAGGAGCAGAACGAGGACCAGAGCGAGAGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2210017I01Rik</td>\n",
       "      <td>A0A0A6YVV1</td>\n",
       "      <td>1</td>\n",
       "      <td>MSSQQHQQKCKLSANCPPKCPQQGPQVPTSCLPLSAPPAPACCVST...</td>\n",
       "      <td>ATGAGCAGCCAGCAGCACCAGCAGAAGTGCAAGCTGAGCGCCAACT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2310009B15Rik</td>\n",
       "      <td>A0A0A6YWI4</td>\n",
       "      <td>1</td>\n",
       "      <td>MAAQGIWTLARVSLGRPPPRAEGFRQQLNSTLGSTSEGARGGSAAD...</td>\n",
       "      <td>ATGGCCGCCCAGGGCATCTGGACCCTGGCCAGGGTGAGCCTGGGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1527</th>\n",
       "      <td>Gm262</td>\n",
       "      <td>A0PK84</td>\n",
       "      <td>1</td>\n",
       "      <td>MLALRLLNVVAPAYFLCISLVTFVLQLFLFLPSMREDPTATPLFSP...</td>\n",
       "      <td>ATGCTGGCCCTGAGGCTGCTGAACGTGGTGGCCCCCGCCTACTTCC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1528</th>\n",
       "      <td>Znf335</td>\n",
       "      <td>A2A5K6</td>\n",
       "      <td>1</td>\n",
       "      <td>MEENEVESSSDAAPRPGQPEEPSESGLGVCTSEAVSADSSDAATVP...</td>\n",
       "      <td>ATGGAGGAGAACGAGGTGGAGAGCAGCAGCGACGCCGCCCCCAGGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1529</th>\n",
       "      <td>Zfp335</td>\n",
       "      <td>A2A5K6</td>\n",
       "      <td>1</td>\n",
       "      <td>MEENEVESSSDAAPRPGQPEEPSESGLGVCTSEAVSADSSDAATVP...</td>\n",
       "      <td>ATGGAGGAGAACGAGGTGGAGAGCAGCAGCGACGCCGCCCCCAGGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1530</th>\n",
       "      <td>Znf385c</td>\n",
       "      <td>A2A5E6</td>\n",
       "      <td>1</td>\n",
       "      <td>MLLGPASGSPSPLLASLTLPARPLQPPLDLKHLLAFHLNGTTPLSL...</td>\n",
       "      <td>ATGCTGCTGGGCCCCGCCAGCGGCAGCCCCAGCCCCCTGCTGGCCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1531</th>\n",
       "      <td>Zfp385c</td>\n",
       "      <td>A2A5E6</td>\n",
       "      <td>1</td>\n",
       "      <td>MLLGPASGSPSPLLASLTLPARPLQPPLDLKHLLAFHLNGTTPLSL...</td>\n",
       "      <td>ATGCTGCTGGGCCCCGCCAGCGGCAGCCCCAGCCCCCTGCTGGCCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1532 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gene  Uniprot_ID  F  \\\n",
       "0     1700017N19Rik  A0A087WPJ1  1   \n",
       "1     1700019A02Rik  A0A087WPV9  1   \n",
       "2     1700025B11Rik  A0A087WNZ6  1   \n",
       "3     2210017I01Rik  A0A0A6YVV1  1   \n",
       "4     2310009B15Rik  A0A0A6YWI4  1   \n",
       "...             ...         ... ..   \n",
       "1527          Gm262      A0PK84  1   \n",
       "1528         Znf335      A2A5K6  1   \n",
       "1529         Zfp335      A2A5K6  1   \n",
       "1530        Znf385c      A2A5E6  1   \n",
       "1531        Zfp385c      A2A5E6  1   \n",
       "\n",
       "                                                    Seq  \\\n",
       "0     MAEVGNDCYFFVNSTCIKGSQCRFRHCEEALGSDTVCSLWRERKCL...   \n",
       "1     MVGSSWCLHPHGGLLEIIEMNDKLRNRQEIVNGLNIFLEENEEEAY...   \n",
       "2     MSTKNEEQNEDQSESVVIPHIQDHHCLAILAFCFFFPLGYLAYRSS...   \n",
       "3     MSSQQHQQKCKLSANCPPKCPQQGPQVPTSCLPLSAPPAPACCVST...   \n",
       "4     MAAQGIWTLARVSLGRPPPRAEGFRQQLNSTLGSTSEGARGGSAAD...   \n",
       "...                                                 ...   \n",
       "1527  MLALRLLNVVAPAYFLCISLVTFVLQLFLFLPSMREDPTATPLFSP...   \n",
       "1528  MEENEVESSSDAAPRPGQPEEPSESGLGVCTSEAVSADSSDAATVP...   \n",
       "1529  MEENEVESSSDAAPRPGQPEEPSESGLGVCTSEAVSADSSDAATVP...   \n",
       "1530  MLLGPASGSPSPLLASLTLPARPLQPPLDLKHLLAFHLNGTTPLSL...   \n",
       "1531  MLLGPASGSPSPLLASLTLPARPLQPPLDLKHLLAFHLNGTTPLSL...   \n",
       "\n",
       "                                                Seq_dna  \n",
       "0     ATGGCCGAGGTGGGCAACGACTGCTACTTCTTCGTGAACAGCACCT...  \n",
       "1     ATGGTGGGCAGCAGCTGGTGCCTGCACCCCCACGGCGGCCTGCTGG...  \n",
       "2     ATGAGCACCAAGAACGAGGAGCAGAACGAGGACCAGAGCGAGAGCG...  \n",
       "3     ATGAGCAGCCAGCAGCACCAGCAGAAGTGCAAGCTGAGCGCCAACT...  \n",
       "4     ATGGCCGCCCAGGGCATCTGGACCCTGGCCAGGGTGAGCCTGGGCA...  \n",
       "...                                                 ...  \n",
       "1527  ATGCTGGCCCTGAGGCTGCTGAACGTGGTGGCCCCCGCCTACTTCC...  \n",
       "1528  ATGGAGGAGAACGAGGTGGAGAGCAGCAGCGACGCCGCCCCCAGGC...  \n",
       "1529  ATGGAGGAGAACGAGGTGGAGAGCAGCAGCGACGCCGCCCCCAGGC...  \n",
       "1530  ATGCTGCTGGGCCCCGCCAGCGGCAGCCCCAGCCCCCTGCTGGCCA...  \n",
       "1531  ATGCTGCTGGGCCCCGCCAGCGGCAGCCCCAGCCCCCTGCTGGCCA...  \n",
       "\n",
       "[1532 rows x 5 columns]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_df_extra = add_extra_genes_to_seq_df(seq_df, uniprot_to_gene_dict)\n",
    "seq_df_extra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7ebe602-77a8-4a5e-80d5-f28154c7711b",
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_to_gene_dict = None\n",
    "output_seq_df = None\n",
    "\n",
    "# Load Uniprot ID to HUGO mapping\n",
    "if uniprot_to_gene_dict is not None:\n",
    "    uniprot_to_gene_dict = json.load(open(uniprot_to_gene_dict)) \n",
    "else:\n",
    "    uniprot_to_gene_dict = uniprot_to_hugo()  \n",
    "\n",
    "# Create a dataframe with protein sequences\n",
    "print(\"\\nGenerating sequence df..\")\n",
    "seq_df = get_seq_df_from_dir(input, uniprot_to_gene_dict)\n",
    "\n",
    "# Annotate df with DNA sequences\n",
    "print(\"Performing back translation\")\n",
    "seq_df = batch_backtranseq(seq_df, 500)\n",
    "#seq_df.to_csv(output_seq_df, index=False)\n",
    "\n",
    "sim_ratio = sum(seq_df.apply(lambda x: get_seq_similarity(x.Seq, translate_dna(x.Seq_dna)), axis=1)) / len(seq_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd0701e1-a486-49e3-82d1-0efe47b123dd",
   "metadata": {},
   "source": [
    "## Get mut rate (BBG signature)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e5f786e-7f5d-4680-953c-f09b902e6f3b",
   "metadata": {},
   "source": [
    "### Parse input for bbg signature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4b98a3c8-9806-4ba2-9be6-da1c15feee65",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gzip\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6ed9d5b2-5de9-494a-af8a-4b234c65c71a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/jobs/spellegrini/8898314/ipykernel_2312/2070478027.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  mouse_df_original = pd.read_csv(path, sep=\"\\t\")\n"
     ]
    }
   ],
   "source": [
    "mouse_path = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data\" \n",
    "path = f\"{mouse_path}/Liver_DEN_strelka_July2017_mm10.txt\"\n",
    "mouse_df_original = pd.read_csv(path, sep=\"\\t\")\n",
    "\n",
    "# Save DataFrame as a compressed TSV file\n",
    "with gzip.open(f\"{mouse_path}/liver_strelka.in.tsv.gz\", 'wt') as file:\n",
    "    mouse_df_original.drop(columns=[\"TYPE\", \"CANCER_TYPE\"]).to_csv(file, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1477cd7a-0c86-4e96-ad18-a9b2f1f58af0",
   "metadata": {},
   "source": [
    "### Get regions file from bgreference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "6f0a97e9-9b1d-4256-905e-7186ec436a4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "\n",
    "from bgreference import refseq\n",
    "\n",
    "CHR = [str(i) for i in range(1, 23)] + ['X', 'Y']\n",
    "\n",
    "\n",
    "def compute_sizes(genome, kmer):\n",
    "    sizes = []\n",
    "    for chr_ in CHR:\n",
    "        seq = refseq(genome, chr_, start=(1 + kmer // 2), size=None)\n",
    "        sizes.append(tuple(map(str, (chr_, 1 + kmer // 2, len(seq) - kmer // 2))))\n",
    "    return sizes\n",
    "\n",
    "\n",
    "def write(sizes):\n",
    "    print('\\t'.join(('CHROMOSOME', 'START', 'END')))\n",
    "    for s in sizes:\n",
    "        print('\\t'.join(s))\n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    genome = sys.argv[1]\n",
    "    kmer = int(sys.argv[2])\n",
    "    write(compute_sizes(genome, kmer))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23657708-c35e-4997-bda3-871fce58fb16",
   "metadata": {},
   "source": [
    "### Download WG regions from biomart\n",
    "\n",
    "Downloaded from "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ba432f14-4548-492d-aaf6-6bb48c37a076",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Liver_DEN_strelka_July2017_mm10.txt  mm10ToMm39.over.chain.gz\n",
      "liftover.py\t\t\t     mm39_wg_regions.tsv.gz\n"
     ]
    }
   ],
   "source": [
    "!ls /workspace/projects/clustering_3d/o3d_analysys/datasets/input/cancer_mouse/raw_data/liver_strelka_mm39/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "3a96848e-7304-4ab1-ae18-56f085bc1cbc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>SIZE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>195154279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>181755017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>169476592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3</td>\n",
       "      <td>159745316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4</td>\n",
       "      <td>156860686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>chrUn_GL456396v1</td>\n",
       "      <td>21240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chrUn_GL456368v1</td>\n",
       "      <td>20208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>chrM</td>\n",
       "      <td>16299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>chr1_MU069434v1_random</td>\n",
       "      <td>8412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>chr4_JH584295v1_random</td>\n",
       "      <td>1976</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                CHROMOSOME       SIZE\n",
       "0                     chr1  195154279\n",
       "1                     chr2  181755017\n",
       "2                     chrX  169476592\n",
       "3                     chr3  159745316\n",
       "4                     chr4  156860686\n",
       "..                     ...        ...\n",
       "56        chrUn_GL456396v1      21240\n",
       "57        chrUn_GL456368v1      20208\n",
       "58                    chrM      16299\n",
       "59  chr1_MU069434v1_random       8412\n",
       "60  chr4_JH584295v1_random       1976\n",
       "\n",
       "[61 rows x 2 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path_regions = \"/workspace/projects/clustering_3d/o3d_analysys/datasets/input/cancer_mouse/raw_data/liver_strelka_mm39/mm39_wg_regions.tsv.gz\"\n",
    "wg_regions = pd.read_csv(path_regions, sep=\"\\t\", header=None)\n",
    "wg_regions.columns = \"CHROMOSOME\", \"SIZE\", \"INFO\"\n",
    "wg_regions = wg_regions.drop(columns=[\"INFO\"])\n",
    "wg_regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d990660-ab78-48d7-9ecc-c069fd784391",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sort_regions(regions):\n",
    "    \"\"\"\n",
    "    Sort regions file.\n",
    "    \"\"\"\n",
    "    \n",
    "    df = regions\n",
    "\n",
    "    # Define the custom sort order for Chromosome\n",
    "    chromosome_order = {'1': 1, '2': 2, '3': 3, '4': 4, '5': 5, '6': 6, '7': 7, '8': 8, '9': 9, '10': 10,\n",
    "                        '11': 11, '12': 12, '13': 13, '14': 14, '15': 15, '16': 16, '17': 17, '18': 18, '19': 19,\n",
    "                        '20': 20, '21': 21, 'X': 22, 'Y': 23, 'M': 24}\n",
    "\n",
    "    # Convert 'Chromosome' column to a temporary categorical column for sorting\n",
    "    df['CHROMOSOME'] = df['CHROMOSOME'].astype(str)\n",
    "    df['Chromosome_sort'] = df['CHROMOSOME'].map(chromosome_order)\n",
    "    # Sort and drop temporary col\n",
    "    sorted_df = df.sort_values(by=['Chromosome_sort'], ignore_index=True)\n",
    "    sorted_df.drop('Chromosome_sort', axis=1, inplace=True)\n",
    "\n",
    "    return sorted_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "d79b2faa-ed3b-4e6f-a29b-6aa86cda9936",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_regions = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data/mm10_wg_regions.tsv\"\n",
    "wg_regions = pd.read_csv(path_regions, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "226ec730-eb00-4fb2-bfd7-1b15762099c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Depending on pytabix version used, it can be 0-index or 1-index. \n",
    "\n",
    "wg_regions[\"START\"] = 2\n",
    "wg_regions[\"END\"] = wg_regions[\"size\"] - wg_regions[\"START\"]\n",
    "wg_regions = wg_regions.rename(columns={\"#chrom\" : \"CHROMOSOME\"}).drop(columns=[\"size\"])\n",
    "wg_regions[\"CHROMOSOME\"] = wg_regions[\"CHROMOSOME\"].apply(lambda x: x[3:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2b243e0d-be78-47d6-9bf9-fa992893e68d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "195471972"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Se chr1 size = 195471971, Perche non START 1, END 1 + 195471971 ?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "d71964b2-6755-4d07-bfdc-b2e31772b5a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>195471969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>182113222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>160039678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>156508114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>151834682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>149736544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>145441457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>129401211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>124595108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>130694991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>122082541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>120129020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>120421637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>124902242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>104043683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>98207766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>94987269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>90702637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>61431564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>X</td>\n",
       "      <td>2</td>\n",
       "      <td>171031297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Y</td>\n",
       "      <td>2</td>\n",
       "      <td>91744696</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROMOSOME  START        END\n",
       "0           1      2  195471969\n",
       "1           2      2  182113222\n",
       "2           3      2  160039678\n",
       "3           4      2  156508114\n",
       "4           5      2  151834682\n",
       "5           6      2  149736544\n",
       "6           7      2  145441457\n",
       "7           8      2  129401211\n",
       "8           9      2  124595108\n",
       "9          10      2  130694991\n",
       "10         11      2  122082541\n",
       "11         12      2  120129020\n",
       "12         13      2  120421637\n",
       "13         14      2  124902242\n",
       "14         15      2  104043683\n",
       "15         16      2   98207766\n",
       "16         17      2   94987269\n",
       "17         18      2   90702637\n",
       "18         19      2   61431564\n",
       "19          X      2  171031297\n",
       "20          Y      2   91744696"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wg_regions = sort_regions(wg_regions)\n",
    "wg_regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0ea21db4-004c-4e69-bdc7-aa66326385d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# out_regions = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data/mm10_wg.regions.gz\"\n",
    "# wg_regions.to_csv(out_regions, compression='gzip', index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b499e59d-cc69-4e39-ab03-3ed6580356d8",
   "metadata": {},
   "source": [
    "### Run bbgsignature content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "925d0a2b-46d8-4ef3-b8e6-bf9f4c6f0b25",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/jobs/spellegrini/8879724/ipykernel_25678/106531960.py:2: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  pd.read_csv(\"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data/liver_strelka.in.tsv.gz\", sep=\"\\t\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>do9714_91620_N3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>do9714_91620_N3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>do9786_91932_N7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>3145729</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>do9786_91932_N7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>3278912</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>do9386_90793_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921896</th>\n",
       "      <td>3</td>\n",
       "      <td>4767854</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>do9442_91837_N5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921897</th>\n",
       "      <td>Y</td>\n",
       "      <td>90793815</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>do12459_91948_N4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921898</th>\n",
       "      <td>Y</td>\n",
       "      <td>90793868</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>do9172_90170_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921899</th>\n",
       "      <td>Y</td>\n",
       "      <td>90794273</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>do9201_90802_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22921900</th>\n",
       "      <td>Y</td>\n",
       "      <td>90805575</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>do9197_90797_N2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22921901 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROMOSOME  POSITION REF ALT            SAMPLE\n",
       "0                10   3145729   A   G   do9714_91620_N3\n",
       "1                10   3145729   A   T   do9714_91620_N3\n",
       "2                10   3145729   A   G   do9786_91932_N7\n",
       "3                10   3145729   A   T   do9786_91932_N7\n",
       "4                10   3278912   T   C   do9386_90793_N2\n",
       "...             ...       ...  ..  ..               ...\n",
       "22921896          3   4767854   A   T   do9442_91837_N5\n",
       "22921897          Y  90793815   T   A  do12459_91948_N4\n",
       "22921898          Y  90793868   C   T   do9172_90170_N2\n",
       "22921899          Y  90794273   C   G   do9201_90802_N1\n",
       "22921900          Y  90805575   T   A   do9197_90797_N2\n",
       "\n",
       "[22921901 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.read_csv(\"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data/liver_strelka.in.tsv.gz\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1281bdf2-82d1-46c3-9083-8486df38793d",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda env activate bgsignature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "096a56f1-dd8c-4b0e-b2c3-9a2415a83fe4",
   "metadata": {},
   "outputs": [],
   "source": [
    "bgsignature count -r /workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data/mm10_wg.regions.gz -s 3 -g mm10 --exclude-N --collapse -o /workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/mut_profile/wg.counts.gz --cores 4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42680911-b4ad-40a7-b35d-ba5276384a47",
   "metadata": {},
   "source": [
    "### Run bbgsignature normalizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81c37745-b888-4496-8a57-f959691d2d77",
   "metadata": {},
   "outputs": [],
   "source": [
    "bgsignature normalize -m /workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data/liver_strelka.in.tsv.gz -r /workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/raw_data/mm10_wg.regions.gz --normalize /workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/mut_profile/wg.counts.gz -s 3 -g mm10 --collapse --cores 4 -o /workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/mut_profile/liver_strelka.sig.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "042a2bd9-d4ea-4d0d-9bf8-61b2efc412b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "mut_profile_192_path = \"/workspace/projects/intogen_plus/fixdatasets-20230223/intogen-plus/run_20230224_ALL/intogen_analysis/steps/signature/CBIOP_WGS_PRAD_EURUROL_2017.sig.json\"\n",
    "mut_profile_192 = json.load(open(mut_profile_192_path)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "ad5e7277-4969-4bff-b30c-d8e667042d22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not isinstance(mut_profile_192, dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2559c122-3884-49ac-8d4e-fbc327966ee3",
   "metadata": {},
   "source": [
    "### Parse mut profile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2784d1a9-a481-437e-8853-dea4acadb907",
   "metadata": {},
   "outputs": [],
   "source": [
    "def convert_mut_rate_192_to_96(mut_rate_dict):\n",
    "    \"\"\"\n",
    "    Convert a mut_profile dictionary of 192 items into a list \n",
    "    of 96 elements (keep only non-redundant mutations). \n",
    "    \"\"\"\n",
    "    \n",
    "    lst_rate = []\n",
    "    cb  = dict(zip('ACGT', 'TGCA'))\n",
    "    for ref in ['C', 'T']:\n",
    "        for alt in cb.keys():\n",
    "            if ref == alt:\n",
    "                continue\n",
    "            else:\n",
    "                for p in product(cb.keys(), repeat=2):\n",
    "                    \n",
    "                    mut = f\"{p[0]}{ref}{p[1]}>{alt}\"\n",
    "                    lst_rate.append(mut_rate_dict[mut])\n",
    "        \n",
    "    return lst_rate\n",
    "\n",
    "import json\n",
    "mut_profile_192_path = \"/workspace/projects/intogen_plus/fixdatasets-20230223/intogen-plus/run_20230224_ALL/intogen_analysis/steps/signature/CBIOP_WGS_PRAD_EURUROL_2017.sig.json\"\n",
    "mut_profile_192 = json.load(open(mut_profile_192_path)) \n",
    "\n",
    "mut_profile_96 = convert_mut_rate_dict_192_to_vec_96(mut_profile_192)\n",
    "mut_profile_96_path = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/mut_profile/liver_strelka.mutrate.json\"\n",
    "\n",
    "with open(mut_profile_96_path, 'w') as file:\n",
    "    json.dump(mut_profile_96, file)   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba14c1ef-ae6e-4248-877e-c95639189c03",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "# Debug main"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c1bd5cfb-dac6-4f79-aea7-376f453389c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_maf_input(maf_input_path, keep_samples_id=False):\n",
    "    \"\"\"\n",
    "    Parse in.maf file which is used as \n",
    "    input for the HotMAPS method.\n",
    "    \"\"\"\n",
    "\n",
    "    # Load\n",
    "    maf = pd.read_csv(maf_input_path, sep=\"\\t\", dtype={'Chromosome': str})\n",
    "\n",
    "    # Select only missense mutation and extract Gene name and mut\n",
    "    maf = maf[maf['Variant_Classification'].str.contains('missense_variant')]\n",
    "    maf[\"Pos\"] = maf.loc[:, \"HGVSp_Short\"].apply(lambda x: int(re.sub(\"\\\\D\", \"\", (x[2:]))))\n",
    "    maf[\"WT\"] = maf[\"HGVSp_Short\"].apply(lambda x: re.findall(\"\\\\D\", x[2:])[0])\n",
    "    maf[\"Mut\"] = maf[\"HGVSp_Short\"].apply(lambda x: re.findall(\"\\\\D\", x[2:])[1])\n",
    "    maf = maf[[\"Hugo_Symbol\", \"Pos\", \"WT\", \"Mut\", \"Tumor_Sample_Barcode\"]]\n",
    "    maf = maf.sort_values(\"Pos\").rename(columns={\"Hugo_Symbol\" : \"Gene\"})\n",
    "\n",
    "    if keep_samples_id == False:\n",
    "        maf = maf.drop(columns=[\"Tumor_Sample_Barcode\"])\n",
    "    \n",
    "    return maf.reset_index(drop=True)\n",
    "\n",
    "def parse2_maf_input(maf_input_path, keep_samples_id=False):\n",
    "    \"\"\"\n",
    "    Parse in.maf file which is used as \n",
    "    input for the HotMAPS method.\n",
    "    \"\"\"\n",
    "\n",
    "    # Load\n",
    "    maf = pd.read_csv(maf_input_path, sep=\"\\t\", dtype={'Chromosome': str})\n",
    "\n",
    "    # Select only missense mutation and extract Gene name and mut\n",
    "    maf = maf[maf['Variant_Classification'].str.contains('missense_variant')]\n",
    "    maf['Pos'] = maf['HGVSp_Short'].str.extract(r'(\\d+)').astype(int)\n",
    "    maf['WT'] = maf['HGVSp_Short'].str.extract(r'(\\D)')[0]\n",
    "    maf['Mut'] = maf['HGVSp_Short'].str.extract(r'\\D(\\D)')[0]\n",
    "    maf = maf[[\"Hugo_Symbol\", \"Pos\", \"WT\", \"Mut\", \"Tumor_Sample_Barcode\"]]\n",
    "    maf = maf.sort_values(\"Pos\").rename(columns={\"Hugo_Symbol\" : \"Gene\"})\n",
    "\n",
    "    if keep_samples_id == False:\n",
    "        maf = maf.drop(columns=[\"Tumor_Sample_Barcode\"])\n",
    "    \n",
    "    return maf.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e1c9021d-e2d3-4fe2-b114-af69c5e8b4a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "maf_input_path = \"/workspace/projects/clustering_3d/evaluation/datasets/datasets_mouse/input/maf/liver_strelka_parsed.maf\"\n",
    "seq_df_path = \"/workspace/projects/clustering_3d/clustering_3d/datasets_mouse/seq_for_mut_prob.csv\"\n",
    "plddt_path = \"/workspace/projects/clustering_3d/clustering_3d/datasets_mouse/confidence.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a5514f29-6840-4966-949a-37e528ae3077",
   "metadata": {},
   "outputs": [],
   "source": [
    "maf = pd.read_csv(maf_input_path, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "02f04814-634e-4873-8d53-7b1b5dd37700",
   "metadata": {},
   "outputs": [],
   "source": [
    "maf = asd[asd['Variant_Classification'].str.contains('missense_variant')].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea6d942b-2a02-4ee5-92ed-00db33954677",
   "metadata": {},
   "outputs": [],
   "source": [
    "maf[\"Pos\"] = maf.loc[:, \"HGVSp_Short\"].apply(lambda x: int(re.sub(\"\\\\D\", \"\", (x[2:]))))\n",
    "maf[\"WT\"] = maf[\"HGVSp_Short\"].apply(lambda x: re.findall(\"\\\\D\", x[2:])[0])\n",
    "maf[\"Mut\"] = maf[\"HGVSp_Short\"].apply(lambda x: re.findall(\"\\\\D\", x[2:])[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84e503d7-b906-4fa9-a8de-5088614df9d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "maf = maf[[\"Hugo_Symbol\", \"Pos\", \"WT\", \"Mut\", \"Tumor_Sample_Barcode\"]]\n",
    "maf = maf.sort_values(\"Pos\").rename(columns={\"Hugo_Symbol\" : \"Gene\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "ca981f5f-c345-42e9-a05e-f7ba68a4fcc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load input and df of DNA sequences\n",
    "\n",
    "# MAF input\n",
    "data = parse_maf_input(maf_input_path, keep_samples_id=True)\n",
    "\n",
    "# Seq df for missense mut prob\n",
    "seq_df = pd.read_csv(seq_df_path)\n",
    "\n",
    "# Model confidence\n",
    "plddt_df = pd.read_csv(plddt_path)\n",
    "\n",
    "\n",
    "## Run\n",
    "\n",
    "result_pos_lst = []\n",
    "result_gene_lst = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "806c1744-07a9-471c-ac2f-d3c9431d4053",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 15.5 s, sys: 675 ms, total: 16.2 s\n",
      "Wall time: 16.2 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Pos</th>\n",
       "      <th>WT</th>\n",
       "      <th>Mut</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Odf2l</td>\n",
       "      <td>2</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9699_90721_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Mt4</td>\n",
       "      <td>2</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9853_94314_N6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ighd</td>\n",
       "      <td>2</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9735_92187_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ghsr</td>\n",
       "      <td>2</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9712_91599_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Fam227b</td>\n",
       "      <td>2</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9159_89986_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180223</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33302</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9711_91416_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180224</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33392</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9788_93096_N4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180225</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33400</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9818_93243_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180226</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33412</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9395_91217_N3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180227</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33463</td>\n",
       "      <td>p</td>\n",
       "      <td>.</td>\n",
       "      <td>do9817_93242_N2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>180228 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Gene    Pos WT Mut Tumor_Sample_Barcode\n",
       "0         Odf2l      2  p   .      do9699_90721_N1\n",
       "1           Mt4      2  p   .      do9853_94314_N6\n",
       "2          Ighd      2  p   .      do9735_92187_N2\n",
       "3          Ghsr      2  p   .      do9712_91599_N1\n",
       "4       Fam227b      2  p   .      do9159_89986_N1\n",
       "...         ...    ... ..  ..                  ...\n",
       "180223      Ttn  33302  p   .      do9711_91416_N2\n",
       "180224      Ttn  33392  p   .      do9788_93096_N4\n",
       "180225      Ttn  33400  p   .      do9818_93243_N1\n",
       "180226      Ttn  33412  p   .      do9395_91217_N3\n",
       "180227      Ttn  33463  p   .      do9817_93242_N2\n",
       "\n",
       "[180228 rows x 5 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "# MAF input\n",
    "data = parse2_maf_input(maf_input_path, keep_samples_id=True)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "4b787eec-3664-4b79-8529-7430c739cf36",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 15 s, sys: 632 ms, total: 15.6 s\n",
      "Wall time: 15.6 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Pos</th>\n",
       "      <th>WT</th>\n",
       "      <th>Mut</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Odf2l</td>\n",
       "      <td>2</td>\n",
       "      <td>E</td>\n",
       "      <td>G</td>\n",
       "      <td>do9699_90721_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Mt4</td>\n",
       "      <td>2</td>\n",
       "      <td>D</td>\n",
       "      <td>V</td>\n",
       "      <td>do9853_94314_N6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ighd</td>\n",
       "      <td>2</td>\n",
       "      <td>N</td>\n",
       "      <td>K</td>\n",
       "      <td>do9735_92187_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ghsr</td>\n",
       "      <td>2</td>\n",
       "      <td>W</td>\n",
       "      <td>R</td>\n",
       "      <td>do9712_91599_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Fam227b</td>\n",
       "      <td>2</td>\n",
       "      <td>E</td>\n",
       "      <td>K</td>\n",
       "      <td>do9159_89986_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180223</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33302</td>\n",
       "      <td>K</td>\n",
       "      <td>M</td>\n",
       "      <td>do9711_91416_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180224</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33392</td>\n",
       "      <td>A</td>\n",
       "      <td>P</td>\n",
       "      <td>do9788_93096_N4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180225</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33400</td>\n",
       "      <td>T</td>\n",
       "      <td>S</td>\n",
       "      <td>do9818_93243_N1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180226</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33412</td>\n",
       "      <td>I</td>\n",
       "      <td>V</td>\n",
       "      <td>do9395_91217_N3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180227</th>\n",
       "      <td>Ttn</td>\n",
       "      <td>33463</td>\n",
       "      <td>N</td>\n",
       "      <td>I</td>\n",
       "      <td>do9817_93242_N2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>180228 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Gene    Pos WT Mut Tumor_Sample_Barcode\n",
       "0         Odf2l      2  E   G      do9699_90721_N1\n",
       "1           Mt4      2  D   V      do9853_94314_N6\n",
       "2          Ighd      2  N   K      do9735_92187_N2\n",
       "3          Ghsr      2  W   R      do9712_91599_N1\n",
       "4       Fam227b      2  E   K      do9159_89986_N1\n",
       "...         ...    ... ..  ..                  ...\n",
       "180223      Ttn  33302  K   M      do9711_91416_N2\n",
       "180224      Ttn  33392  A   P      do9788_93096_N4\n",
       "180225      Ttn  33400  T   S      do9818_93243_N1\n",
       "180226      Ttn  33412  I   V      do9395_91217_N3\n",
       "180227      Ttn  33463  N   I      do9817_93242_N2\n",
       "\n",
       "[180228 rows x 5 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "# MAF input\n",
    "data = parse_maf_input(maf_input_path, keep_samples_id=True)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "09fe5dae-868a-4e92-a322-c1fd10f8c705",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Run\n",
    "\n",
    "result_pos_lst = []\n",
    "result_gene_lst = []\n",
    "\n",
    "# Get genes with enough mut\n",
    "genes = data.groupby(\"Gene\").apply(lambda x: len(x))\n",
    "genes_mut = genes[genes >= 2]\n",
    "genes_no_mut = genes[genes < 2].index\n",
    "result_gene = pd.DataFrame({\"Gene\" : genes_no_mut,\n",
    "                            \"Uniprot_ID\" : np.nan,\n",
    "                            \"F\" : np.nan,\n",
    "                            \"Mut_in_gene\" : 1,\n",
    "                            \"Max_mut_pos\" : np.nan,\n",
    "                            \"Structure_max_pos\" : np.nan,\n",
    "                            \"Status\" : \"No_mut\"})\n",
    "result_gene_lst.append(result_gene)   \n",
    "\n",
    "# Get genes with corresponding Uniprot-ID mapping\n",
    "gene_to_uniprot_dict = {gene : uni_id for gene, uni_id in seq_df[[\"Gene\", \"Uniprot_ID\"]].drop_duplicates().values}\n",
    "genes_to_process = [gene for gene in genes_mut.index if gene in gene_to_uniprot_dict.keys()]\n",
    "seq_df = seq_df[[gene in genes_to_process for gene in seq_df[\"Gene\"]]].reset_index(drop=True)\n",
    "genes_no_mapping = genes[[gene in genes_mut.index and gene not in gene_to_uniprot_dict.keys() for gene in genes.index]]\n",
    "result_gene = pd.DataFrame({\"Gene\" : genes_no_mapping.index,\n",
    "                            \"Uniprot_ID\" : np.nan,\n",
    "                            \"F\" : np.nan,\n",
    "                            \"Mut_in_gene\" : genes_no_mapping.values,\n",
    "                            \"Max_mut_pos\" : np.nan,\n",
    "                            \"Structure_max_pos\" : np.nan,\n",
    "                            \"Status\" : \"No_ID_mapping\"})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "642ba311-5c65-4d73-a285-da542d351a5a",
   "metadata": {},
   "source": [
    "# Mouse analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "664cb4ed-c0cb-4d89-ab6f-099e6a2693b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Pos</th>\n",
       "      <th>WT</th>\n",
       "      <th>Mut</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>O3D_transcript_ID</th>\n",
       "      <th>Transcript_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0610009B22Rik</td>\n",
       "      <td>53</td>\n",
       "      <td>M</td>\n",
       "      <td>K</td>\n",
       "      <td>do9706_90887_N2</td>\n",
       "      <td>ENSMUST00000109098</td>\n",
       "      <td>ENSMUST00000007921</td>\n",
       "      <td>Mismatch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0610010F05Rik</td>\n",
       "      <td>66</td>\n",
       "      <td>V</td>\n",
       "      <td>A</td>\n",
       "      <td>do9411_91598_N1</td>\n",
       "      <td>ENSMUST00000043356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0610010F05Rik</td>\n",
       "      <td>179</td>\n",
       "      <td>Y</td>\n",
       "      <td>H</td>\n",
       "      <td>do9736_92187_N3</td>\n",
       "      <td>ENSMUST00000043356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0610010F05Rik</td>\n",
       "      <td>252</td>\n",
       "      <td>H</td>\n",
       "      <td>Y</td>\n",
       "      <td>do9852_94314_N5</td>\n",
       "      <td>ENSMUST00000043356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0610010F05Rik</td>\n",
       "      <td>362</td>\n",
       "      <td>I</td>\n",
       "      <td>R</td>\n",
       "      <td>do9245_91113_N2</td>\n",
       "      <td>ENSMUST00000043356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180223</th>\n",
       "      <td>ccdc198</td>\n",
       "      <td>21</td>\n",
       "      <td>Q</td>\n",
       "      <td>L</td>\n",
       "      <td>do9450_91839_N2</td>\n",
       "      <td>ENSMUST00000228936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180224</th>\n",
       "      <td>ccdc198</td>\n",
       "      <td>210</td>\n",
       "      <td>E</td>\n",
       "      <td>K</td>\n",
       "      <td>do9878_91943_N1</td>\n",
       "      <td>ENSMUST00000228936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180225</th>\n",
       "      <td>ccdc198</td>\n",
       "      <td>250</td>\n",
       "      <td>A</td>\n",
       "      <td>V</td>\n",
       "      <td>do9826_93244_N8</td>\n",
       "      <td>ENSMUST00000228936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180226</th>\n",
       "      <td>ccdc198</td>\n",
       "      <td>264</td>\n",
       "      <td>D</td>\n",
       "      <td>E</td>\n",
       "      <td>do9435_91604_C1</td>\n",
       "      <td>ENSMUST00000228936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180227</th>\n",
       "      <td>ccdc198</td>\n",
       "      <td>284</td>\n",
       "      <td>I</td>\n",
       "      <td>F</td>\n",
       "      <td>do9771_93096_N1</td>\n",
       "      <td>ENSMUST00000228936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>180228 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene  Pos WT Mut Tumor_Sample_Barcode       Transcript_ID  \\\n",
       "0       0610009B22Rik   53  M   K      do9706_90887_N2  ENSMUST00000109098   \n",
       "1       0610010F05Rik   66  V   A      do9411_91598_N1  ENSMUST00000043356   \n",
       "2       0610010F05Rik  179  Y   H      do9736_92187_N3  ENSMUST00000043356   \n",
       "3       0610010F05Rik  252  H   Y      do9852_94314_N5  ENSMUST00000043356   \n",
       "4       0610010F05Rik  362  I   R      do9245_91113_N2  ENSMUST00000043356   \n",
       "...               ...  ... ..  ..                  ...                 ...   \n",
       "180223        ccdc198   21  Q   L      do9450_91839_N2  ENSMUST00000228936   \n",
       "180224        ccdc198  210  E   K      do9878_91943_N1  ENSMUST00000228936   \n",
       "180225        ccdc198  250  A   V      do9826_93244_N8  ENSMUST00000228936   \n",
       "180226        ccdc198  264  D   E      do9435_91604_C1  ENSMUST00000228936   \n",
       "180227        ccdc198  284  I   F      do9771_93096_N1  ENSMUST00000228936   \n",
       "\n",
       "         O3D_transcript_ID Transcript_status  \n",
       "0       ENSMUST00000007921          Mismatch  \n",
       "1                      NaN       O3D_missing  \n",
       "2                      NaN       O3D_missing  \n",
       "3                      NaN       O3D_missing  \n",
       "4                      NaN       O3D_missing  \n",
       "...                    ...               ...  \n",
       "180223                 NaN       O3D_missing  \n",
       "180224                 NaN       O3D_missing  \n",
       "180225                 NaN       O3D_missing  \n",
       "180226                 NaN       O3D_missing  \n",
       "180227                 NaN       O3D_missing  \n",
       "\n",
       "[180228 rows x 8 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut = pd.read_table(\"/workspace/projects/clustering_3d/o3d_analysys/datasets/output/cancer_mouse/run_2024-11-04_13-40-54/liver_strelka/liver_strelka.mutations.processed.tsv\")\n",
    "mut[mut[\"Gene\"] == \"Braf\"][:60]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "bfb2080e-3e14-45dc-86f5-a4a5d5d3a3cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Braf</td>\n",
       "      <td>do9457_91930_N1</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Braf</td>\n",
       "      <td>do9695_90446_N1</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Braf</td>\n",
       "      <td>do9844_93260_N3</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Braf</td>\n",
       "      <td>do9408_91539_N3</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Braf</td>\n",
       "      <td>do9710_91945_N1</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1158</th>\n",
       "      <td>Kras</td>\n",
       "      <td>do12479_93100_N6</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1159</th>\n",
       "      <td>Kras</td>\n",
       "      <td>do12474_92188_N3</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1160</th>\n",
       "      <td>Kras</td>\n",
       "      <td>do9838_93258_N12</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1161</th>\n",
       "      <td>Kras</td>\n",
       "      <td>do9823_93244_N4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1162</th>\n",
       "      <td>Kras</td>\n",
       "      <td>do9875_91932_N4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1163 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo_Symbol Tumor_Sample_Barcode Variant_Classification  \\\n",
       "0           Braf      do9457_91930_N1    3_prime_UTR_variant   \n",
       "1           Braf      do9695_90446_N1    3_prime_UTR_variant   \n",
       "2           Braf      do9844_93260_N3    3_prime_UTR_variant   \n",
       "3           Braf      do9408_91539_N3    3_prime_UTR_variant   \n",
       "4           Braf      do9710_91945_N1    3_prime_UTR_variant   \n",
       "...          ...                  ...                    ...   \n",
       "1158        Kras     do12479_93100_N6  upstream_gene_variant   \n",
       "1159        Kras     do12474_92188_N3  upstream_gene_variant   \n",
       "1160        Kras     do9838_93258_N12  upstream_gene_variant   \n",
       "1161        Kras      do9823_93244_N4  upstream_gene_variant   \n",
       "1162        Kras      do9875_91932_N4  upstream_gene_variant   \n",
       "\n",
       "           Transcript_ID HGVSp_Short  \n",
       "0     ENSMUST00000002487         NaN  \n",
       "1     ENSMUST00000002487         NaN  \n",
       "2     ENSMUST00000002487         NaN  \n",
       "3     ENSMUST00000002487         NaN  \n",
       "4     ENSMUST00000002487         NaN  \n",
       "...                  ...         ...  \n",
       "1158  ENSMUST00000032399         NaN  \n",
       "1159  ENSMUST00000032399         NaN  \n",
       "1160  ENSMUST00000032399         NaN  \n",
       "1161  ENSMUST00000032399         NaN  \n",
       "1162  ENSMUST00000032399         NaN  \n",
       "\n",
       "[1163 rows x 5 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf = pd.read_table(\"/workspace/projects/clustering_3d/o3d_analysys/datasets/input/cancer_mouse/maf/liver_strelka.in.maf\")\n",
    "maf = maf[maf.Hugo_Symbol.isin([\"Braf\", \"Kras\"])].reset_index(drop=True)\n",
    "maf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "1c66d7b6-c1fe-43a9-b991-b2435b0ab5be",
   "metadata": {},
   "outputs": [],
   "source": [
    "maf.to_csv(\"/workspace/projects/clustering_3d/o3d_analysys/datasets/input/cancer_mouse/maf/liver_strelka.toy.in.maf\", index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "7ba17569-68bd-4aef-ab55-15e608f243a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Pos</th>\n",
       "      <th>WT</th>\n",
       "      <th>Mut</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>O3D_transcript_ID</th>\n",
       "      <th>Transcript_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Braf</td>\n",
       "      <td>45</td>\n",
       "      <td>E</td>\n",
       "      <td>G</td>\n",
       "      <td>do9458_91930_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Braf</td>\n",
       "      <td>77</td>\n",
       "      <td>H</td>\n",
       "      <td>N</td>\n",
       "      <td>do9170_90022_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Braf</td>\n",
       "      <td>127</td>\n",
       "      <td>D</td>\n",
       "      <td>E</td>\n",
       "      <td>do9789_93096_N5</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Braf</td>\n",
       "      <td>320</td>\n",
       "      <td>S</td>\n",
       "      <td>L</td>\n",
       "      <td>do9832_93258_N6</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Braf</td>\n",
       "      <td>408</td>\n",
       "      <td>L</td>\n",
       "      <td>P</td>\n",
       "      <td>do9433_91620_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>Kras</td>\n",
       "      <td>61</td>\n",
       "      <td>Q</td>\n",
       "      <td>R</td>\n",
       "      <td>do9428_91618_N1</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>ENSMUST00000111710</td>\n",
       "      <td>Mismatch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>Kras</td>\n",
       "      <td>61</td>\n",
       "      <td>Q</td>\n",
       "      <td>L</td>\n",
       "      <td>do9428_91618_N1</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>ENSMUST00000111710</td>\n",
       "      <td>Mismatch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>Kras</td>\n",
       "      <td>61</td>\n",
       "      <td>Q</td>\n",
       "      <td>R</td>\n",
       "      <td>do9799_93129_N2</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>ENSMUST00000111710</td>\n",
       "      <td>Mismatch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>Kras</td>\n",
       "      <td>61</td>\n",
       "      <td>Q</td>\n",
       "      <td>L</td>\n",
       "      <td>do9799_93129_N2</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>ENSMUST00000111710</td>\n",
       "      <td>Mismatch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>Kras</td>\n",
       "      <td>76</td>\n",
       "      <td>E</td>\n",
       "      <td>K</td>\n",
       "      <td>do12459_91948_N4</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>ENSMUST00000111710</td>\n",
       "      <td>Mismatch</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>184 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene  Pos WT Mut Tumor_Sample_Barcode       Transcript_ID  \\\n",
       "0    Braf   45  E   G      do9458_91930_N2  ENSMUST00000002487   \n",
       "1    Braf   77  H   N      do9170_90022_N1  ENSMUST00000002487   \n",
       "2    Braf  127  D   E      do9789_93096_N5  ENSMUST00000002487   \n",
       "3    Braf  320  S   L      do9832_93258_N6  ENSMUST00000002487   \n",
       "4    Braf  408  L   P      do9433_91620_N2  ENSMUST00000002487   \n",
       "..    ...  ... ..  ..                  ...                 ...   \n",
       "179  Kras   61  Q   R      do9428_91618_N1  ENSMUST00000032399   \n",
       "180  Kras   61  Q   L      do9428_91618_N1  ENSMUST00000032399   \n",
       "181  Kras   61  Q   R      do9799_93129_N2  ENSMUST00000032399   \n",
       "182  Kras   61  Q   L      do9799_93129_N2  ENSMUST00000032399   \n",
       "183  Kras   76  E   K     do12459_91948_N4  ENSMUST00000032399   \n",
       "\n",
       "      O3D_transcript_ID Transcript_status  \n",
       "0                   NaN       O3D_missing  \n",
       "1                   NaN       O3D_missing  \n",
       "2                   NaN       O3D_missing  \n",
       "3                   NaN       O3D_missing  \n",
       "4                   NaN       O3D_missing  \n",
       "..                  ...               ...  \n",
       "179  ENSMUST00000111710          Mismatch  \n",
       "180  ENSMUST00000111710          Mismatch  \n",
       "181  ENSMUST00000111710          Mismatch  \n",
       "182  ENSMUST00000111710          Mismatch  \n",
       "183  ENSMUST00000111710          Mismatch  \n",
       "\n",
       "[184 rows x 8 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = \"/workspace/projects/clustering_3d/o3d_analysys/datasets/output/cancer_mouse/run_2024-11-05_20-26-44/liver_strelka.toy\"\n",
    "mut_path = f\"{path}/liver_strelka.toy.mutations.processed.tsv\"\n",
    "mut_toy = pd.read_table(mut_path)\n",
    "mut_toy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "3f7e258b-6400-443e-b6cc-68bce942bab6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>F</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Mut_in_gene</th>\n",
       "      <th>Mut_in_res</th>\n",
       "      <th>Mut_in_vol</th>\n",
       "      <th>Score</th>\n",
       "      <th>Score_obs_sim</th>\n",
       "      <th>pval</th>\n",
       "      <th>...</th>\n",
       "      <th>Rank</th>\n",
       "      <th>Mut_in_cl_vol</th>\n",
       "      <th>Res_in_cl</th>\n",
       "      <th>PAE_vol</th>\n",
       "      <th>pLDDT_res</th>\n",
       "      <th>pLDDT_vol</th>\n",
       "      <th>pLDDT_cl_vol</th>\n",
       "      <th>WT_mismatch</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Cohort</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>637</td>\n",
       "      <td>162</td>\n",
       "      <td>155</td>\n",
       "      <td>156</td>\n",
       "      <td>0.924798</td>\n",
       "      <td>83.446518</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.006410</td>\n",
       "      <td>95.48</td>\n",
       "      <td>95.499860</td>\n",
       "      <td>95.499860</td>\n",
       "      <td>155.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>631</td>\n",
       "      <td>162</td>\n",
       "      <td>1</td>\n",
       "      <td>156</td>\n",
       "      <td>0.919119</td>\n",
       "      <td>93.109734</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>156.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.993590</td>\n",
       "      <td>98.58</td>\n",
       "      <td>95.499860</td>\n",
       "      <td>95.499860</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>408</td>\n",
       "      <td>162</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000782</td>\n",
       "      <td>0.085379</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>32.52</td>\n",
       "      <td>32.520000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>320</td>\n",
       "      <td>162</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000672</td>\n",
       "      <td>0.077943</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>27.59</td>\n",
       "      <td>27.590000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>45</td>\n",
       "      <td>162</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000386</td>\n",
       "      <td>0.047099</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>33.42</td>\n",
       "      <td>33.420000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>127</td>\n",
       "      <td>162</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000317</td>\n",
       "      <td>0.040383</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>35.79</td>\n",
       "      <td>35.790000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>77</td>\n",
       "      <td>162</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>0.018465</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>89.04</td>\n",
       "      <td>89.040000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Braf</td>\n",
       "      <td>P28028</td>\n",
       "      <td>1</td>\n",
       "      <td>469</td>\n",
       "      <td>162</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000745</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>88.95</td>\n",
       "      <td>88.950000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Kras</td>\n",
       "      <td>P32883</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>0.796174</td>\n",
       "      <td>6.901697</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.350000</td>\n",
       "      <td>94.74</td>\n",
       "      <td>91.009995</td>\n",
       "      <td>91.009995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Kras</td>\n",
       "      <td>P32883</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>0.794881</td>\n",
       "      <td>8.045598</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>20.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.350000</td>\n",
       "      <td>94.73</td>\n",
       "      <td>91.009995</td>\n",
       "      <td>91.009995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Kras</td>\n",
       "      <td>P32883</td>\n",
       "      <td>1</td>\n",
       "      <td>61</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>0.631085</td>\n",
       "      <td>7.039743</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>20.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.117647</td>\n",
       "      <td>83.37</td>\n",
       "      <td>90.058235</td>\n",
       "      <td>91.009995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Kras</td>\n",
       "      <td>P32883</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>22</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>0.310494</td>\n",
       "      <td>3.733244</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>20.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.846154</td>\n",
       "      <td>97.24</td>\n",
       "      <td>95.123850</td>\n",
       "      <td>91.009995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Kras</td>\n",
       "      <td>P32883</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001837</td>\n",
       "      <td>0.023565</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>95.77</td>\n",
       "      <td>95.770000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Kras</td>\n",
       "      <td>P32883</td>\n",
       "      <td>1</td>\n",
       "      <td>76</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001736</td>\n",
       "      <td>0.023607</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>97.92</td>\n",
       "      <td>97.920000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Uniprot_ID  F  Pos  Mut_in_gene  Mut_in_res  Mut_in_vol     Score  \\\n",
       "0   Braf     P28028  1  637          162         155         156  0.924798   \n",
       "1   Braf     P28028  1  631          162           1         156  0.919119   \n",
       "2   Braf     P28028  1  408          162           1           1  0.000782   \n",
       "3   Braf     P28028  1  320          162           1           1  0.000672   \n",
       "4   Braf     P28028  1   45          162           1           1  0.000386   \n",
       "5   Braf     P28028  1  127          162           1           1  0.000317   \n",
       "6   Braf     P28028  1   77          162           1           1  0.000139   \n",
       "7   Braf     P28028  1  469          162           1           1  0.000005   \n",
       "8   Kras     P32883  1   12           22          10          20  0.796174   \n",
       "9   Kras     P32883  1   13           22           1          20  0.794881   \n",
       "10  Kras     P32883  1   61           22           7          17  0.631085   \n",
       "11  Kras     P32883  1   14           22           2          13  0.310494   \n",
       "12  Kras     P32883  1   24           22           1           1  0.001837   \n",
       "13  Kras     P32883  1   76           22           1           1  0.001736   \n",
       "\n",
       "    Score_obs_sim  pval  ...  Rank  Mut_in_cl_vol  Res_in_cl   PAE_vol  \\\n",
       "0       83.446518   0.0  ...     0          156.0        2.0  0.006410   \n",
       "1       93.109734   0.0  ...     1          156.0        2.0  0.993590   \n",
       "2        0.085379   1.0  ...     2            NaN        NaN  0.000000   \n",
       "3        0.077943   1.0  ...     3            NaN        NaN  0.000000   \n",
       "4        0.047099   1.0  ...     4            NaN        NaN  0.000000   \n",
       "5        0.040383   1.0  ...     5            NaN        NaN  0.000000   \n",
       "6        0.018465   1.0  ...     6            NaN        NaN  0.000000   \n",
       "7        0.000745   1.0  ...     7            NaN        NaN  0.000000   \n",
       "8        6.901697   0.0  ...     0           20.0        4.0  2.350000   \n",
       "9        8.045598   0.0  ...     1           20.0        4.0  2.350000   \n",
       "10       7.039743   0.0  ...     2           20.0        4.0  4.117647   \n",
       "11       3.733244   0.0  ...     3           20.0        4.0  0.846154   \n",
       "12       0.023565   1.0  ...     4            NaN        NaN  0.000000   \n",
       "13       0.023607   1.0  ...     5            NaN        NaN  0.000000   \n",
       "\n",
       "    pLDDT_res  pLDDT_vol  pLDDT_cl_vol  WT_mismatch  Cancer             Cohort  \n",
       "0       95.48  95.499860     95.499860        155.0     NaN  liver_strelka.toy  \n",
       "1       98.58  95.499860     95.499860          1.0     NaN  liver_strelka.toy  \n",
       "2       32.52  32.520000           NaN          1.0     NaN  liver_strelka.toy  \n",
       "3       27.59  27.590000           NaN          0.0     NaN  liver_strelka.toy  \n",
       "4       33.42  33.420000           NaN          0.0     NaN  liver_strelka.toy  \n",
       "5       35.79  35.790000           NaN          0.0     NaN  liver_strelka.toy  \n",
       "6       89.04  89.040000           NaN          0.0     NaN  liver_strelka.toy  \n",
       "7       88.95  88.950000           NaN          1.0     NaN  liver_strelka.toy  \n",
       "8       94.74  91.009995     91.009995          NaN     NaN  liver_strelka.toy  \n",
       "9       94.73  91.009995     91.009995          NaN     NaN  liver_strelka.toy  \n",
       "10      83.37  90.058235     91.009995          NaN     NaN  liver_strelka.toy  \n",
       "11      97.24  95.123850     91.009995          NaN     NaN  liver_strelka.toy  \n",
       "12      95.77  95.770000           NaN          NaN     NaN  liver_strelka.toy  \n",
       "13      97.92  97.920000           NaN          NaN     NaN  liver_strelka.toy  \n",
       "\n",
       "[14 rows x 23 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_pos_path = f\"{path}/liver_strelka.toy.3d_clustering_pos.csv\"\n",
    "result_pos = pd.read_csv(result_pos_path)\n",
    "result_pos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "d3cddcd7-d2c9-4414-b4a4-b4319f7fdc5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Ens_Gene_ID</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>C_gene</th>\n",
       "      <th>C_pos</th>\n",
       "      <th>C_label</th>\n",
       "      <th>Score_obs_sim_top_vol</th>\n",
       "      <th>Clust_res</th>\n",
       "      <th>...</th>\n",
       "      <th>Ratio_not_in_structure</th>\n",
       "      <th>Ratio_WT_mismatch</th>\n",
       "      <th>Mut_zero_mut_prob</th>\n",
       "      <th>Pos_zero_mut_prob</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Cohort</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>O3D_transcript_ID</th>\n",
       "      <th>Transcript_status</th>\n",
       "      <th>Status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Braf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P28028</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>[637 631]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>93.109734</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.975309</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O3D_missing</td>\n",
       "      <td>Processed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Kras</td>\n",
       "      <td>ENSMUSG00000030265</td>\n",
       "      <td>P32883</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>[12 13 61 14]</td>\n",
       "      <td>[0. 0. 0. 0.]</td>\n",
       "      <td>8.045598</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver_strelka.toy</td>\n",
       "      <td>ENSMUST00000032399</td>\n",
       "      <td>ENSMUST00000111710</td>\n",
       "      <td>Mismatch</td>\n",
       "      <td>Processed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene         Ens_Gene_ID Uniprot_ID  pval  qval  C_gene          C_pos  \\\n",
       "0  Braf                 NaN     P28028   0.0   0.0       1      [637 631]   \n",
       "1  Kras  ENSMUSG00000030265     P32883   0.0   0.0       1  [12 13 61 14]   \n",
       "\n",
       "         C_label  Score_obs_sim_top_vol  Clust_res  ...  \\\n",
       "0        [0. 0.]              93.109734          2  ...   \n",
       "1  [0. 0. 0. 0.]               8.045598          4  ...   \n",
       "\n",
       "   Ratio_not_in_structure  Ratio_WT_mismatch  Mut_zero_mut_prob  \\\n",
       "0                       0           0.975309                  0   \n",
       "1                       0           0.000000                  0   \n",
       "\n",
       "   Pos_zero_mut_prob  Cancer             Cohort       Transcript_ID  \\\n",
       "0                NaN     NaN  liver_strelka.toy  ENSMUST00000002487   \n",
       "1                NaN     NaN  liver_strelka.toy  ENSMUST00000032399   \n",
       "\n",
       "    O3D_transcript_ID  Transcript_status     Status  \n",
       "0                 NaN        O3D_missing  Processed  \n",
       "1  ENSMUST00000111710           Mismatch  Processed  \n",
       "\n",
       "[2 rows x 27 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_gene_path = f\"{path}/liver_strelka.toy.3d_clustering_genes.csv\"\n",
    "result_gene = pd.read_csv(result_gene_path)\n",
    "result_gene"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6407d58e-7f72-4ef6-a99f-7b9135d8127e",
   "metadata": {},
   "source": [
    "### End cazzeggio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d8178ad3-5309-4132-9f94-1872ea5a2833",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "59f9ac29-4044-4ce9-8448-09cb685a3428",
   "metadata": {},
   "outputs": [],
   "source": [
    "mouse_pos_result = pd.read_csv(\"/workspace/projects/clustering_3d/o3d_analysys/datasets/output/cancer_mouse/run_2024-11-04_13-40-54/liver_strelka/liver_strelka.3d_clustering_pos.csv\")\n",
    "mouse_gene_result = pd.read_csv(\"/workspace/projects/clustering_3d/o3d_analysys/datasets/output/cancer_mouse/run_2024-11-04_13-40-54/liver_strelka/liver_strelka.3d_clustering_genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0d28d238-0fd2-4485-ae98-3eef18ee3f36",
   "metadata": {},
   "outputs": [],
   "source": [
    "mouse_pos_result = pd.read_csv(\"/workspace/projects/clustering_3d/evaluation/tool_output/run_20230608_mouse/Liver_strelka_2017.3d_clustering_pos.csv\")\n",
    "mouse_gene_result = pd.read_csv(\"/workspace/projects/clustering_3d/evaluation/tool_output/run_20230608_mouse/Liver_strelka_2017.3d_clustering_genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e5e68316-1cd7-4049-8d13-20e106a5dffd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>qval</th>\n",
       "      <th>C_gene</th>\n",
       "      <th>C_pos</th>\n",
       "      <th>C_community</th>\n",
       "      <th>Ratio_obs_sim_top_vol</th>\n",
       "      <th>Mut_in_gene</th>\n",
       "      <th>Clust_mut</th>\n",
       "      <th>Mut_in_top_vol</th>\n",
       "      <th>Mut_in_top_comm_vol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Hras</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[61 12 13]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>inf</td>\n",
       "      <td>350</td>\n",
       "      <td>341.0</td>\n",
       "      <td>341.0</td>\n",
       "      <td>341.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Braf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[637 631]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>90.675025</td>\n",
       "      <td>162</td>\n",
       "      <td>156.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>156.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H2-M5</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[347]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>65.432984</td>\n",
       "      <td>157</td>\n",
       "      <td>151.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>151.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Nrcam</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[714 676 716 688]</td>\n",
       "      <td>[0. 0. 0. 0.]</td>\n",
       "      <td>56.050080</td>\n",
       "      <td>125</td>\n",
       "      <td>99.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>99.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ncl</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[107]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>34.644913</td>\n",
       "      <td>66</td>\n",
       "      <td>61.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Itga9</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[179]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>27.512761</td>\n",
       "      <td>109</td>\n",
       "      <td>70.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Nkx2-5</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[273]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>21.778911</td>\n",
       "      <td>50</td>\n",
       "      <td>42.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>42.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Olfr1242</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[ 36 279  73]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>17.554267</td>\n",
       "      <td>40</td>\n",
       "      <td>38.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Egfr</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[254]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>16.310410</td>\n",
       "      <td>47</td>\n",
       "      <td>34.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Gm5134</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[635]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>16.281568</td>\n",
       "      <td>70</td>\n",
       "      <td>44.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Oaf</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[133]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>14.697380</td>\n",
       "      <td>37</td>\n",
       "      <td>35.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Rdh10</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[136 133]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>12.687365</td>\n",
       "      <td>32</td>\n",
       "      <td>28.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9530002B09Rik</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[5]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>12.566828</td>\n",
       "      <td>35</td>\n",
       "      <td>31.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Chd5</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[1888]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>11.923891</td>\n",
       "      <td>72</td>\n",
       "      <td>28.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Fam126b</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[490]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>11.314176</td>\n",
       "      <td>27</td>\n",
       "      <td>22.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Zfp184</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[168]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>9.380280</td>\n",
       "      <td>22</td>\n",
       "      <td>17.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Pbp2</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[123 122  17]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>7.941862</td>\n",
       "      <td>26</td>\n",
       "      <td>23.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Kras</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[12 13 61 14]</td>\n",
       "      <td>[0. 0. 0. 0.]</td>\n",
       "      <td>7.673587</td>\n",
       "      <td>22</td>\n",
       "      <td>20.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Olfr945</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[92]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>7.395317</td>\n",
       "      <td>34</td>\n",
       "      <td>23.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Ccdc9</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[224]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>6.702145</td>\n",
       "      <td>11</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Gm20721</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[22]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>6.456593</td>\n",
       "      <td>19</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Rap1gap</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[405]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>6.312924</td>\n",
       "      <td>21</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Mylk</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[1641]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>6.294105</td>\n",
       "      <td>32</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Unk</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[591]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>6.164195</td>\n",
       "      <td>22</td>\n",
       "      <td>12.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Tdrd6</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[298]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>6.009603</td>\n",
       "      <td>72</td>\n",
       "      <td>16.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Mab21l4</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[130]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>5.838393</td>\n",
       "      <td>26</td>\n",
       "      <td>16.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Zmynd10</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[192]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>5.249385</td>\n",
       "      <td>17</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Plcl1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[663]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>4.347430</td>\n",
       "      <td>39</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Vmn2r74</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[51 94]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>4.032435</td>\n",
       "      <td>39</td>\n",
       "      <td>12.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>St8sia3</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[113]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>3.630678</td>\n",
       "      <td>22</td>\n",
       "      <td>12.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Soat2</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[84]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>3.527611</td>\n",
       "      <td>22</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Ebf3</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[556]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>3.492045</td>\n",
       "      <td>30</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Saxo1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[232 230]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>3.418260</td>\n",
       "      <td>23</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Evpl</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[457]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>3.397278</td>\n",
       "      <td>46</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Tnc</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[62]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>3.210614</td>\n",
       "      <td>61</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Txk</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[106 103 109]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>3.132542</td>\n",
       "      <td>26</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Tepsin</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[362 321 361]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>3.097745</td>\n",
       "      <td>9</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Angel2</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[97]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>3.076529</td>\n",
       "      <td>8</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Ankrd63</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[ 86 114  88]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>3.064409</td>\n",
       "      <td>7</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Bpifb5</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[18]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>2.982920</td>\n",
       "      <td>23</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Rps25</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[118]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>2.905756</td>\n",
       "      <td>8</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Nol6</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[207]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>2.746241</td>\n",
       "      <td>41</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Lrfn5</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[512 429]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>2.713499</td>\n",
       "      <td>19</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Ifi47</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[45 44]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>2.695163</td>\n",
       "      <td>6</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Npas1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[351 383]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>2.666975</td>\n",
       "      <td>12</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Pdxdc1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[162 163 324]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>2.665633</td>\n",
       "      <td>11</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Ccdc178</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[739 736]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>2.661947</td>\n",
       "      <td>29</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Krt77</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[251 257 249]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>2.655064</td>\n",
       "      <td>16</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Dot1l</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[994]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>2.637831</td>\n",
       "      <td>17</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Sh2d3c</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[47]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>2.624367</td>\n",
       "      <td>16</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Vmn2r22</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[653 730 668 712 714 707 708 758 718 674 663]</td>\n",
       "      <td>[0. 0. 1. 1. 1. 1. 1. 1. 1. 1. 1.]</td>\n",
       "      <td>2.624205</td>\n",
       "      <td>34</td>\n",
       "      <td>22.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Ccdc54</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[41 36 44]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>2.618848</td>\n",
       "      <td>13</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Adamtsl3</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[1380 1335]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>2.523232</td>\n",
       "      <td>54</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Pum1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[ 996 1013  992]</td>\n",
       "      <td>[0. 0. 0.]</td>\n",
       "      <td>2.488469</td>\n",
       "      <td>18</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Ccdc33</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[297]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>2.457333</td>\n",
       "      <td>26</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Scn1a</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[975 973 970 947 903]</td>\n",
       "      <td>[0. 0. 0. 0. 0.]</td>\n",
       "      <td>2.386572</td>\n",
       "      <td>47</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Nbas</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[1855 1895 1899 1854]</td>\n",
       "      <td>[0. 0. 0. 0.]</td>\n",
       "      <td>2.367992</td>\n",
       "      <td>38</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Zswim4</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>[216 221]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>2.202616</td>\n",
       "      <td>14</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>1700012B07Rik</td>\n",
       "      <td>0.02787</td>\n",
       "      <td>0</td>\n",
       "      <td>[112]</td>\n",
       "      <td>[0.]</td>\n",
       "      <td>3.158133</td>\n",
       "      <td>16</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>Kif5b</td>\n",
       "      <td>0.02787</td>\n",
       "      <td>0</td>\n",
       "      <td>[156 165]</td>\n",
       "      <td>[0. 0.]</td>\n",
       "      <td>2.717943</td>\n",
       "      <td>9</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene     qval  C_gene  \\\n",
       "0            Hras  0.00000       1   \n",
       "1            Braf  0.00000       1   \n",
       "2           H2-M5  0.00000       1   \n",
       "3           Nrcam  0.00000       1   \n",
       "4             Ncl  0.00000       1   \n",
       "5           Itga9  0.00000       1   \n",
       "6          Nkx2-5  0.00000       1   \n",
       "7        Olfr1242  0.00000       1   \n",
       "8            Egfr  0.00000       1   \n",
       "9          Gm5134  0.00000       1   \n",
       "10            Oaf  0.00000       1   \n",
       "11          Rdh10  0.00000       1   \n",
       "12  9530002B09Rik  0.00000       1   \n",
       "13           Chd5  0.00000       1   \n",
       "14        Fam126b  0.00000       1   \n",
       "15         Zfp184  0.00000       1   \n",
       "16           Pbp2  0.00000       1   \n",
       "17           Kras  0.00000       1   \n",
       "18        Olfr945  0.00000       1   \n",
       "19          Ccdc9  0.00000       1   \n",
       "20        Gm20721  0.00000       1   \n",
       "21        Rap1gap  0.00000       1   \n",
       "22           Mylk  0.00000       1   \n",
       "23            Unk  0.00000       1   \n",
       "24          Tdrd6  0.00000       1   \n",
       "25        Mab21l4  0.00000       1   \n",
       "26        Zmynd10  0.00000       1   \n",
       "27          Plcl1  0.00000       1   \n",
       "28        Vmn2r74  0.00000       1   \n",
       "29        St8sia3  0.00000       1   \n",
       "30          Soat2  0.00000       1   \n",
       "31           Ebf3  0.00000       1   \n",
       "32          Saxo1  0.00000       1   \n",
       "33           Evpl  0.00000       1   \n",
       "34            Tnc  0.00000       1   \n",
       "35            Txk  0.00000       1   \n",
       "36         Tepsin  0.00000       1   \n",
       "37         Angel2  0.00000       1   \n",
       "38        Ankrd63  0.00000       1   \n",
       "39         Bpifb5  0.00000       1   \n",
       "40          Rps25  0.00000       1   \n",
       "41           Nol6  0.00000       1   \n",
       "42          Lrfn5  0.00000       1   \n",
       "43          Ifi47  0.00000       1   \n",
       "44          Npas1  0.00000       1   \n",
       "45         Pdxdc1  0.00000       1   \n",
       "46        Ccdc178  0.00000       1   \n",
       "47          Krt77  0.00000       1   \n",
       "48          Dot1l  0.00000       1   \n",
       "49         Sh2d3c  0.00000       1   \n",
       "50        Vmn2r22  0.00000       1   \n",
       "51         Ccdc54  0.00000       1   \n",
       "52       Adamtsl3  0.00000       1   \n",
       "53           Pum1  0.00000       1   \n",
       "54         Ccdc33  0.00000       1   \n",
       "55          Scn1a  0.00000       1   \n",
       "56           Nbas  0.00000       1   \n",
       "57         Zswim4  0.00000       1   \n",
       "58  1700012B07Rik  0.02787       0   \n",
       "59          Kif5b  0.02787       0   \n",
       "\n",
       "                                            C_pos  \\\n",
       "0                                      [61 12 13]   \n",
       "1                                       [637 631]   \n",
       "2                                           [347]   \n",
       "3                               [714 676 716 688]   \n",
       "4                                           [107]   \n",
       "5                                           [179]   \n",
       "6                                           [273]   \n",
       "7                                   [ 36 279  73]   \n",
       "8                                           [254]   \n",
       "9                                           [635]   \n",
       "10                                          [133]   \n",
       "11                                      [136 133]   \n",
       "12                                            [5]   \n",
       "13                                         [1888]   \n",
       "14                                          [490]   \n",
       "15                                          [168]   \n",
       "16                                  [123 122  17]   \n",
       "17                                  [12 13 61 14]   \n",
       "18                                           [92]   \n",
       "19                                          [224]   \n",
       "20                                           [22]   \n",
       "21                                          [405]   \n",
       "22                                         [1641]   \n",
       "23                                          [591]   \n",
       "24                                          [298]   \n",
       "25                                          [130]   \n",
       "26                                          [192]   \n",
       "27                                          [663]   \n",
       "28                                        [51 94]   \n",
       "29                                          [113]   \n",
       "30                                           [84]   \n",
       "31                                          [556]   \n",
       "32                                      [232 230]   \n",
       "33                                          [457]   \n",
       "34                                           [62]   \n",
       "35                                  [106 103 109]   \n",
       "36                                  [362 321 361]   \n",
       "37                                           [97]   \n",
       "38                                  [ 86 114  88]   \n",
       "39                                           [18]   \n",
       "40                                          [118]   \n",
       "41                                          [207]   \n",
       "42                                      [512 429]   \n",
       "43                                        [45 44]   \n",
       "44                                      [351 383]   \n",
       "45                                  [162 163 324]   \n",
       "46                                      [739 736]   \n",
       "47                                  [251 257 249]   \n",
       "48                                          [994]   \n",
       "49                                           [47]   \n",
       "50  [653 730 668 712 714 707 708 758 718 674 663]   \n",
       "51                                     [41 36 44]   \n",
       "52                                    [1380 1335]   \n",
       "53                               [ 996 1013  992]   \n",
       "54                                          [297]   \n",
       "55                          [975 973 970 947 903]   \n",
       "56                          [1855 1895 1899 1854]   \n",
       "57                                      [216 221]   \n",
       "58                                          [112]   \n",
       "59                                      [156 165]   \n",
       "\n",
       "                           C_community  Ratio_obs_sim_top_vol  Mut_in_gene  \\\n",
       "0                           [0. 0. 0.]                    inf          350   \n",
       "1                              [0. 0.]              90.675025          162   \n",
       "2                                 [0.]              65.432984          157   \n",
       "3                        [0. 0. 0. 0.]              56.050080          125   \n",
       "4                                 [0.]              34.644913           66   \n",
       "5                                 [0.]              27.512761          109   \n",
       "6                                 [0.]              21.778911           50   \n",
       "7                           [0. 0. 0.]              17.554267           40   \n",
       "8                                 [0.]              16.310410           47   \n",
       "9                                 [0.]              16.281568           70   \n",
       "10                                [0.]              14.697380           37   \n",
       "11                             [0. 0.]              12.687365           32   \n",
       "12                                [0.]              12.566828           35   \n",
       "13                                [0.]              11.923891           72   \n",
       "14                                [0.]              11.314176           27   \n",
       "15                                [0.]               9.380280           22   \n",
       "16                          [0. 0. 0.]               7.941862           26   \n",
       "17                       [0. 0. 0. 0.]               7.673587           22   \n",
       "18                                [0.]               7.395317           34   \n",
       "19                                [0.]               6.702145           11   \n",
       "20                                [0.]               6.456593           19   \n",
       "21                                [0.]               6.312924           21   \n",
       "22                                [0.]               6.294105           32   \n",
       "23                                [0.]               6.164195           22   \n",
       "24                                [0.]               6.009603           72   \n",
       "25                                [0.]               5.838393           26   \n",
       "26                                [0.]               5.249385           17   \n",
       "27                                [0.]               4.347430           39   \n",
       "28                             [0. 0.]               4.032435           39   \n",
       "29                                [0.]               3.630678           22   \n",
       "30                                [0.]               3.527611           22   \n",
       "31                                [0.]               3.492045           30   \n",
       "32                             [0. 0.]               3.418260           23   \n",
       "33                                [0.]               3.397278           46   \n",
       "34                                [0.]               3.210614           61   \n",
       "35                          [0. 0. 0.]               3.132542           26   \n",
       "36                          [0. 0. 0.]               3.097745            9   \n",
       "37                                [0.]               3.076529            8   \n",
       "38                          [0. 0. 0.]               3.064409            7   \n",
       "39                                [0.]               2.982920           23   \n",
       "40                                [0.]               2.905756            8   \n",
       "41                                [0.]               2.746241           41   \n",
       "42                             [0. 0.]               2.713499           19   \n",
       "43                             [0. 0.]               2.695163            6   \n",
       "44                             [0. 0.]               2.666975           12   \n",
       "45                          [0. 0. 0.]               2.665633           11   \n",
       "46                             [0. 0.]               2.661947           29   \n",
       "47                          [0. 0. 0.]               2.655064           16   \n",
       "48                                [0.]               2.637831           17   \n",
       "49                                [0.]               2.624367           16   \n",
       "50  [0. 0. 1. 1. 1. 1. 1. 1. 1. 1. 1.]               2.624205           34   \n",
       "51                          [0. 0. 0.]               2.618848           13   \n",
       "52                             [0. 0.]               2.523232           54   \n",
       "53                          [0. 0. 0.]               2.488469           18   \n",
       "54                                [0.]               2.457333           26   \n",
       "55                    [0. 0. 0. 0. 0.]               2.386572           47   \n",
       "56                       [0. 0. 0. 0.]               2.367992           38   \n",
       "57                             [0. 0.]               2.202616           14   \n",
       "58                                [0.]               3.158133           16   \n",
       "59                             [0. 0.]               2.717943            9   \n",
       "\n",
       "    Clust_mut  Mut_in_top_vol  Mut_in_top_comm_vol  \n",
       "0       341.0           341.0                341.0  \n",
       "1       156.0           156.0                156.0  \n",
       "2       151.0           151.0                151.0  \n",
       "3        99.0            99.0                 99.0  \n",
       "4        61.0            61.0                 61.0  \n",
       "5        70.0            70.0                 70.0  \n",
       "6        42.0            42.0                 42.0  \n",
       "7        38.0            38.0                 38.0  \n",
       "8        34.0            34.0                 34.0  \n",
       "9        44.0            44.0                 44.0  \n",
       "10       35.0            35.0                 35.0  \n",
       "11       28.0            28.0                 28.0  \n",
       "12       31.0            31.0                 31.0  \n",
       "13       28.0            28.0                 28.0  \n",
       "14       22.0            22.0                 22.0  \n",
       "15       17.0            17.0                 17.0  \n",
       "16       23.0            23.0                 23.0  \n",
       "17       20.0            20.0                 20.0  \n",
       "18       23.0            23.0                 23.0  \n",
       "19       10.0            10.0                 10.0  \n",
       "20       13.0            13.0                 13.0  \n",
       "21       14.0            14.0                 14.0  \n",
       "22       13.0            13.0                 13.0  \n",
       "23       12.0            12.0                 12.0  \n",
       "24       16.0            16.0                 16.0  \n",
       "25       16.0            16.0                 16.0  \n",
       "26       13.0            13.0                 13.0  \n",
       "27       14.0            14.0                 14.0  \n",
       "28       12.0            12.0                 12.0  \n",
       "29       12.0            12.0                 12.0  \n",
       "30       10.0            10.0                 10.0  \n",
       "31       10.0            10.0                 10.0  \n",
       "32        8.0             8.0                  8.0  \n",
       "33       10.0            10.0                 10.0  \n",
       "34        9.0             9.0                  9.0  \n",
       "35       10.0            10.0                 10.0  \n",
       "36        6.0             6.0                  6.0  \n",
       "37        6.0             6.0                  6.0  \n",
       "38        6.0             6.0                  6.0  \n",
       "39       10.0            10.0                 10.0  \n",
       "40        7.0             7.0                  7.0  \n",
       "41       10.0            10.0                 10.0  \n",
       "42        7.0             7.0                  7.0  \n",
       "43        5.0             5.0                  5.0  \n",
       "44        6.0             6.0                  6.0  \n",
       "45        6.0             6.0                  6.0  \n",
       "46        7.0             7.0                  7.0  \n",
       "47        6.0             6.0                  6.0  \n",
       "48        5.0             5.0                  5.0  \n",
       "49        6.0             6.0                  6.0  \n",
       "50       22.0             9.0                  9.0  \n",
       "51        6.0             6.0                  6.0  \n",
       "52        9.0             9.0                  9.0  \n",
       "53        6.0             6.0                  6.0  \n",
       "54        7.0             7.0                  7.0  \n",
       "55        8.0             8.0                  8.0  \n",
       "56        7.0             7.0                  7.0  \n",
       "57        5.0             5.0                  5.0  \n",
       "58        8.0             8.0                  8.0  \n",
       "59        5.0             5.0                  5.0  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mouse_gene_result[:60].drop(columns=['Uniprot_ID', \"pval\", 'pLDDT_top_vol', 'pLDDT_top_comm_vol', 'Status', 'Cancer', 'Cohort', \"Tot_samples\", \"Samples_in_top_vol\", \"Samples_in_top_comm_vol\", \"Clust_res\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "58f35390-4442-48c2-894d-36ecc692ba0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Uniprot_ID</th>\n",
       "      <th>F</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Mut_in_gene</th>\n",
       "      <th>Mut_in_res</th>\n",
       "      <th>Mut_in_vol</th>\n",
       "      <th>Obs_anomaly</th>\n",
       "      <th>Ratio_obs_sim</th>\n",
       "      <th>pval</th>\n",
       "      <th>...</th>\n",
       "      <th>Tot_samples</th>\n",
       "      <th>Samples_in_vol</th>\n",
       "      <th>Samples_in_comm_vol</th>\n",
       "      <th>Mut_in_comm_vol</th>\n",
       "      <th>Res_in_comm</th>\n",
       "      <th>pLDDT_res</th>\n",
       "      <th>pLDDT_vol</th>\n",
       "      <th>pLDDT_comm_vol</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Cohort</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1520401A03Rik</td>\n",
       "      <td>A0A0J9YVH6</td>\n",
       "      <td>1</td>\n",
       "      <td>561</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.135332</td>\n",
       "      <td>0.943700</td>\n",
       "      <td>0.5018</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.62</td>\n",
       "      <td>33.105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1520401A03Rik</td>\n",
       "      <td>A0A0J9YVH6</td>\n",
       "      <td>1</td>\n",
       "      <td>559</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.134568</td>\n",
       "      <td>0.995312</td>\n",
       "      <td>0.4354</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.59</td>\n",
       "      <td>33.105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1520401A03Rik</td>\n",
       "      <td>A0A0J9YVH6</td>\n",
       "      <td>1</td>\n",
       "      <td>503</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.064167</td>\n",
       "      <td>0.509408</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.93</td>\n",
       "      <td>33.930</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1520401A03Rik</td>\n",
       "      <td>A0A0J9YVH6</td>\n",
       "      <td>1</td>\n",
       "      <td>596</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.062337</td>\n",
       "      <td>0.527303</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.94</td>\n",
       "      <td>48.940</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1520401A03Rik</td>\n",
       "      <td>A0A0J9YVH6</td>\n",
       "      <td>1</td>\n",
       "      <td>462</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.061768</td>\n",
       "      <td>0.577629</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.65</td>\n",
       "      <td>38.650</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117281</th>\n",
       "      <td>Zzz3</td>\n",
       "      <td>Q6KAQ7</td>\n",
       "      <td>1</td>\n",
       "      <td>611</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023702</td>\n",
       "      <td>0.457698</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.06</td>\n",
       "      <td>47.060</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117282</th>\n",
       "      <td>Zzz3</td>\n",
       "      <td>Q6KAQ7</td>\n",
       "      <td>1</td>\n",
       "      <td>605</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.023573</td>\n",
       "      <td>0.471635</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.99</td>\n",
       "      <td>48.990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117283</th>\n",
       "      <td>Zzz3</td>\n",
       "      <td>Q6KAQ7</td>\n",
       "      <td>1</td>\n",
       "      <td>220</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.022321</td>\n",
       "      <td>0.462874</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.26</td>\n",
       "      <td>40.260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117284</th>\n",
       "      <td>Zzz3</td>\n",
       "      <td>Q6KAQ7</td>\n",
       "      <td>1</td>\n",
       "      <td>509</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.020447</td>\n",
       "      <td>0.439525</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.91</td>\n",
       "      <td>88.910</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117285</th>\n",
       "      <td>Zzz3</td>\n",
       "      <td>Q6KAQ7</td>\n",
       "      <td>1</td>\n",
       "      <td>705</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.020061</td>\n",
       "      <td>0.447043</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90.45</td>\n",
       "      <td>90.450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liver</td>\n",
       "      <td>Liver_strelka_2017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117286 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene  Uniprot_ID  F  Pos  Mut_in_gene  Mut_in_res  \\\n",
       "0       1520401A03Rik  A0A0J9YVH6  1  561            9           1   \n",
       "1       1520401A03Rik  A0A0J9YVH6  1  559            9           1   \n",
       "2       1520401A03Rik  A0A0J9YVH6  1  503            9           1   \n",
       "3       1520401A03Rik  A0A0J9YVH6  1  596            9           1   \n",
       "4       1520401A03Rik  A0A0J9YVH6  1  462            9           1   \n",
       "...               ...         ... ..  ...          ...         ...   \n",
       "117281           Zzz3      Q6KAQ7  1  611           18           1   \n",
       "117282           Zzz3      Q6KAQ7  1  605           18           1   \n",
       "117283           Zzz3      Q6KAQ7  1  220           18           1   \n",
       "117284           Zzz3      Q6KAQ7  1  509           18           1   \n",
       "117285           Zzz3      Q6KAQ7  1  705           18           1   \n",
       "\n",
       "        Mut_in_vol  Obs_anomaly  Ratio_obs_sim    pval  ...  Tot_samples  \\\n",
       "0                2     0.135332       0.943700  0.5018  ...            9   \n",
       "1                2     0.134568       0.995312  0.4354  ...            9   \n",
       "2                1     0.064167       0.509408  1.0000  ...            9   \n",
       "3                1     0.062337       0.527303  1.0000  ...            9   \n",
       "4                1     0.061768       0.577629  1.0000  ...            9   \n",
       "...            ...          ...            ...     ...  ...          ...   \n",
       "117281           1     0.023702       0.457698  1.0000  ...           15   \n",
       "117282           1     0.023573       0.471635  1.0000  ...           15   \n",
       "117283           1     0.022321       0.462874  1.0000  ...           15   \n",
       "117284           1     0.020447       0.439525  1.0000  ...           15   \n",
       "117285           1     0.020061       0.447043  1.0000  ...           15   \n",
       "\n",
       "        Samples_in_vol  Samples_in_comm_vol  Mut_in_comm_vol  Res_in_comm  \\\n",
       "0                    2                  NaN              NaN          NaN   \n",
       "1                    2                  NaN              NaN          NaN   \n",
       "2                    1                  NaN              NaN          NaN   \n",
       "3                    1                  NaN              NaN          NaN   \n",
       "4                    1                  NaN              NaN          NaN   \n",
       "...                ...                  ...              ...          ...   \n",
       "117281               1                  NaN              NaN          NaN   \n",
       "117282               1                  NaN              NaN          NaN   \n",
       "117283               1                  NaN              NaN          NaN   \n",
       "117284               1                  NaN              NaN          NaN   \n",
       "117285               1                  NaN              NaN          NaN   \n",
       "\n",
       "        pLDDT_res  pLDDT_vol  pLDDT_comm_vol  Cancer              Cohort  \n",
       "0           31.62     33.105             NaN   Liver  Liver_strelka_2017  \n",
       "1           34.59     33.105             NaN   Liver  Liver_strelka_2017  \n",
       "2           33.93     33.930             NaN   Liver  Liver_strelka_2017  \n",
       "3           48.94     48.940             NaN   Liver  Liver_strelka_2017  \n",
       "4           38.65     38.650             NaN   Liver  Liver_strelka_2017  \n",
       "...           ...        ...             ...     ...                 ...  \n",
       "117281      47.06     47.060             NaN   Liver  Liver_strelka_2017  \n",
       "117282      48.99     48.990             NaN   Liver  Liver_strelka_2017  \n",
       "117283      40.26     40.260             NaN   Liver  Liver_strelka_2017  \n",
       "117284      88.91     88.910             NaN   Liver  Liver_strelka_2017  \n",
       "117285      90.45     90.450             NaN   Liver  Liver_strelka_2017  \n",
       "\n",
       "[117286 rows x 24 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mouse_pos_result"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f957b425-d980-41d0-8acc-2bdc9cf8661b",
   "metadata": {},
   "source": [
    "# LiftOver from mm38 to mm39"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
