{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "21688689-7a7a-4aab-8260-f6d2ef6bb21c",
   "metadata": {},
   "source": [
    "# Liver mouse preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9953e776-c098-4118-be46-fa446155e06b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "45cf56d2-c640-4552-8349-b68745cb9d43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>do9806_93131_N1</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>do9414_91603_N1</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>do9786_91932_N7</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>do9219_90886_N4</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>do9429_91618_N2</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55406524</th>\n",
       "      <td>do9818_93243_N1</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55406525</th>\n",
       "      <td>do9878_91943_N1</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55406526</th>\n",
       "      <td>do9864_94315_N10</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55406527</th>\n",
       "      <td>do9403_91415_N2</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55406528</th>\n",
       "      <td>do9379_91216_N1</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55406529 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #Uploaded_variation Feature         Consequence Protein_position  \\\n",
       "0            do9806_93131_N1       -  intergenic_variant                -   \n",
       "1            do9414_91603_N1       -  intergenic_variant                -   \n",
       "2            do9786_91932_N7       -  intergenic_variant                -   \n",
       "3            do9219_90886_N4       -  intergenic_variant                -   \n",
       "4            do9429_91618_N2       -  intergenic_variant                -   \n",
       "...                      ...     ...                 ...              ...   \n",
       "55406524     do9818_93243_N1       -  intergenic_variant                -   \n",
       "55406525     do9878_91943_N1       -  intergenic_variant                -   \n",
       "55406526    do9864_94315_N10       -  intergenic_variant                -   \n",
       "55406527     do9403_91415_N2       -  intergenic_variant                -   \n",
       "55406528     do9379_91216_N1       -  intergenic_variant                -   \n",
       "\n",
       "         Amino_acids SYMBOL CANONICAL  \n",
       "0                  -      -         -  \n",
       "1                  -      -         -  \n",
       "2                  -      -         -  \n",
       "3                  -      -         -  \n",
       "4                  -      -         -  \n",
       "...              ...    ...       ...  \n",
       "55406524           -      -         -  \n",
       "55406525           -      -         -  \n",
       "55406526           -      -         -  \n",
       "55406527           -      -         -  \n",
       "55406528           -      -         -  \n",
       "\n",
       "[55406529 rows x 7 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_table(\"/workspace/projects/clustering_3d/o3d_analysys/datasets/input/cancer_mouse/mm39/vep/liver_strelka_mm39.vep.tsv.gz\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "53f0b621-0564-440b-ac3a-02a53d90f8e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'G'"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_df[seq_df[\"Gene\"] == \"Braf\"].Seq.values[0][637-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "ad7b5af1-a687-4ff9-bd98-33470572acd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "braf_df = braf_df[braf_df[\"CANONICAL\"] == \"YES\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "a92b1df0-cfb7-47ff-95d2-08fbef07f3a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>CANONICAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>do12458_90793_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>do12459_91948_N4</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>do12460_92057_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>do12461_92057_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>do12465_92058_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>do12466_92104_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>do12469_92105_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>do12471_92105_N4</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>do12473_91100_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>do12477_93091_N8</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>do12480_93129_N7</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>do9156_89985_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>do9157_89985_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>do9160_89986_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>do9165_90004_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>do9166_90004_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>do9167_90004_N4</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>do9170_90022_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>do9172_90170_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>do9174_90400_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>do9176_90400_N5</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>do9178_90446_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>do9182_90732_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>do9184_90791_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>do9202_90803_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>do9205_90803_N6</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>do9210_90810_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>do9211_90810_N4</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>do9214_90880_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>do9216_90885_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>do9217_90886_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>do9219_90886_N4</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>do9230_90887_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>do9232_90888_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>do9234_90888_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>do9235_90888_N4</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>do9236_91097_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>do9237_91097_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>do9240_91100_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>do9242_91100_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>do9243_91100_N4</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>do9248_91113_N6</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>do9250_91208_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>do9370_91213_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>do9371_91213_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>do9378_91215_N9</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>do9380_91216_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>do9381_91216_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>do9383_91216_N5</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>do9384_91216_N6</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>do9385_91216_N7</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>do9386_90793_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>do9387_90793_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>do9389_90888_N5</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>do9391_91215_N5</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>do9398_91218_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>do9401_91414_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>do9402_91414_N2</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>do9404_91415_N3</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>do9406_91539_N1</td>\n",
       "      <td>ENSMUST00000002487</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>637</td>\n",
       "      <td>V/E</td>\n",
       "      <td>Braf</td>\n",
       "      <td>YES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #Uploaded_variation             Feature       Consequence  \\\n",
       "32     do12458_90793_N2  ENSMUST00000002487  missense_variant   \n",
       "34     do12459_91948_N4  ENSMUST00000002487  missense_variant   \n",
       "36     do12460_92057_N1  ENSMUST00000002487  missense_variant   \n",
       "38     do12461_92057_N2  ENSMUST00000002487  missense_variant   \n",
       "40     do12465_92058_N3  ENSMUST00000002487  missense_variant   \n",
       "42     do12466_92104_N1  ENSMUST00000002487  missense_variant   \n",
       "44     do12469_92105_N1  ENSMUST00000002487  missense_variant   \n",
       "46     do12471_92105_N4  ENSMUST00000002487  missense_variant   \n",
       "48     do12473_91100_N3  ENSMUST00000002487  missense_variant   \n",
       "50     do12477_93091_N8  ENSMUST00000002487  missense_variant   \n",
       "52     do12480_93129_N7  ENSMUST00000002487  missense_variant   \n",
       "54      do9156_89985_N1  ENSMUST00000002487  missense_variant   \n",
       "56      do9157_89985_N2  ENSMUST00000002487  missense_variant   \n",
       "58      do9160_89986_N2  ENSMUST00000002487  missense_variant   \n",
       "60      do9165_90004_N1  ENSMUST00000002487  missense_variant   \n",
       "62      do9166_90004_N2  ENSMUST00000002487  missense_variant   \n",
       "64      do9167_90004_N4  ENSMUST00000002487  missense_variant   \n",
       "66      do9170_90022_N1  ENSMUST00000002487  missense_variant   \n",
       "68      do9172_90170_N2  ENSMUST00000002487  missense_variant   \n",
       "70      do9174_90400_N1  ENSMUST00000002487  missense_variant   \n",
       "72      do9176_90400_N5  ENSMUST00000002487  missense_variant   \n",
       "74      do9178_90446_N3  ENSMUST00000002487  missense_variant   \n",
       "76      do9182_90732_N2  ENSMUST00000002487  missense_variant   \n",
       "78      do9184_90791_N2  ENSMUST00000002487  missense_variant   \n",
       "80      do9202_90803_N3  ENSMUST00000002487  missense_variant   \n",
       "82      do9205_90803_N6  ENSMUST00000002487  missense_variant   \n",
       "84      do9210_90810_N3  ENSMUST00000002487  missense_variant   \n",
       "86      do9211_90810_N4  ENSMUST00000002487  missense_variant   \n",
       "88      do9214_90880_N2  ENSMUST00000002487  missense_variant   \n",
       "90      do9216_90885_N2  ENSMUST00000002487  missense_variant   \n",
       "92      do9217_90886_N2  ENSMUST00000002487  missense_variant   \n",
       "94      do9219_90886_N4  ENSMUST00000002487  missense_variant   \n",
       "96      do9230_90887_N1  ENSMUST00000002487  missense_variant   \n",
       "98      do9232_90888_N1  ENSMUST00000002487  missense_variant   \n",
       "100     do9234_90888_N3  ENSMUST00000002487  missense_variant   \n",
       "102     do9235_90888_N4  ENSMUST00000002487  missense_variant   \n",
       "104     do9236_91097_N1  ENSMUST00000002487  missense_variant   \n",
       "106     do9237_91097_N2  ENSMUST00000002487  missense_variant   \n",
       "108     do9240_91100_N1  ENSMUST00000002487  missense_variant   \n",
       "110     do9242_91100_N3  ENSMUST00000002487  missense_variant   \n",
       "112     do9243_91100_N4  ENSMUST00000002487  missense_variant   \n",
       "114     do9248_91113_N6  ENSMUST00000002487  missense_variant   \n",
       "116     do9250_91208_N1  ENSMUST00000002487  missense_variant   \n",
       "118     do9370_91213_N1  ENSMUST00000002487  missense_variant   \n",
       "120     do9371_91213_N2  ENSMUST00000002487  missense_variant   \n",
       "122     do9378_91215_N9  ENSMUST00000002487  missense_variant   \n",
       "124     do9380_91216_N2  ENSMUST00000002487  missense_variant   \n",
       "126     do9381_91216_N3  ENSMUST00000002487  missense_variant   \n",
       "128     do9383_91216_N5  ENSMUST00000002487  missense_variant   \n",
       "130     do9384_91216_N6  ENSMUST00000002487  missense_variant   \n",
       "132     do9385_91216_N7  ENSMUST00000002487  missense_variant   \n",
       "134     do9386_90793_N2  ENSMUST00000002487  missense_variant   \n",
       "136     do9387_90793_N2  ENSMUST00000002487  missense_variant   \n",
       "138     do9389_90888_N5  ENSMUST00000002487  missense_variant   \n",
       "140     do9391_91215_N5  ENSMUST00000002487  missense_variant   \n",
       "142     do9398_91218_N3  ENSMUST00000002487  missense_variant   \n",
       "144     do9401_91414_N1  ENSMUST00000002487  missense_variant   \n",
       "146     do9402_91414_N2  ENSMUST00000002487  missense_variant   \n",
       "148     do9404_91415_N3  ENSMUST00000002487  missense_variant   \n",
       "150     do9406_91539_N1  ENSMUST00000002487  missense_variant   \n",
       "\n",
       "    Protein_position Amino_acids SYMBOL CANONICAL  \n",
       "32               637         V/E   Braf       YES  \n",
       "34               637         V/E   Braf       YES  \n",
       "36               637         V/E   Braf       YES  \n",
       "38               637         V/E   Braf       YES  \n",
       "40               637         V/E   Braf       YES  \n",
       "42               637         V/E   Braf       YES  \n",
       "44               637         V/E   Braf       YES  \n",
       "46               637         V/E   Braf       YES  \n",
       "48               637         V/E   Braf       YES  \n",
       "50               637         V/E   Braf       YES  \n",
       "52               637         V/E   Braf       YES  \n",
       "54               637         V/E   Braf       YES  \n",
       "56               637         V/E   Braf       YES  \n",
       "58               637         V/E   Braf       YES  \n",
       "60               637         V/E   Braf       YES  \n",
       "62               637         V/E   Braf       YES  \n",
       "64               637         V/E   Braf       YES  \n",
       "66               637         V/E   Braf       YES  \n",
       "68               637         V/E   Braf       YES  \n",
       "70               637         V/E   Braf       YES  \n",
       "72               637         V/E   Braf       YES  \n",
       "74               637         V/E   Braf       YES  \n",
       "76               637         V/E   Braf       YES  \n",
       "78               637         V/E   Braf       YES  \n",
       "80               637         V/E   Braf       YES  \n",
       "82               637         V/E   Braf       YES  \n",
       "84               637         V/E   Braf       YES  \n",
       "86               637         V/E   Braf       YES  \n",
       "88               637         V/E   Braf       YES  \n",
       "90               637         V/E   Braf       YES  \n",
       "92               637         V/E   Braf       YES  \n",
       "94               637         V/E   Braf       YES  \n",
       "96               637         V/E   Braf       YES  \n",
       "98               637         V/E   Braf       YES  \n",
       "100              637         V/E   Braf       YES  \n",
       "102              637         V/E   Braf       YES  \n",
       "104              637         V/E   Braf       YES  \n",
       "106              637         V/E   Braf       YES  \n",
       "108              637         V/E   Braf       YES  \n",
       "110              637         V/E   Braf       YES  \n",
       "112              637         V/E   Braf       YES  \n",
       "114              637         V/E   Braf       YES  \n",
       "116              637         V/E   Braf       YES  \n",
       "118              637         V/E   Braf       YES  \n",
       "120              637         V/E   Braf       YES  \n",
       "122              637         V/E   Braf       YES  \n",
       "124              637         V/E   Braf       YES  \n",
       "126              637         V/E   Braf       YES  \n",
       "128              637         V/E   Braf       YES  \n",
       "130              637         V/E   Braf       YES  \n",
       "132              637         V/E   Braf       YES  \n",
       "134              637         V/E   Braf       YES  \n",
       "136              637         V/E   Braf       YES  \n",
       "138              637         V/E   Braf       YES  \n",
       "140              637         V/E   Braf       YES  \n",
       "142              637         V/E   Braf       YES  \n",
       "144              637         V/E   Braf       YES  \n",
       "146              637         V/E   Braf       YES  \n",
       "148              637         V/E   Braf       YES  \n",
       "150              637         V/E   Braf       YES  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "braf_df[braf_df[\"Protein_position\"] == \"637\"][:60]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2a9fe4b-691e-47fa-a16b-faaf6894ab3b",
   "metadata": {},
   "source": [
    "## Parse .txt file to .vcf for VEP (if not using liftover)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3393e21-4d4e-4088-b439-5ee958a75a90",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sort_vcf(vcf_file):\n",
    "    \"\"\"\n",
    "    Sort the vcf file from by chromosomes and position.\n",
    "    \"\"\"\n",
    "    \n",
    "    df = pd.DataFrame(vcf_file)\n",
    "\n",
    "    # Define the custom sort order for Chromosome\n",
    "    chromosome_order = {'1': 1, '2': 2, '3': 3, '4': 4, '5': 5, '6': 6, '7': 7, '8': 8, '9': 9, '10': 10,\n",
    "                        '11': 11, '12': 12, '13': 13, '14': 14, '15': 15, '16': 16, '17': 17, '18': 18, '19': 19,\n",
    "                        '20': 20, '21': 21, 'X': 22, 'Y': 23, 'M': 24}\n",
    "\n",
    "    # Convert 'Chromosome' column to a temporary categorical column for sorting\n",
    "    df['#CHROM'] = df['#CHROM'].astype(str)\n",
    "    df['Chromosome_sort'] = df['#CHROM'].map(chromosome_order)\n",
    "    # Sort and drop temporary col\n",
    "    sorted_df = df.sort_values(by=['Chromosome_sort', 'POS'], ignore_index=True)\n",
    "    sorted_df.drop('Chromosome_sort', axis=1, inplace=True)\n",
    "\n",
    "    return sorted_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c1b3083-b2ad-42b5-8fff-74e0fe6e5933",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_input = \"../Liver_DEN_strelka_July2017_mm10.txt\"\n",
    "path_output = \"liver_strelka_mm39.vcf\"\n",
    "\n",
    "mouse_df = pd.read_csv(path_input, sep=\"\\t\")\n",
    "mouse_df = mouse_df[[\"CHROMOSOME\", \"POSITION\", \"SAMPLE\", \"REF\", \"ALT\"]]\n",
    "mouse_df = mouse_df.rename(columns={\"CHROMOSOME\" : \"#CHROM\",\n",
    "                                    \"POSITION\" : \"POS\",\n",
    "                                    \"SAMPLE\" : \"ID\"})\n",
    "mouse_df = sort_vcf(mouse_df)\n",
    "mouse_df.to_csv(fpath_output, sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d72b24fc-8c46-44f0-aa11-4f6a6e9e9a47",
   "metadata": {},
   "source": [
    "## Liftover from mm10 to mm39"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ed25622-7801-42ca-89d3-ca0ad035abfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "from pyliftover import LiftOver\n",
    "\n",
    "\n",
    "# Convert coordinates\n",
    "def liftover_vcf(df, path_chain_file):\n",
    "    \n",
    "    lo = LiftOver(path_chain_file)\n",
    "\n",
    "    df['CHROM_mm39'] = np.nan\n",
    "    df['POS_mm39'] = np.nan\n",
    "    for index, row in df.iterrows():\n",
    "        chrom = f\"chr{str(row['CHROMOSOME']).lower()}\"\n",
    "        pos = int(row['POSITION'])\n",
    "        \n",
    "        # Perform liftover\n",
    "        result = lo.convert_coordinate(chrom, pos - 1)  # convert to 0-based for liftover\n",
    "        if result:\n",
    "            new_chrom, new_pos = result[0][0], result[0][1] + 1  # convert back to 1-based\n",
    "            df.at[index, 'CHROM_mm39'] = new_chrom\n",
    "            df.at[index, 'POS_mm39'] = new_pos\n",
    "        else:\n",
    "            df.at[index, 'CHROM_mm39'] = np.nan\n",
    "            df.at[index, 'POS_mm39'] = np.nan\n",
    "            \n",
    "    if (index + 1) % 1000 == 0:\n",
    "            print(f\"Processed {index + 1} rows\")\n",
    "            \n",
    "    return df\n",
    "\n",
    "\n",
    "def main(input_path, chain_file_path, output_path):\n",
    "    \n",
    "    # Load VCF\n",
    "    print(\"Loading mutations..\")\n",
    "    vcf_df = pd.read_table(input_path, low_memory=False)\n",
    "\n",
    "    # Liftover\n",
    "    print(\"Starting liftover..\")\n",
    "    vcf_mm39 = liftover_vcf(vcf_df, chain_file_path)\n",
    "\n",
    "    # Filter out unmapped entries\n",
    "    print(\"Filtering NAs..\")\n",
    "    len_a = len(vcf_mm39)\n",
    "    vcf_mm39 = vcf_mm39.dropna(subset=['CHROM_mm39', 'POS_mm39'])\n",
    "    len_b = len(vcf_mm39)\n",
    "    print(f\"Filtered {len_a - len_b} NAs\")\n",
    "\n",
    "    # Save to new VCF file\n",
    "    vcf_mm39['#CHROM'] = vcf_mm39['CHROM_mm39'].str.replace(\"chr\", \"\").astype(int)\n",
    "    vcf_mm39['POS'] = vcf_mm39['POS_mm39'].astype(int)\n",
    "    vcf_mm39 = vcf_mm39.rename(columns={\"SAMPLE\" : \"ID\"})\n",
    "    output_columns = ['#CHROM', 'POS', 'ID', 'REF', 'ALT']\n",
    "    print(\"Saving..\")\n",
    "    vcf_mm39[output_columns].to_csv(output_path, sep='\\t', index=False)\n",
    "\n",
    "\n",
    "# if __name__ == '__main__':\n",
    "#     input_path = sys.argv[1]\n",
    "#     chain_file_path = sys.argv[2]\n",
    "#     output_path = sys.argv[3]\n",
    "    \n",
    "#     print(\"Input:\", input_path)\n",
    "#     print(\"Chain:\", chain_file_path)\n",
    "#     print(\"Output:\", output_path)\n",
    "    \n",
    "#     main(input_path, chain_file_path, output_path)\n",
    "    \n",
    "main(\"Liver_DEN_strelka_July2017_mm10.txt\", \"mm10ToMm39.over.chain.gz\", \"liver_strelka_mm39.vcf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f130964-940c-49a2-a7fd-00d4327f55b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "python3 liftover.py Liver_DEN_strelka_July2017_mm10.txt mm10ToMm39.over.chain.gz liver_strelka_mm39.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fd75226-f58c-414f-9b09-3e40a4c2f4ad",
   "metadata": {},
   "source": [
    "## Run VEP and parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9298aef-75e1-47bb-b517-8468a2cbc11e",
   "metadata": {},
   "outputs": [],
   "source": [
    "/workspace/projects/clustering_3d/clustering_3d/build/preprocessing/vep_and_parse_mouse.sh liver_strelka_mm39.vcf liver_strelka_mm39.vep.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77b229fa-2009-455b-9c63-043f8fc1148c",
   "metadata": {},
   "source": [
    "## Convert tsv for VEP to tsv for bgsignature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b37c2140-6e7f-48ab-9605-be006e9a7f29",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_input = \"liver_strelka_mm39.vcf\"\n",
    "path_output = \"liver_strelka_mm39.in.tsv.gz\"\n",
    "\n",
    "df = pd.read_table(path_input)\n",
    "df = df.rename(columns={\"#CHROM\" : \"CHROMOSOME\",\n",
    "                           \"POS\" : \"POSITION\",\n",
    "                           \"ID\" : \"SAMPLE\"})\n",
    "df[\"SAMPLE\"] = df.pop(\"SAMPLE\")\n",
    "\n",
    "df.to_csv(path_output, sep=\"\\t\", compression=\"gzip\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a7d6463-49fb-4481-a097-c24d89a3c4ce",
   "metadata": {},
   "source": [
    "## Generate mut profile"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8e5c7e0-ec09-4177-999e-dd2bb74242c0",
   "metadata": {},
   "source": [
    "### Get regions file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "812257d7-a5c3-4cd1-87c9-aea7912e33ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from bgreference import refseq\n",
    "\n",
    "CHR = [str(i) for i in range(1, 20)] + ['X', 'Y', 'M']\n",
    "\n",
    "\n",
    "def compute_sizes(genome, kmer):\n",
    "    sizes = []\n",
    "    for chr_ in CHR:\n",
    "        seq = refseq(genome, f\"chr{chr_}\", start=(1 + kmer // 2), size=None)\n",
    "        sizes.append(tuple(map(str, (chr_, 1 + kmer // 2, len(seq) - kmer // 2))))\n",
    "    return sizes\n",
    "\n",
    "\n",
    "def write(sizes):\n",
    "    print('\\t'.join(('CHROMOSOME', 'START', 'END')))\n",
    "    for s in sizes:\n",
    "        print('\\t'.join(s))\n",
    "\n",
    "# if __name__ == '__main__':\n",
    "#     genome = sys.argv[1]\n",
    "#     kmer = int(sys.argv[2])\n",
    "#     write(compute_sizes(genome, kmer))\n",
    "\n",
    "write(compute_sizes(genome=\"mm39\", kmer=3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d08c05f-898a-485b-864f-24b5e6488370",
   "metadata": {},
   "outputs": [],
   "source": [
    "python3 /workspace/projects/clustering_3d/o3d_analysys/datasets/input/cancer_mouse/raw_data/liver_strelka_mm39/get_regions_file.py mm39 3 > mm39_wg_regions.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66cbd451-dda5-4941-8546-1813facfdaf3",
   "metadata": {},
   "source": [
    "### Run bgsignature content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54841f3a-7b7c-49f2-b42a-2666a23e49fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda env activate bgsignature\n",
    "\n",
    "bgsignature count -r mm39_wg_regions.tsv -s 3 -g mm39 --exclude-N --collapse -o wg.counts.gz --cores 8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf0d5b99-a746-44b0-9e68-974e664bc28d",
   "metadata": {},
   "source": [
    "### Run bgsignature normalizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06a0ea38-cab4-482d-98a8-541b730d1c66",
   "metadata": {},
   "outputs": [],
   "source": [
    "bgsignature normalize -m liver_strelka_mm39.in.tsv.gz -r mm39_wg_regions.tsv --normalize wg.counts.gz -s 3 -g mm39 --collapse --cores 8 -o liver_strelka_mm39.sig.json"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "o3d_nb",
   "language": "python",
   "name": "o3d_nb"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
